
---
title: 'biocMultiAssay: exploring unified representations of multi-omics data'
author: "Vincent J. Carey, Levi Waldron et al."
date: "October 15, 2014"
output:
  html_document:
    highlight: pygments
    number_sections: yes
    theme: united
    toc: yes
  pdf_document:
    toc: yes
---

# Background

An increasingly common use case involves a set of
samples or patients who provide measurements on
multiple data types, such as gene expression,
genotype, miRNA abundance.  It will frequently
be the case that not all samples will contribute to
all assays, so some sparsity in the
set of
samples $\times$ assays is expected.

# Basic demonstrative resources

Here are some very simple manipulations with
TCGA ovarian cancer data.  The data sizes are manageable
enough that the loadHub function is used to deserialize
all relevant data.

```{r chk,cache=FALSE}
suppressPackageStartupMessages(library(biocMultiAssay))
#
# crude way of enumerating RDA files planted in extdata
#
ov = dir(system.file("extdata/tcga_ov",
   package="biocMultiAssay"), full=TRUE, pattern="\\.rda$")
drop = grep("pheno", ov)
if (length(drop)>0) {
  pdpath=ov[drop]
  ov=ov[-drop]
  }
#
# informal labels for constituents
#
tags = c("ov RNA-seq", "ov agilent", "ov mirna", "ov affy", "ov CNV gistic",
  "ov methy 450k")
#
# construct expt instances from ExpressionSets
#
exptlist = lapply(1:length(ov), function(x) new("expt",
     serType="RData", assayPath=ov[x], tag=tags[x]))
#
# populate an eHub, witha master phenotype data frame
#
ovhub = new("eHub", hub=exptlist, masterSampleData = get(load(pdpath)))
ovhub
```

This is a lightweight representation of the scope of data
identified to an eHub.  We have as well a class that
includes materializations of all the experimental data.
Constructing it is currently slow.

```{r doload, cache=TRUE}
lovhub = loadHub(ovhub)
```
```{r lkloaded}
lovhub
format(object.size(lovhub), units = "Mb")
```

This is a heavy representation but manageable at this level
of data reduction.

We can determine the set of common identifiers.
```{r getcomm}
allid = lapply(lovhub@elist, sampleNames)
commids = allid[[1]]
for (i in 2:length(allid))
 commids = intersect(commids, allid[[i]])
length(commids)
```

We can now generate the MultiAssayExperiment instance with only the
common samples.
```{r docomm}
locomm = lovhub
locomm@elist = lapply(locomm@elist, function(x) x[,commids])
locomm
```

Where to put these abstractions for both the light and
heavy representations is a point of discussion.

# createMA shortcut function for creating a MultiAssayExperiment

```{r createMA}
ovlist <- lapply(ov, function(x) get(load(x)))
names(ovlist) <- tags

lovhub2 <- createMA(masterpheno=pData(ovlist[[2]]), objlist=ovlist, drop=TRUE)
lovhub2
object.size(lovhub2)
```
