---
title: "Creating an eHub for TCGA ovarian cancer dataset"
author: "Levi Waldron"
date: "December 10, 2014"
output:
  html_document:
    number_sections: yes
    toc: yes
---

```{r, echo=FALSE, message=FALSE}
library(biocMultiAssay)
```

Data are now downloaded from Amazon S3 bucket:
```{r download}
downloadData(datadir="downloads")
```

# Links
* Compiled html vignette from [Rpubs](http://rpubs.com/lwaldron/ovc_ehub).  
* .rda files from [Dropbox](https://www.dropbox.com/sh/po98a41h779j6mv/AACMbyHQgc2Yd0oi8XJL5iNIa?dl=0).

# GISTIC copy number

## GISTIC centered CN estimates in an ExpressionSet

This yields `eset.cn` in working environment.
```{r loadg}
load("downloads/tcga_ov/ovc_eset_copynumber_gistic_2013_01_16.rda")
```

## Called loss/gain/amplification as SummarizedExperiment

Read data:
```{r}
cn <- read.table("downloads/tcga_ov/raw/all_lesions.conf_99.txt", 
                 row.names=1, header=TRUE, sep="\t", as.is=TRUE)
```

Extract copy number ranges, using region limits. First the chromosomes:
```{r}
chr <- sub(":.+", "", cn$Region.Limits)
head(chr)
```

Then the ranges:
```{r}
head(cn$Region.Limits, 1)
ranges <- sub(".+?:", "", cn$Region.Limits)
head(ranges, 1)
ranges <- sub("\\(.+", "", ranges)
head(ranges, 1)
ranges <- data.frame(t(do.call(cbind, strsplit(ranges, "-"))), stringsAsFactors=FALSE)
ranges[, 1] <- as.integer(ranges[, 1])
ranges[, 2] <- as.integer(ranges[, 2])
colnames(ranges) <- c("start", "end")
head(ranges, 1)
```

Construct a GRanges object:
```{r, message=FALSE}
library(GenomicRanges)
gr <- GRanges(seqnames=Rle(chr),
        ranges=IRanges(start=ranges$start, end=ranges$end),
        q.values=cn$q.values)
genome(gr) <- "hg19"
gr
```

Create a matrix of peak calls, with the patient component of the barcode as column names:
```{r}
cna.dat <- cn[, -1:-8]
colnames(cna.dat) <- sub("\\.[0-9]*[ABCD].+", "", colnames(cna.dat))
cna.dat$X <- NULL
table(sapply(cna.dat, class))
cna.dat <- as.matrix(cna.dat)
class(cna.dat)
dim(cna.dat)
cna.dat[1:3, 1:3]
```

And finally create a SummarizedExperiment with the copy number data:
```{r}
cna01.se <- SummarizedExperiment(assays=SimpleList(cna.dat), 
                                  rowData=gr)
colnames(cna01.se) <- colnames(cna.dat)
save(cna01.se, file="cna01.se.rda")
cna01.se
```

## Raw copy number as GRangesList

This section creates a GRangesList object containing raw copy number numeric (non-integer) values.  _Levi: I am unsure what class to use when there is a different set of GRanges for each patient, with copy number data associated with each of those ranges_.  In this section I create a GRangesList, even though I don't imagine that is the correct solution.

```{r}
cn <- read.table("downloads/tcga_ov/raw/focal_input.seg.txt",
                 header=TRUE, sep="\t", as.is=TRUE)
cn$id <- make.names(sub("\\-[0-9]*[ABCD].+", "", cn[, 1]))
```

Make a GRangesList, one element per patient.  
```{r}
grl <- lapply(unique(cn$id), function(x){
  cn1 <- cn[cn$id==x, ]
  gr <- GRanges(seqnames=Rle(paste0("chr", cn1$Chromosome)),
          ranges=IRanges(start=cn1$Start.bp, end=cn1$End.bp),
          Num.Markers=cn1$Num.Markers,
          Seg.CN=cn1$Seg.CN)
  genome(gr) <- "hg19"
  gr
})
names(grl) <- unique(cn$id)
cna.grl <- GRangesList(grl)
save(cna.grl, file="cna.grl.rda")
```

## Raw copy number as SummarizedExperiment

For methods to obtain the file `all_data_by_genes.txt`, see the downloads/raw directory.

```{r}
if (!file.exists("all_data_by_genes.txt.gz")) {
download.file("http://s3.amazonaws.com/cancerhub/all_data_by_genes.txt.gz",
              destfile="all_data_by_genes.txt.gz", method="wget")
}
cn <- read.table(gzfile("all_data_by_genes.txt.gz"), as.is=TRUE, row.names=1, sep="\t", header=TRUE)
colnames(cn) <- sub("\\.[0-9]*[ABCD].+", "", colnames(cn))
cn <- cn[, -1:-2]
```

Get ranges for each gene:
```{r, message=FALSE, warning=FALSE}
rowd <- key2GRanges(rownames(cn))
hasrowd = match(names(rowd), rownames(cn), nomatch=0)
ex = as.matrix(cn[hasrowd,])
stopifnot(identical(names(rowd), rownames(ex)))
cna.se <- SummarizedExperiment(assays=SimpleList(ex), 
                                  rowData=rowd)
colnames(cna.se) <- colnames(ex)
save(cna.se, file="cna.se.rda")
cna.se
```

# Load Gene Expression Data

## As ExpressionSet

```{r}
library(affy)
load("downloads/tcga_ov/TCGA_eset.rda")
expr.eset <- TCGA_eset
pdata <- pData(TCGA_eset)
save(expr.eset, file="expr.eset.rda")
save(pdata, file="pdata.rda")
```

## As SummarizedExperiment

```{r, warning=FALSE, message=FALSE}
featureNames(expr.eset) <- as.character(featureData(expr.eset)$probeset)
expr.eset@annotation <- "hgu133a"
expr.se <- exs2se(expr.eset)
save(expr.se, file="expr.se.rda")
```

# construct expt instances from ExpressionSets
```{r}
tags = c("ov affy", "ov CNV gistic")
elist = lapply(1:length(tags), function(x) new("expt",
        serType="RData", assayPath="", tag=tags[x], sampleDataPath=""))
```

# populate an eHub, with a master phenotype data frame 

```{r}
ovhub2 = new("eHub", hub=elist, masterSampleData = pData(TCGA_eset))
```

