---
title: "Correlation between copy number and gene expression for quality control of ovarian cancer experiments"
output: html_document
---

# Motivation

The ovarian cancer genome is characterized profound defects in DNA damage repair mechanisms, including almost universal loss of TP53 function.  This results in extensive somatic mutation mutation and copy number variation.  Copy number variation is associated with changes in gene expression, and lack of this association can be used as a multi-assay quality control measure that is not possible from a single assay.   This case study demonstrates the identification of TCGA ovarian cancer Affymetrix microarray sample mix-ups through correlation to copy number, and confirms these mix-ups by comparison to Agilent microarray and level III RNA-seq data.  

# Methods
```{r, echo=FALSE, message=FALSE, warning=FALSE}
library(biocMultiAssay)
library(Biobase)
library(sva)
library(impute)
```

Data are now downloaded from Amazon S3 bucket:
```{r download}
datadir <- "downloads"
downloadData(datadir)
```

First, load TCGA ovarian cancer ExpressionSets from the biocMultiAssay package:
```{r}
load("downloads/tcga_ov/ovc_eset_copynumber_gistic_2013_01_16.rda")
load("downloads/tcga_ov/TCGA_eset.rda")
load("downloads/tcga_ov/TCGA.agilent_eset.rda")
```

These ExpressionSets have HGNC symbols as featureNames, and some have featureData.
```{r}
head(pData(featureData(eset.cn)))
head(pData(featureData(TCGA_eset)))
```

I invisibly define a function "corFinder()" to calculate pair-wise correlations between samples using the expr() slots of a list of two ExpressionSets.  It subsets each ExpressionSet to the intersection of common features, performs ComBat batch correction using each experiment as a batch, then calculates column-wise Pearson correlation between all pairs of samples.
```{r, echo=FALSE}
corFinder <- function  # Calculate pair-wise correlations between samples using the expr() slots of a list of two ExpressionSets.
### This function acts as a wrapper around ComBat (sva package) and
### cor(), to calculate pairwise correlations within one or between
### two ExpressionSets.
(eset.pair,
### a list of ExpressionSets, with two elements.  If the two
### elements are identical, return the correlation matrix for pairs
### of samples in the first element.  If not identical, return pairs
### between the two elements.
separator=":",
### Separator between dataset name and sample name.  Dataset names are
### added to sample names to keep track of dataset of origin.
use.ComBat=TRUE,
### Use the sva::ComBat function for batch correction of the expr()
### data between the two datasets.
...
### Extra arguments passed to the cor() function.
 ){
    if((class(eset.pair) != "ExpressionSet")
       & (class(eset.pair) != "list" || length(eset.pair) > 2))
        stop("eset.pair should be a list of two esets")
    if(is.null(names(eset.pair)))
        names(eset.pair) <- paste("eset", 1:2, sep="")
    if( identical(class(eset.pair), "list") & !identical(eset.pair[[1]], eset.pair[[2]]) ){
        genes.intersect <- intersect(featureNames(eset.pair[[1]]), featureNames(eset.pair[[2]]))
        samples.intersect <- intersect(sampleNames(eset.pair[[1]]), sampleNames(eset.pair[[2]]))
        for (i in 1:length(eset.pair)){
            eset.pair[[i]] <- eset.pair[[i]][genes.intersect, samples.intersect]
            sampleNames(eset.pair[[i]]) <- paste(names(eset.pair)[i], sampleNames(eset.pair[[i]]), sep=separator)
        }
        ## Calculate correlation matrix for a pair of ExpressionSets:
        if(use.ComBat){
            big.matrix <- do.call(cbind, lapply(eset.pair, exprs))
            batch.var <- lapply(names(eset.pair), function(x) rep(x, ncol(eset.pair[[x]])))
            batch.var <- do.call(c, batch.var)
            big.matrix.combat <- sva::ComBat(big.matrix, mod=model.matrix(~(rep(1, length(batch.var)))), batch=batch.var)
            matrix.pair <- lapply(unique(batch.var), function(x) big.matrix.combat[, batch.var %in% x])
            names(matrix.pair) <- unique(batch.var)
        }else{
            matrix.pair <- lapply(eset.pair, exprs)
            names(matrix.pair) <- names(eset.pair)
        }
        cormat <- cor(matrix.pair[[1]], matrix.pair[[2]], ...)
    }else{
        ##Calculate correlation matrix for a single ExpressionSet:
        if(identical(class(eset.pair), "list")){
            matrix.one <- exprs(eset.pair[[1]])
            colnames(matrix.one) <- paste(names(eset.pair)[1], colnames(matrix.one), sep=separator)
        }else{
            matrix.one <- exprs(eset.pair)
        }
        cormat <- cor(matrix.one, ...)
        cormat[!upper.tri(cormat)] <- NA  ##NA for diagonal
    }
    return(cormat)
###   Returns a matrix of sample-wise Pearson Correlations.
}
```

```{r, cache=TRUE, message=FALSE, results='hide'}
cn.cor <- corFinder(list(microarray=TCGA_eset, cn=eset.cn), use.ComBat=FALSE)
agilent.cor <- corFinder(list(microarray=TCGA_eset, agilent=TCGA.agilent_eset), use.ComBat=TRUE)
```

Just to give an idea of what these results look like, they are a matrix of sample pairwise Pearson correlations:

```{r}
dim(cn.cor)
cn.cor[1:4, 1:4]
```

```{r, echo=FALSE}
## fix row and column names
colnames(cn.cor) <- sub(".+:", "", colnames(cn.cor))
rownames(cn.cor) <- sub(".+:", "", rownames(cn.cor))
colnames(agilent.cor) <- sub(".+:", "", colnames(agilent.cor))
rownames(agilent.cor) <- sub(".+:", "", rownames(agilent.cor))
```

# Initial identification of sample mix-ups

This histogram shows the correlation between genome-wide copy number and expression for each sample.  The lower mode is caused by mixed-up samples.
```{r, echo=FALSE}
par(mar=c(4, 4, 0.1, 0.1))
hist(diag(cn.cor), breaks="FD", main="", xlab="correlation")
abline(v=0.066, col="red", lw=2)
arrows(x0=0.06, y0=9, x1=0.002, y1=9, col="red", lw=2, length=0.1)
text(x=0.025, y=c(25, 18), labels=c("flagged", "samples"), col="red")
```

These samples are also suspect from low correlation between Affy - Agilent technical replicates:

```{r, echo=FALSE}
hist(diag(agilent.cor), breaks="FD", xlim=c(0.59, 1))
abline(v=0.87, col="red", lw=2)
```

There are more systematic ways to determine a cutoff, but for now I just set it manually and count the number of likely sample mix-ups by Agilent and by Copy Number.  First the number of suspect samples identified by CN that are also assayed by Agilent:

```{r}
samples.intersect <- intersect(colnames(cn.cor), colnames(agilent.cor))
sum(diag(cn.cor[samples.intersect, samples.intersect]) < 0.066)
```

And the number of samples identified as suspect by both Affy/CN and Affy/Agilent:
```{r}
sum(diag(cn.cor[samples.intersect, samples.intersect]) < 0.066 & diag(agilent.cor[samples.intersect, samples.intersect]) < 0.87)
```

# Using methylation for QC

Methylation is not nearly so informative for QC in this case, but I
left this analysis in place anyways.  It still identifies three
outliers, two of which overlap with the CN outliers.  There is
overall a strong negative correlation between expression and
methylation (mean cor ~ -0.35)

```{r}
load("downloads/tcga_ov/ovc_eset_methylation.rda")
```

Drop features with too many missing values, then impute:

```{r, cache=TRUE, message=FALSE, results='hide'}
eset.meth <- eset.meth[esApply(eset.meth, 1, function(x) sum(is.na(x))) < 300, ]
exprs(eset.meth) <- impute.knn(exprs(eset.meth))$data
```

Repeat the analysis with Affy expression and methylation:

```{r, cache=TRUE}
meth.cor <- corFinder(list(microarray=TCGA_eset, meth=eset.meth), use.ComBat=FALSE)
```

```{r}
colnames(meth.cor) <- sub(".+:", "", colnames(meth.cor))
rownames(meth.cor) <- sub(".+:", "", rownames(meth.cor))
hist(diag(meth.cor), breaks="FD")
abline(v=-0.2, col="red")
arrows(x0=-0.2, y0=20, x1=-0.1, y1=20, col="red", lw=2, length=0.1)
text(x=-0.15, y=c(40, 30), labels=c("flagged", "samples"), col="red")
```

There are only three samples flagged:
```{r}
sum(diag(meth.cor) > -0.2)
```

And two of these were also identified by lack of correlation between CN and expression:

```{r}
samples.intersect <- intersect(colnames(meth.cor), colnames(cn.cor))
summary(diag(meth.cor[samples.intersect, samples.intersect]) > -0.2 & diag(cn.cor[samples.intersect, samples.intersect]) < 0.066)
```

# Summary

## Short-term objectives
* create SummarizedExperiment objects for these TCGA ovarian cancer datasets
* use biocMultiAssay objects to repeat this correlation analysis
 
## Longer-term future directions
I would like to establish for which assay combinations in which cancer types this is possible for, using all of TCGA.  It is likely to be cancer specific; for example, many prostate cancers have ``quiet'' genomes with little copy number variation, and the procedure demonstrated in this vignette would not work.  However correlation between other regulatory elements and downstream elements may be effective, for example:
* promotor region methylation and gene expression
* somatic mutation of particular genes and associated gene expression
* microRNA expression and target gene expression
* transcription factor and target gene expression

It is easy to simulate sample mix-ups simply by swapping sample labels, making testing feasible in datasets that do not contain actual mix-ups.  