
\documentclass{article}

<<style, eval=TRUE, echo=FALSE, results=tex>>=
BiocStyle::latex()
@

\begin{document}

\title{biocMultiAssay: exploratory work on managing
multiple assays on a group of samples}
\author{Bioconductor MultiAssay SIG}
\maketitle

\tableofcontents

\section{Introduction}

An increasingly common use case involves a set of
samples or patients who provide measurements on
multiple data types, such as gene expression,
genotype, miRNA abundance.  It will frequently
be the case that not all samples will contribute to
all assays, so some sparsity in the
set of 
samples $\times$ assays is expected.

\section{Basic demonstrative resources}

<<lookDataCode>>=
library(biocMultiAssay)
downloadData(datadir="downloads")
dir("downloads/tcga_ov")
getClass("TTMergedDataSet") # from Tim Triche 25 July 2013
@

Here are some very simple manipulations with
TCGA ovarian cancer data.  The data sizes are manageable
enough that the loadHub function is used to deserialize
all relevant data.

<<chk,cache=TRUE>>=
ov = dir("downloads/tcga_ov", pattern="\\.rda$", full=TRUE)
drop = grep("pheno", ov)
if (length(drop)>0) {
  pdpath=ov[drop]
  ov=ov[-drop]
  }
tags = sub("\\.rda", "", basename(ov))
elist = lapply(1:length(ov), function(x) new("expt", 
     serType="RData", assayPath=ov[x], tag=tags[x], sampleDataPath=ov[x]))
ovhub = new("eHub", hub=elist, masterSampleData = get(load(pdpath)))
lo = loadHub(ovhub)
@
<<reset>>=
ovhub@allids = as.character(unique(unlist(lapply(lo@elist, sampleNames))))
ovhub
sapply(lo@elist, function(x) dim(pData(x)))
@


\section{Session information}

<<sessi>>=
toLatex(sessionInfo())
@

\end{document}
