<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Save a MultiAssayExperiment class object to HDF5 and Rds files — saveHDF5MultiAssayExperiment • MultiAssayExperiment</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Save a MultiAssayExperiment class object to HDF5 and Rds files — saveHDF5MultiAssayExperiment"><meta name="description" content="This function takes a MultiAssayExperiment object and uses the assays
functionality to obtain data matrices out of the experiments. These are
then saved into the .h5 file format. This function relies heavily on
the HDF5Array package whose installation is required before use.
saveHDF5MultiAssayExpeirment preserves the classes contained in the
ExperimentList with the exception of matrix which is
converted to HDF5Matrix. Internal SummarizedExperiment assays are
converted to HDF5-backed assays as in
HDF5Array::saveHDF5SummarizedExperiment. SummarizedExperiment
objects with multiple i-th assays will have the first assay take
precedence and others assays will be dropped with a warning.
If the first assay in a SummarizedExperiment contains an array,
the array is preserved in the process of saving and loading the
HDF5-backed MultiAssayExperiment."><meta property="og:description" content="This function takes a MultiAssayExperiment object and uses the assays
functionality to obtain data matrices out of the experiments. These are
then saved into the .h5 file format. This function relies heavily on
the HDF5Array package whose installation is required before use.
saveHDF5MultiAssayExpeirment preserves the classes contained in the
ExperimentList with the exception of matrix which is
converted to HDF5Matrix. Internal SummarizedExperiment assays are
converted to HDF5-backed assays as in
HDF5Array::saveHDF5SummarizedExperiment. SummarizedExperiment
objects with multiple i-th assays will have the first assay take
precedence and others assays will be dropped with a warning.
If the first assay in a SummarizedExperiment contains an array,
the array is preserved in the process of saving and loading the
HDF5-backed MultiAssayExperiment."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="default" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MultiAssayExperiment</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.30.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Quick Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><a class="dropdown-item" href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a></li>
    <li><a class="dropdown-item" href="../articles/MultiAssayExperiment.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a></li>
    <li><a class="dropdown-item" href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a></li>
  </ul></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news"><li><a class="dropdown-item" href="../news/index.html">Release Notes</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="nav-link" href="https://github.com/waldronlab/MultiAssayExperiment"><span class="fa fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Save a MultiAssayExperiment class object to HDF5 and Rds files</h1>
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/RELEASE_3_19/R/saveHDF5MultiAssayExperiment.R"><code>R/saveHDF5MultiAssayExperiment.R</code></a></small>
      <div class="d-none name"><code>HDF5MultiAssayExperiment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function takes a <code>MultiAssayExperiment</code> object and uses the <code>assays</code>
functionality to obtain data matrices out of the experiments. These are
then saved into the <code>.h5</code> file format. This function relies heavily on
the <code>HDF5Array</code> package whose installation is required before use.
<code>saveHDF5MultiAssayExpeirment</code> preserves the classes contained in the
<code><a href="ExperimentList.html">ExperimentList</a></code> with the exception of <code>matrix</code> which is
converted to <code>HDF5Matrix</code>. Internal <code>SummarizedExperiment</code> assays are
converted to HDF5-backed assays as in
<code><a href="https://rdrr.io/pkg/HDF5Array/man/saveHDF5SummarizedExperiment.html" class="external-link">HDF5Array::saveHDF5SummarizedExperiment</a></code>. <code>SummarizedExperiment</code>
objects with multiple <code>i</code>-th assays will have the first assay take
precedence and others assays will be dropped with a warning.
If the first assay in a <code>SummarizedExperiment</code> contains an array,
the array is preserved in the process of saving and loading the
HDF5-backed <code>MultiAssayExperiment</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">saveHDF5MultiAssayExperiment</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  dir <span class="op">=</span> <span class="st">"h5_mae"</span>,</span>
<span>  prefix <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  replace <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  chunkdim <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  level <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  as.sparse <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">loadHDF5MultiAssayExperiment</span><span class="op">(</span>dir <span class="op">=</span> <span class="st">"h5_mae"</span>, prefix <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>A <code><a href="MultiAssayExperiment.html">MultiAssayExperiment</a></code> object or derivative</p></dd>


<dt id="arg-dir">dir<a class="anchor" aria-label="anchor" href="#arg-dir"></a></dt>
<dd><p>The path (as a single string) to the directory where to save the
HDF5-based <code><a href="MultiAssayExperiment.html">MultiAssayExperiment</a></code> object or to load it from.</p>
<p>When saving, the directory will be created if it doesn't already exist.
If the directory already exists and no prefix is specified and
<code>replace</code> is set to <code>TRUE</code>, then it's replaced with an
empty directory.</p></dd>


<dt id="arg-prefix">prefix<a class="anchor" aria-label="anchor" href="#arg-prefix"></a></dt>
<dd><p>An optional prefix to add to the names of the files created
inside <code>dir</code>. This allows saving more than one object in the same
directory. When the prefix is <code>NULL</code>, the name of the <code>x</code> input
<code>MultiAssayExperiment</code> is used. To avoid the default setting
use an empty character string i.e., <code>""</code>. An underscore (<code>_</code>) is
appended to the prefix when provided; therefore, typical inputs should be
words, e.g., "test".</p></dd>


<dt id="arg-replace">replace<a class="anchor" aria-label="anchor" href="#arg-replace"></a></dt>
<dd><p>When no prefix is specified, should a pre-existing directory be replaced
    with a new empty one? The content of the pre-existing directory will
    be lost!</p></dd>


<dt id="arg-chunkdim-level">chunkdim, level<a class="anchor" aria-label="anchor" href="#arg-chunkdim-level"></a></dt>
<dd><p>The dimensions of the chunks and the compression level to use for
    writing the assay data to disk.</p>
<p>Passed to the internal calls to <code>writeHDF5Array</code>.
    See <code>?<a href="https://rdrr.io/pkg/HDF5Array/man/writeHDF5Array.html" class="external-link">writeHDF5Array</a></code> for more information.</p></dd>


<dt id="arg-as-sparse">as.sparse<a class="anchor" aria-label="anchor" href="#arg-as-sparse"></a></dt>
<dd><p>Whether the assay data should be flagged as sparse or not. If set to
    <code>NA</code> (the default), then the specific <code>as.sparse</code> value to
    use for each assay is determined by calling <code>is_sparse()</code> on them.</p>
<p>Passed to the internal calls to <code>writeHDF5Array</code>.
    See <code>?<a href="https://rdrr.io/pkg/HDF5Array/man/writeHDF5Array.html" class="external-link">writeHDF5Array</a></code> for more information and an
    IMPORTANT NOTE.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Set to <code>TRUE</code> to make the function display progress.</p>
<p>In the case of <code>saveHDF5MultiAssayExperiment()</code>, <code>verbose</code>
is set to <code>NA</code> by default, in which case verbosity is controlled
by <code>DelayedArray.verbose.block.processing</code> option. Setting
<code>verbose</code> to <code>TRUE</code> or <code>FALSE</code> overrides the option.</p></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"miniACC"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">testDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"test_mae"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">saveHDF5MultiAssayExperiment</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">miniACC</span>, dir <span class="op">=</span> <span class="va">testDir</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span>, replace <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 1/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ok</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 1/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 2/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ok</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 2/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 3/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ok</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 3/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 4/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ok</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 4/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 5/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ok</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 5/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Serialize MultiAssayExperiment object to RDS file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpXNWNbz/test_mae/miniACC_mae.rds</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## inspect the files in the dir</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">testDir</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "miniACC_experiments.h5" "miniACC_mae.rds"       </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">loadHDF5MultiAssayExperiment</span><span class="op">(</span></span></span>
<span class="r-in"><span>    dir <span class="op">=</span> <span class="va">testDir</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 5 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 5:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] gistict: SummarizedExperiment with 198 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] Mutations: HDF5Matrix with 97 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## remove example files</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">testDir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Marcel Ramos, Martin Morgan, Vincent J Carey, Levi Waldron.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

