<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Save a MultiAssayExperiment class object to HDF5 and Rds files — saveHDF5MultiAssayExperiment • MultiAssayExperiment</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Save a MultiAssayExperiment class object to HDF5 and Rds files — saveHDF5MultiAssayExperiment"><meta property="og:description" content="
This function takes a `MultiAssayExperiment` object and uses the `assays`
functionality to obtain data matrices out of the experiments. These are
then saved into the `.h5` file format. This function relies heavily on
the `HDF5Array` package whose installation is required before use.
`saveHDF5MultiAssayExpeirment` preserves the classes contained in the
\linkS4class{ExperimentList} with the exception of `matrix` which is
converted to `HDF5Matrix`. Internal `SummarizedExperiment` assays are
converted to HDF5-backed assays as in
`HDF5Array::saveHDF5SummarizedExperiment`. `SummarizedExperiment`
objects with multiple `i`-th assays will have the first assay take
precedence and others assays will be dropped with a warning.
If the first assay in a `SummarizedExperiment` contains an array,
the array is preserved in the process of saving and loading the
HDF5-backed `MultiAssayExperiment`.

"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiAssayExperiment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.22.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/MultiAssayExperiment.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a>
    </li>
    <li>
      <a href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../news/index.html">Release Notes</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/waldronlab/MultiAssayExperiment" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Save a MultiAssayExperiment class object to HDF5 and Rds files</h1>
    <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/HEAD/R/saveHDF5MultiAssayExperiment.R" class="external-link"><code>R/saveHDF5MultiAssayExperiment.R</code></a></small>
    <div class="hidden name"><code>HDF5MultiAssayExperiment.Rd</code></div>
    </div>

    <div class="ref-description">
    
<div class="sourceCode"><pre><code>This function takes a `MultiAssayExperiment` object and uses the `assays`
functionality to obtain data matrices out of the experiments. These are
then saved into the `.h5` file format. This function relies heavily on
the `HDF5Array` package whose installation is required before use.
`saveHDF5MultiAssayExpeirment` preserves the classes contained in the
\linkS4class{ExperimentList} with the exception of `matrix` which is
converted to `HDF5Matrix`. Internal `SummarizedExperiment` assays are
converted to HDF5-backed assays as in
`HDF5Array::saveHDF5SummarizedExperiment`. `SummarizedExperiment`
objects with multiple `i`-th assays will have the first assay take
precedence and others assays will be dropped with a warning.
If the first assay in a `SummarizedExperiment` contains an array,
the array is preserved in the process of saving and loading the
HDF5-backed `MultiAssayExperiment`.
</code></pre></div>

    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">saveHDF5MultiAssayExperiment</span><span class="op">(</span>
  <span class="va">x</span>,
  dir <span class="op">=</span> <span class="st">"h5_mae"</span>,
  prefix <span class="op">=</span> <span class="cn">NULL</span>,
  replace <span class="op">=</span> <span class="cn">FALSE</span>,
  chunkdim <span class="op">=</span> <span class="cn">NULL</span>,
  level <span class="op">=</span> <span class="cn">NULL</span>,
  as.sparse <span class="op">=</span> <span class="cn">NA</span>,
  verbose <span class="op">=</span> <span class="cn">NA</span>
<span class="op">)</span>

<span class="fu">loadHDF5MultiAssayExperiment</span><span class="op">(</span>dir <span class="op">=</span> <span class="st">"h5_mae"</span>, prefix <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>A <a href="MultiAssayExperiment-class.html">MultiAssayExperiment</a> object or derivative</p></dd>
<dt>dir</dt>
<dd><p>The path (as a single string) to the directory where to save the
HDF5-based <a href="MultiAssayExperiment-class.html">MultiAssayExperiment</a> object or to load it from.</p><div class="sourceCode"><pre><code>When saving, the directory will be created if it doesn't already exist.
If the directory already exists and no prefix is specified and
`replace` is set to `TRUE`, then it's replaced with an
empty directory.
</code></pre></div></dd>
<dt>prefix</dt>
<dd><p>An optional prefix to add to the names of the files created
inside <code>dir</code>. Allows saving more than one object in the same
directory. When the prefix is <code>NULL</code>, the name of the
<code>MultiAssayExperiment</code> object is used. To avoid the default setting
use an empty character string i.e., <code>""</code>.</p></dd>
<dt>replace</dt>
<dd><p>When no prefix is specified, should a pre-existing directory be replaced
    with a new empty one? The content of the pre-existing directory will
    be lost!</p></dd>
<dt>chunkdim</dt>
<dd><p>The dimensions of the chunks and the compression level to use for
    writing the assay data to disk.</p>
<p>Passed to the internal calls to <code>writeHDF5Array</code>.
    See <code>?<a href="https://rdrr.io/pkg/HDF5Array/man/writeHDF5Array.html" class="external-link">writeHDF5Array</a></code> for more information.</p></dd>
<dt>level</dt>
<dd><p>The dimensions of the chunks and the compression level to use for
    writing the assay data to disk.</p>
<p>Passed to the internal calls to <code>writeHDF5Array</code>.
    See <code>?<a href="https://rdrr.io/pkg/HDF5Array/man/writeHDF5Array.html" class="external-link">writeHDF5Array</a></code> for more information.</p></dd>
<dt>as.sparse</dt>
<dd><p>Whether the assay data should be flagged as sparse or not. If set to
    <code>NA</code> (the default), then the specific <code>as.sparse</code> value to
    use for each assay is determined by calling <code>is_sparse()</code> on them.</p>
<p>Passed to the internal calls to <code>writeHDF5Array</code>.
    See <code>?<a href="https://rdrr.io/pkg/HDF5Array/man/writeHDF5Array.html" class="external-link">writeHDF5Array</a></code> for more information and an
    IMPORTANT NOTE.</p></dd>
<dt>verbose</dt>
<dd><p>Set to <code>TRUE</code> to make the function display progress.</p><div class="sourceCode"><pre><code>In the case of `saveHDF5MultiAssayExperiment()`, `verbose`
is set to `NA` by default, in which case verbosity is controlled
by `DelayedArray:::get_verbose_block_processing()`. Setting
`verbose` to `TRUE` or `FALSE` overrides this.
</code></pre></div></dd>
</dl></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"miniACC"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">testDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"test_mae"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">saveHDF5MultiAssayExperiment</span><span class="op">(</span></span>
<span class="r-in">    <span class="va">miniACC</span>, dir <span class="op">=</span> <span class="va">testDir</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span>, replace <span class="op">=</span> <span class="cn">TRUE</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 1/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / Reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 1/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 2/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / Reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 2/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 3/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / Reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 3/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 4/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / Reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 4/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start writing assay 5/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> / Reading and realizing block 1/1 ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> \ Writing it ... </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> OK</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finished writing assay 5/5 to HDF5 file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_experiments.h5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Serialize MultiAssayExperiment object to RDS file:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   /tmp/RtmpDn9MMF/test_mae/miniACC_mae.rds</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## inspect the files in the dir</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">testDir</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "miniACC_experiments.h5" "miniACC_mae.rds"       </span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">loadHDF5MultiAssayExperiment</span><span class="op">(</span></span>
<span class="r-in">    dir <span class="op">=</span> <span class="va">testDir</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 5 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 5:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] gistict: SummarizedExperiment with 198 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] Mutations: HDF5Matrix with 97 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## remove example files</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">testDir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

