<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Construct an integrative representation of multi-omic data with
MultiAssayExperiment — MultiAssayExperiment • MultiAssayExperiment</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Construct an integrative representation of multi-omic data with
MultiAssayExperiment — MultiAssayExperiment" />
<meta property="og:description" content="The constructor function for the MultiAssayExperiment-class combines
multiple data elements from the different hierarchies of data
(study, experiments, and samples). It can create instances where neither
a sampleMap or a colData set is provided. Please see the
MultiAssayExperiment API documentation for more information." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiAssayExperiment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/MultiAssayExperiment.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a>
    </li>
    <li>
      <a href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../news/index.html">Release Notes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/waldronlab/MultiAssayExperiment">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Construct an integrative representation of multi-omic data with
<code>MultiAssayExperiment</code></h1>
    <small class="dont-index">Source: <a href='https://github.com/waldronlab/MultiAssayExperiment/blob/master/R/MultiAssayExperiment-class.R'><code>R/MultiAssayExperiment-class.R</code></a></small>
    <div class="hidden name"><code>MultiAssayExperiment.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The constructor function for the <a href='MultiAssayExperiment-class.html'>MultiAssayExperiment-class</a> combines
multiple data elements from the different hierarchies of data
(study, experiments, and samples). It can create instances where neither
a <code>sampleMap</code> or a <code>colData</code> set is provided. Please see the
MultiAssayExperiment API documentation for more information.</p>
    </div>

    <pre class="usage"><span class='fu'>MultiAssayExperiment</span><span class='op'>(</span>
  experiments <span class='op'>=</span> <span class='fu'><a href='ExperimentList.html'>ExperimentList</a></span><span class='op'>(</span><span class='op'>)</span>,
  colData <span class='op'>=</span> <span class='fu'>S4Vectors</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html'>DataFrame</a></span><span class='op'>(</span><span class='op'>)</span>,
  sampleMap <span class='op'>=</span> <span class='fu'>S4Vectors</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html'>DataFrame</a></span><span class='op'>(</span>assay <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='op'>)</span>, primary <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/character.html'>character</a></span><span class='op'>(</span><span class='op'>)</span>, colname <span class='op'>=</span>
    <span class='fu'><a href='https://rdrr.io/r/base/character.html'>character</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>,
  metadata <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>,
  drops <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>experiments</th>
      <td><p>A <code>list</code> or <a href='ExperimentList.html'>ExperimentList</a> of all
combined experiments</p></td>
    </tr>
    <tr>
      <th>colData</th>
      <td><p>A <code>DataFrame</code> or <code>data.frame</code> of
characteristics for all biological units</p></td>
    </tr>
    <tr>
      <th>sampleMap</th>
      <td><p>A <code>DataFrame</code> or <code>data.frame</code> of assay names,
sample identifiers, and colname samples</p></td>
    </tr>
    <tr>
      <th>metadata</th>
      <td><p>An optional argument of "ANY" class (usually list) for
content describing the experiments</p></td>
    </tr>
    <tr>
      <th>drops</th>
      <td><p>A <code>list</code> of unmatched information
(included after subsetting)</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>MultiAssayExperiment</code> object that can store
experiment and phenotype data</p>
    <h2 class="hasAnchor" id="coldata"><a class="anchor" href="#coldata"></a>colData</h2>

    

<p>The `colData` input can be either `DataFrame` or `data.frame` with
subsequent coercion to DataFrame. The rownames in the `colData` must match
the colnames in the experiments if no sampleMap is provided.</p>
    <h2 class="hasAnchor" id="experiments"><a class="anchor" href="#experiments"></a>experiments</h2>

    

<p>The `experiments` input can be of class SimpleList or `list`.
This input becomes the <code><a href='ExperimentList.html'>ExperimentList</a></code>. Each element of the
input `list` or `List` must be named, rectangular with two dimensions, and
have `dimnames`.</p>
    <h2 class="hasAnchor" id="samplemap"><a class="anchor" href="#samplemap"></a>sampleMap</h2>

    

<p>The <code><a href='MultiAssayExperiment-methods.html'>sampleMap</a></code> can either be input as `DataFrame` or
`data.frame` with eventual coercion to `DataFrame`. The `sampleMap` relates
biological units and biological measurements within each assay. Each row in
the `sampleMap` is a single such link. The standard column names of the
`sampleMap` are "assay", "primary", and "colname".  Note that the "assay"
column is a factor corresponding to the names of each experiment in the
`ExperimentList`. In the case where these names do not match between the
`sampleMap` and the experiments, the documented experiments in the
`sampleMap` take precedence and experiments are dropped by the harmonization
procedure. The constructor function will generate a `sampleMap` in the case
where it is not provided and this method may be a 'safer' alternative for
creating the `MultiAssayExperiment` (so long as the rownames are identical
in the `colData`, if provided).  An empty `sampleMap` may produce empty
experiments if the levels of the "assay" factor in the `sampleMap` do not
match the names in the `ExperimentList`.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='MultiAssayExperiment-class.html'>MultiAssayExperiment-class</a></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='co'>## Run the example ExperimentList</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/example.html'>example</a></span><span class='op'>(</span><span class='st'>"ExperimentList"</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ## Create an empty ExperimentList instance</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ExperimentList()</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExperimentList class object of length 0:</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>  </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ## Create array matrix and AnnotatedDataFrame to create an ExpressionSet class</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; arraydat &lt;- matrix(data = seq(101, length.out = 20), ncol = 4,</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     dimnames = list(</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+         c("ENST00000294241", "ENST00000355076",</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+         "ENST00000383706","ENST00000234812", "ENST00000383323"),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+         c("array1", "array2", "array3", "array4")</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     ))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; colDat &lt;- data.frame(slope53 = rnorm(4),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     row.names = c("array1", "array2", "array3", "array4"))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ## SummarizedExperiment constructor</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; exprdat &lt;- SummarizedExperiment::SummarizedExperiment(arraydat,</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     colData = colDat)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ## Create a sample methylation dataset</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; methyldat &lt;- matrix(data = seq(1, length.out = 25), ncol = 5,</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     dimnames = list(</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+         c("ENST00000355076", "ENST00000383706",</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+           "ENST00000383323", "ENST00000234812", "ENST00000294241"),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+         c("methyl1", "methyl2", "methyl3",</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+           "methyl4", "methyl5")</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     ))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ## Create a sample RNASeqGene dataset</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; rnadat &lt;- matrix(</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     data = sample(c(46851, 5, 19, 13, 2197, 507,</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+         84318, 126, 17, 21, 23979, 614), size = 20, replace = TRUE),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     ncol = 4,</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     dimnames = list(</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+         c("XIST", "RPS4Y1", "KDM5D", "ENST00000383323", "ENST00000234812"),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+         c("samparray1", "samparray2", "samparray3", "samparray4")</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     ))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ## Create a mock RangedSummarizedExperiment from a data.frame</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; rangedat &lt;- data.frame(chr="chr2", start = 11:15, end = 12:16,</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     strand = c("+", "-", "+", "*", "."),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     samp0 = c(0,0,1,1,1), samp1 = c(1,0,1,0,1), samp2 = c(0,1,0,1,0),</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+     row.names = c(paste0("ENST", "00000", 135411:135414), "ENST00000383323"))</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; rangeSE &lt;- SummarizedExperiment::makeSummarizedExperimentFromDataFrame(rangedat)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ## Combine to a named list and call the ExperimentList constructor function</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; assayList &lt;- list(Affy = exprdat, Methyl450k = methyldat, RNASeqGene = rnadat,</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL+                 GISTIC = rangeSE)</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ## Use the ExperimentList constructor</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> ExprmL&gt; ExpList &lt;- ExperimentList(assayList)</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Create sample maps for each experiment</span></span>
<span class='r-in'><span class='va'>exprmap</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span class='r-in'>    primary <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Jack"</span>, <span class='st'>"Jill"</span>, <span class='st'>"Barbara"</span>, <span class='st'>"Bob"</span><span class='op'>)</span>,</span>
<span class='r-in'>    colname <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"array1"</span>, <span class='st'>"array2"</span>, <span class='st'>"array3"</span>, <span class='st'>"array4"</span><span class='op'>)</span>,</span>
<span class='r-in'>    stringsAsFactors <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>methylmap</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span class='r-in'>    primary <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Jack"</span>, <span class='st'>"Jack"</span>, <span class='st'>"Jill"</span>, <span class='st'>"Barbara"</span>, <span class='st'>"Bob"</span><span class='op'>)</span>,</span>
<span class='r-in'>    colname <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"methyl1"</span>, <span class='st'>"methyl2"</span>, <span class='st'>"methyl3"</span>, <span class='st'>"methyl4"</span>, <span class='st'>"methyl5"</span><span class='op'>)</span>,</span>
<span class='r-in'>    stringsAsFactors <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>rnamap</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span class='r-in'>    primary <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Jack"</span>, <span class='st'>"Jill"</span>, <span class='st'>"Bob"</span>, <span class='st'>"Barbara"</span><span class='op'>)</span>,</span>
<span class='r-in'>    colname <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"samparray1"</span>, <span class='st'>"samparray2"</span>, <span class='st'>"samparray3"</span>, <span class='st'>"samparray4"</span><span class='op'>)</span>,</span>
<span class='r-in'>    stringsAsFactors <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>gistmap</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span></span>
<span class='r-in'>    primary <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Jack"</span>, <span class='st'>"Bob"</span>, <span class='st'>"Jill"</span><span class='op'>)</span>,</span>
<span class='r-in'>    colname <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"samp0"</span>, <span class='st'>"samp1"</span>, <span class='st'>"samp2"</span><span class='op'>)</span>,</span>
<span class='r-in'>    stringsAsFactors <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Combine as a named list and convert to a DataFrame</span></span>
<span class='r-in'><span class='va'>maplist</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Affy <span class='op'>=</span> <span class='va'>exprmap</span>, Methyl450k <span class='op'>=</span> <span class='va'>methylmap</span>,</span>
<span class='r-in'>    RNASeqGene <span class='op'>=</span> <span class='va'>rnamap</span>, GISTIC <span class='op'>=</span> <span class='va'>gistmap</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Create a sampleMap</span></span>
<span class='r-in'><span class='va'>sampMap</span> <span class='op'>&lt;-</span> <span class='fu'><a href='listToMap.html'>listToMap</a></span><span class='op'>(</span><span class='va'>maplist</span><span class='op'>)</span></span>
<span class='r-in'><span class='co'>## Create an example phenotype data</span></span>
<span class='r-in'><span class='va'>colDat</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>sex <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"M"</span>, <span class='st'>"F"</span>, <span class='st'>"M"</span>, <span class='st'>"F"</span><span class='op'>)</span>, age <span class='op'>=</span> <span class='fl'>38</span><span class='op'>:</span><span class='fl'>41</span>,</span>
<span class='r-in'>    row.names <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Jack"</span>, <span class='st'>"Jill"</span>, <span class='st'>"Bob"</span>, <span class='st'>"Barbara"</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Create a MultiAssayExperiment instance</span></span>
<span class='r-in'><span class='va'>mae</span> <span class='op'>&lt;-</span> <span class='fu'>MultiAssayExperiment</span><span class='op'>(</span>experiments <span class='op'>=</span> <span class='va'>ExpList</span>, colData <span class='op'>=</span> <span class='va'>colDat</span>,</span>
<span class='r-in'>    sampleMap <span class='op'>=</span> <span class='va'>sampMap</span><span class='op'>)</span></span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


