<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>MultiAssayExperiment - An integrative multi-assay class for experiment data — MultiAssayExperiment-class • MultiAssayExperiment</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="MultiAssayExperiment - An integrative multi-assay class for experiment data — MultiAssayExperiment-class"><meta property="og:description" content="The MultiAssayExperiment class can be used to manage results of
diverse assays on a collection of specimen. Currently,  the class can handle
assays that are organized instances of
SummarizedExperiment,
ExpressionSet, matrix,
RaggedExperiment
(inherits from GRangesList), and RangedVcfStack.
Create new MultiAssayExperiment instances with the homonymous
constructor, minimally with the argument ExperimentList,
potentially also with the arguments colData (see section below) and
sampleMap."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiAssayExperiment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.22.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/MultiAssayExperiment.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a>
    </li>
    <li>
      <a href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../news/index.html">Release Notes</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/waldronlab/MultiAssayExperiment" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>MultiAssayExperiment - An integrative multi-assay class for experiment data</h1>
    <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/HEAD/R/MultiAssayExperiment-class.R" class="external-link"><code>R/MultiAssayExperiment-class.R</code></a>, <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/HEAD/R/MultiAssayExperiment-methods.R" class="external-link"><code>R/MultiAssayExperiment-methods.R</code></a>, <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/HEAD/R/assay-methods.R" class="external-link"><code>R/assay-methods.R</code></a></small>
    <div class="hidden name"><code>MultiAssayExperiment-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The <code>MultiAssayExperiment</code> class can be used to manage results of
diverse assays on a collection of specimen. Currently,  the class can handle
assays that are organized instances of
<code>SummarizedExperiment</code>,
<code>ExpressionSet</code>, <code>matrix</code>,
<code><a href="https://rdrr.io/pkg/RaggedExperiment/man/RaggedExperiment-class.html" class="external-link">RaggedExperiment</a></code>
(inherits from <code>GRangesList</code>), and <code>RangedVcfStack</code>.
Create new <code>MultiAssayExperiment</code> instances with the homonymous
constructor, minimally with the argument <code><a href="ExperimentList.html">ExperimentList</a></code>,
potentially also with the arguments <code>colData</code> (see section below) and
<code><a href="MultiAssayExperiment-methods.html">sampleMap</a></code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="co"># S4 method for MultiAssayExperiment</span>
<span class="fu">show</span><span class="op">(</span><span class="va">object</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment</span>
<span class="fu">updateObject</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment</span>
<span class="fu"><a href="https://rdrr.io/r/base/dimnames.html" class="external-link">dimnames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment</span>
<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span>, sampleMap <span class="op">=</span> <span class="cn">NULL</span>, mapFrom <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment</span>
<span class="fu">exportClass</span><span class="op">(</span>
  <span class="va">object</span>,
  dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">fmt</span>,
  <span class="va">ext</span>,
  match <span class="op">=</span> <span class="cn">FALSE</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>,
  <span class="va">...</span>
<span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment</span>
<span class="fu">assays</span><span class="op">(</span><span class="va">x</span>, withDimnames <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment,missing</span>
<span class="fu">assay</span><span class="op">(</span><span class="va">x</span>, <span class="va">i</span>, withDimnames <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment,numeric</span>
<span class="fu">assay</span><span class="op">(</span><span class="va">x</span>, <span class="va">i</span>, withDimnames <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">...</span><span class="op">)</span>

<span class="co"># S4 method for MultiAssayExperiment,character</span>
<span class="fu">assay</span><span class="op">(</span><span class="va">x</span>, <span class="va">i</span>, withDimnames <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object, x</dt>
<dd><p>A <code>MultiAssayExperiment</code> object</p></dd>
<dt>...</dt>
<dd><p>Additional arguments for supporting functions. See details.</p></dd>
<dt>verbose</dt>
<dd><p>logical(1) Whether to print additional information (default
TRUE)</p></dd>
<dt>sampleMap</dt>
<dd><p><code>c</code> method: a <code>sampleMap</code> <code>list</code> or
<code>DataFrame</code> to guide merge</p></dd>
<dt>mapFrom</dt>
<dd><p>Either a <code>logical</code>, <code>character</code>, or <code>integer</code>
vector indicating the experiment(s) that have an identical colname order as
the experiment input(s). If using a character input, the name must match
exactly.</p></dd>
<dt>dir</dt>
<dd><p>character(1) A directory for saving exported data (default:
`tempdir()`)</p></dd>
<dt>fmt</dt>
<dd><p>character(1) or function() Either a format character atomic as
supported by `write.table` either ('csv', or 'tsv') or a function whose
first two arguments are 'object to save' and 'file location'</p></dd>
<dt>ext</dt>
<dd><p>character(1) A file extension supported by the format argument</p></dd>
<dt>match</dt>
<dd><p>logical(1) Whether to coerce the current object to a
'MatchedAssayExperiment' object (default: FALSE)</p></dd>
<dt>withDimnames</dt>
<dd><p>logical (default TRUE) whether to return dimension names
included in the object</p></dd>
<dt>i</dt>
<dd><p>An integer or character scalar indicating the assay to return</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <code>MultiAssayExperiment</code> object</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The dots (<code>...</code>) argument allows the user to specify additional
arguments in several instances.</p><ul><li><p>subsetting <strong>[</strong>: additional arguments sent to
<a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a>.</p></li>
<li><p>mergeReplicates: used to specify arguments for the <code>simplify</code>
functional argument</p></li>
<li><p>assay: may contain withDimnames, which is forwarded to assays</p></li>
<li><p>combining <strong>c</strong>: compatible <code>MultiAssayExperiment</code> classes
passed on to the <code><a href="ExperimentList-class.html">ExperimentList</a></code> constructor,
can be a <code>list</code>, <code>List</code>, or a series of
named arguments. See the examples below.</p></li>
</ul></div>
    <div id="methods-by-generic-">
    <h2>Methods (by generic)</h2>
    
<ul><li><p><code>show</code>: Show method for a
<code>MultiAssayExperiment</code></p></li>
<li><p><code>length</code>: Get the length of ExperimentList</p></li>
<li><p><code>names</code>: Get the names of the ExperimentList</p></li>
<li><p><code>updateObject</code>: Update old serialized MultiAssayExperiment
objects to new API</p></li>
<li><p><code>dimnames</code>: Get the dimension names
for a <code>MultiAssayExperiment</code> object</p></li>
<li><p><code>c</code>: Add a supported data class to the
<code>ExperimentList</code></p></li>
<li><p><code>exportClass</code>: Export data from class to a series
of text files</p></li>
<li><p><code>assays</code>: Obtain a <code>SimpleList</code> of assay
data for all available experiments in the object</p></li>
<li><p><code>assay</code>: Convenience function for extracting the
assay of the first element (default) in the <code>ExperimentList</code>. A
<code>numeric</code> or <code>character</code> index can also be provided</p></li>
</ul></div>
    <div id="slots">
    <h2>Slots</h2>
    

<dl><dt><code>ExperimentList</code></dt>
<dd><p>A <code><a href="ExperimentList.html">ExperimentList</a></code> class object for
each assay dataset</p></dd>


<dt><code>colData</code></dt>
<dd><p>A <code>DataFrame</code> of all clinical/specimen data available
across experiments</p></dd>


<dt><code>sampleMap</code></dt>
<dd><p>A <code>DataFrame</code> of translatable identifiers
of samples and participants</p></dd>


<dt><code>metadata</code></dt>
<dd><p>Additional data describing the
<code>MultiAssayExperiment</code> object</p></dd>


<dt><code>drops</code></dt>
<dd><p>A metadata <code>list</code> of dropped information</p></dd>


</dl></div>
    <div id="coldata">
    <h2>colData</h2>
    

<p>The <code>colData</code> slot is a collection of primary specimen data valid
across all experiments. This slot is strictly of class
<code>DataFrame</code> but arguments for the constructor function
allow arguments to be of class <code>data.frame</code> and subsequently coerced.</p>
    </div>
    <div id="experimentlist">
    <h2>ExperimentList</h2>
    

<p>The <code><a href="ExperimentList.html">ExperimentList</a></code> slot is designed to contain results from
each experiment/assay. It contains a SimpleList.</p>
    </div>
    <div id="samplemap">
    <h2>sampleMap</h2>
    

<p>The <code><a href="MultiAssayExperiment-methods.html">sampleMap</a></code> contains a <code>DataFrame</code> of translatable
identifiers of samples and participants or biological units. The standard
column names of the <code>sampleMap</code> are "assay", "primary", and "colname".
Note that the "assay" column is a factor corresponding to the names of each
experiment in the <code>ExperimentList</code>. In the case where these names do
not match between the <code>sampleMap</code> and the experiments, the documented
experiments in the <code>sampleMap</code> take precedence and experiments are
dropped by the harmonization procedure. The constructor function will
generate a <code>sampleMap</code> in the case where it is not provided and this
method may be a 'safer' alternative for creating the <code>MultiAssayExperiment</code>
(so long as the rownames are identical in the <code>colData</code>, if provided).
An empty <code>sampleMap</code> may produce empty experiments if the levels of the
"assay" factor in the <code>sampleMap</code> do not match the names in the
<code>ExperimentList</code>.</p>
    </div>
    <div id="coercion">
    <h2>coercion</h2>
    

<p>Convert a <code>list</code> or S4 <code>List</code> to a MultiAssayExperiment object using the
<a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a> function.</p>
<p>In the following example, <code>x</code> is either a <code>list</code> or List:</p><div class="sourceCode"><pre><code><span class="va">`as(x, "MultiAssayExperiment")`</span></code></pre></div>

<p>Convert a <code>MultiAssayExperiment</code> to <code>MAF</code> class object using the
<a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a> function.</p>
<p>In the following example, <code>x</code> is a MultiAssayExperiment:</p><div class="sourceCode"><pre><code><span class="va">`MultiAssayExperimentToMAF(x)`</span></code></pre></div>

    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="MultiAssayExperiment-methods.html">MultiAssayExperiment-methods</a> for slot modifying methods,
<a href="https://github.com/waldronlab/MultiAssayExperiment/wiki/MultiAssayExperiment-API" class="external-link">MultiAssayExperiment API</a></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="st">"MultiAssayExperiment"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Run the example ExperimentList</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; example("ExperimentList")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create an empty ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExperimentList()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExperimentList class object of length 0:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create array matrix and AnnotatedDataFrame to create an ExpressionSet class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; arraydat &lt;- matrix(data = seq(101, length.out = 20), ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000294241", "ENST00000355076",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         "ENST00000383706","ENST00000234812", "ENST00000383323"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("array1", "array2", "array3", "array4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; colDat &lt;- data.frame(slope53 = rnorm(4),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c("array1", "array2", "array3", "array4"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## SummarizedExperiment constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; exprdat &lt;- SummarizedExperiment::SummarizedExperiment(arraydat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     colData = colDat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample methylation dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; methyldat &lt;- matrix(data = seq(1, length.out = 25), ncol = 5,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000355076", "ENST00000383706",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "ENST00000383323", "ENST00000234812", "ENST00000294241"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("methyl1", "methyl2", "methyl3",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "methyl4", "methyl5")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample RNASeqGene dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rnadat &lt;- matrix(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     data = sample(c(46851, 5, 19, 13, 2197, 507,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         84318, 126, 17, 21, 23979, 614), size = 20, replace = TRUE),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("XIST", "RPS4Y1", "KDM5D", "ENST00000383323", "ENST00000234812"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("samparray1", "samparray2", "samparray3", "samparray4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a mock RangedSummarizedExperiment from a data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangedat &lt;- data.frame(chr="chr2", start = 11:15, end = 12:16,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     strand = c("+", "-", "+", "*", "."),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     samp0 = c(0,0,1,1,1), samp1 = c(1,0,1,0,1), samp2 = c(0,1,0,1,0),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c(paste0("ENST", "00000", 135411:135414), "ENST00000383323"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangeSE &lt;- SummarizedExperiment::makeSummarizedExperimentFromDataFrame(rangedat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Combine to a named list and call the ExperimentList constructor function</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; assayList &lt;- list(Affy = exprdat, Methyl450k = methyldat, RNASeqGene = rnadat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+                 GISTIC = rangeSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Use the ExperimentList constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExpList &lt;- ExperimentList(assayList)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create sample maps for each experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; exprmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("array1", "array2", "array3", "array4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; methylmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("methyl1", "methyl2", "methyl3", "methyl4", "methyl5"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; rnamap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Bob", "Barbara"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samparray1", "samparray2", "samparray3", "samparray4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; gistmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Bob", "Jill"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samp0", "samp1", "samp2"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Combine as a named list and convert to a DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; maplist &lt;- list(Affy = exprmap, Methyl450k = methylmap,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     RNASeqGene = rnamap, GISTIC = gistmap)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a sampleMap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; sampMap &lt;- listToMap(maplist)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create an example phenotype data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; colDat &lt;- data.frame(sex = c("M", "F", "M", "F"), age = 38:41,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     row.names = c("Jack", "Jill", "Bob", "Barbara"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a MultiAssayExperiment instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; mae &lt;- MultiAssayExperiment(experiments = ExpList, colData = colDat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     sampleMap = sampMap)</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Subsetting</span></span>
<span class="r-in"><span class="co"># Rows (i) Rows/Features in each experiment</span></span>
<span class="r-in"><span class="va">mae</span><span class="op">[</span><span class="fl">1</span>, , <span class="op">]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 4 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 1 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 1 rows and 5 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 1 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 1 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"><span class="va">mae</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span>, , <span class="op">]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 4 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 3 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 3 rows and 5 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 3 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 3 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Columns (j) Rows in colData</span></span>
<span class="r-in"><span class="va">mae</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>,  <span class="op">]</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 4 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 5 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 5 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 5 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Assays (k)</span></span>
<span class="r-in"><span class="va">mae</span><span class="op">[</span>, , <span class="st">"Affy"</span><span class="op">]</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>'experiments' dropped; see 'metadata'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> harmonizing input:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   removing 12 sampleMap rows not in names(experiments)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 1 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiment with a user-defined name and respective class.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 1:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Complete cases (returns logical vector)</span></span>
<span class="r-in"><span class="va">completes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span>
<span class="r-in"><span class="va">compMAE</span> <span class="op">&lt;-</span> <span class="va">mae</span><span class="op">[</span>, <span class="va">completes</span>, <span class="op">]</span></span>
<span class="r-in"><span class="va">compMAE</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 4 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 5 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 5 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"><span class="fu">colData</span><span class="op">(</span><span class="va">compMAE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 3 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              sex       age</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      &lt;character&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Jack           M        38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Jill           F        39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bob            M        40</span>
<span class="r-in"></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="st">"MultiAssayExperiment"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Run the example ExperimentList</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; example("ExperimentList")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create an empty ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExperimentList()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExperimentList class object of length 0:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create array matrix and AnnotatedDataFrame to create an ExpressionSet class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; arraydat &lt;- matrix(data = seq(101, length.out = 20), ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000294241", "ENST00000355076",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         "ENST00000383706","ENST00000234812", "ENST00000383323"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("array1", "array2", "array3", "array4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; colDat &lt;- data.frame(slope53 = rnorm(4),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c("array1", "array2", "array3", "array4"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## SummarizedExperiment constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; exprdat &lt;- SummarizedExperiment::SummarizedExperiment(arraydat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     colData = colDat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample methylation dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; methyldat &lt;- matrix(data = seq(1, length.out = 25), ncol = 5,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000355076", "ENST00000383706",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "ENST00000383323", "ENST00000234812", "ENST00000294241"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("methyl1", "methyl2", "methyl3",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "methyl4", "methyl5")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample RNASeqGene dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rnadat &lt;- matrix(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     data = sample(c(46851, 5, 19, 13, 2197, 507,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         84318, 126, 17, 21, 23979, 614), size = 20, replace = TRUE),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("XIST", "RPS4Y1", "KDM5D", "ENST00000383323", "ENST00000234812"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("samparray1", "samparray2", "samparray3", "samparray4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a mock RangedSummarizedExperiment from a data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangedat &lt;- data.frame(chr="chr2", start = 11:15, end = 12:16,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     strand = c("+", "-", "+", "*", "."),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     samp0 = c(0,0,1,1,1), samp1 = c(1,0,1,0,1), samp2 = c(0,1,0,1,0),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c(paste0("ENST", "00000", 135411:135414), "ENST00000383323"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangeSE &lt;- SummarizedExperiment::makeSummarizedExperimentFromDataFrame(rangedat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Combine to a named list and call the ExperimentList constructor function</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; assayList &lt;- list(Affy = exprdat, Methyl450k = methyldat, RNASeqGene = rnadat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+                 GISTIC = rangeSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Use the ExperimentList constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExpList &lt;- ExperimentList(assayList)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create sample maps for each experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; exprmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("array1", "array2", "array3", "array4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; methylmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("methyl1", "methyl2", "methyl3", "methyl4", "methyl5"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; rnamap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Bob", "Barbara"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samparray1", "samparray2", "samparray3", "samparray4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; gistmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Bob", "Jill"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samp0", "samp1", "samp2"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Combine as a named list and convert to a DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; maplist &lt;- list(Affy = exprmap, Methyl450k = methylmap,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     RNASeqGene = rnamap, GISTIC = gistmap)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a sampleMap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; sampMap &lt;- listToMap(maplist)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create an example phenotype data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; colDat &lt;- data.frame(sex = c("M", "F", "M", "F"), age = 38:41,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     row.names = c("Jack", "Jill", "Bob", "Barbara"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a MultiAssayExperiment instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; mae &lt;- MultiAssayExperiment(experiments = ExpList, colData = colDat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     sampleMap = sampMap)</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Add an experiment</span></span>
<span class="r-in"><span class="va">test1</span> <span class="op">&lt;-</span> <span class="va">mae</span><span class="op">[[</span><span class="fl">1L</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Combine current MultiAssayExperiment with additional experiment</span></span>
<span class="r-in"><span class="co">## (no sampleMap)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">mae</span>, newExperiment <span class="op">=</span> <span class="va">test1</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 5 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 5:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 5 rows and 5 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] newExperiment: SummarizedExperiment with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">test2</span> <span class="op">&lt;-</span> <span class="va">mae</span><span class="op">[[</span><span class="fl">3L</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">mae</span>, newExp <span class="op">=</span> <span class="va">test2</span>, mapFrom <span class="op">=</span> <span class="fl">3L</span><span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Assuming column order in the data provided </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>  matches the order in 'mapFrom' experiment(s) colnames</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 5 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 5:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 5 rows and 5 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] newExp: matrix with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Add experiment using experiment name in mapFrom</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">mae</span>, RNASeqGeneV2 <span class="op">=</span> <span class="va">test2</span>, mapFrom <span class="op">=</span> <span class="st">"RNASeqGene"</span><span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Assuming column order in the data provided </span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>  matches the order in 'mapFrom' experiment(s) colnames</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 5 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 5:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 5 rows and 5 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] RNASeqGeneV2: matrix with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

