<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Construct an integrative representation of multi-omic data with MultiAssayExperiment — MultiAssayExperiment • MultiAssayExperiment</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Construct an integrative representation of multi-omic data with MultiAssayExperiment — MultiAssayExperiment"><meta name="description" content="The constructor function for the MultiAssayExperiment combines
multiple data elements from the different hierarchies of data
(study, experiments, and samples). It can create instances where neither
a sampleMap or a colData set is provided. Please see the
MultiAssayExperiment API documentation for more information."><meta property="og:description" content="The constructor function for the MultiAssayExperiment combines
multiple data elements from the different hierarchies of data
(study, experiments, and samples). It can create instances where neither
a sampleMap or a colData set is provided. Please see the
MultiAssayExperiment API documentation for more information."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="default" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MultiAssayExperiment</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.30.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Quick Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><a class="dropdown-item" href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a></li>
    <li><a class="dropdown-item" href="../articles/MultiAssayExperiment.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a></li>
    <li><a class="dropdown-item" href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a></li>
  </ul></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news"><li><a class="dropdown-item" href="../news/index.html">Release Notes</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="nav-link" href="https://github.com/waldronlab/MultiAssayExperiment"><span class="fa fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Construct an integrative representation of multi-omic data with <code>MultiAssayExperiment</code></h1>
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/RELEASE_3_19/R/MultiAssayExperiment-class.R"><code>R/MultiAssayExperiment-class.R</code></a></small>
      <div class="d-none name"><code>MultiAssayExperiment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The constructor function for the <a href="MultiAssayExperiment-class.html">MultiAssayExperiment</a> combines
multiple data elements from the different hierarchies of data
(study, experiments, and samples). It can create instances where neither
a <code>sampleMap</code> or a <code>colData</code> set is provided. Please see the
MultiAssayExperiment API documentation for more information.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MultiAssayExperiment</span><span class="op">(</span></span>
<span>  experiments <span class="op">=</span> <span class="fu"><a href="ExperimentList.html">ExperimentList</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  colData <span class="op">=</span> <span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  sampleMap <span class="op">=</span> <span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span>assay <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="op">)</span>, primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="op">)</span>, colname <span class="op">=</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  drops <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-experiments">experiments<a class="anchor" aria-label="anchor" href="#arg-experiments"></a></dt>
<dd><p>A <code>list</code> or <code><a href="ExperimentList.html">ExperimentList</a></code> of all
combined experiments</p></dd>


<dt id="arg-coldata">colData<a class="anchor" aria-label="anchor" href="#arg-coldata"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code> or <code>data.frame</code> of
characteristics for all biological units</p></dd>


<dt id="arg-samplemap">sampleMap<a class="anchor" aria-label="anchor" href="#arg-samplemap"></a></dt>
<dd><p>A <code>DataFrame</code> or <code>data.frame</code> of assay names,
sample identifiers, and colname samples</p></dd>


<dt id="arg-metadata">metadata<a class="anchor" aria-label="anchor" href="#arg-metadata"></a></dt>
<dd><p>An optional argument of "ANY" class (usually list) for
content describing the experiments</p></dd>


<dt id="arg-drops">drops<a class="anchor" aria-label="anchor" href="#arg-drops"></a></dt>
<dd><p>A <code>list</code> of unmatched information
(included after subsetting)</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code>MultiAssayExperiment</code> object that can store
experiment and phenotype data</p>
    </div>
    <div class="section level2">
    <h2 id="coldata">colData<a class="anchor" aria-label="anchor" href="#coldata"></a></h2>


<p>The <code>colData</code> input can be either <code>DataFrame</code> or <code>data.frame</code> with
subsequent coercion to DataFrame. The rownames in the <code>colData</code> must match
the colnames in the experiments if no sampleMap is provided.</p>
    </div>
    <div class="section level2">
    <h2 id="experiments">experiments<a class="anchor" aria-label="anchor" href="#experiments"></a></h2>


<p>The <code>experiments</code> input can be of class <code>SimpleList</code> or <code>list</code>.
This input becomes the <code><a href="ExperimentList.html">ExperimentList</a></code>. Each element of the
input <code>list</code> or <code>List</code> must be named, rectangular with two dimensions, and
have <code>dimnames</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="samplemap">sampleMap<a class="anchor" aria-label="anchor" href="#samplemap"></a></h2>


<p>The <code><a href="MultiAssayExperiment-methods.html">sampleMap</a></code> can either be input as <code>DataFrame</code> or
<code>data.frame</code> with eventual coercion to <code>DataFrame</code>. The <code>sampleMap</code> relates
biological units and biological measurements within each assay. Each row in
the <code>sampleMap</code> is a single such link. The standard column names of the
<code>sampleMap</code> are "assay", "primary", and "colname".  Note that the "assay"
column is a factor corresponding to the names of each experiment in the
<code>ExperimentList</code>. In the case where these names do not match between the
<code>sampleMap</code> and the experiments, the documented experiments in the
<code>sampleMap</code> take precedence and experiments are dropped by the harmonization
procedure. The constructor function will generate a <code>sampleMap</code> in the case
where it is not provided and this method may be a 'safer' alternative for
creating the <code>MultiAssayExperiment</code> (so long as the rownames are identical
in the <code>colData</code>, if provided).  An empty <code>sampleMap</code> may produce empty
experiments if the levels of the "assay" factor in the <code>sampleMap</code> do not
match the names in the <code>ExperimentList</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><a href="MultiAssayExperiment-class.html">MultiAssayExperiment</a></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Run the example ExperimentList</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="st">"ExperimentList"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create an empty ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExperimentList()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExperimentList class object of length 0:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create array matrix and AnnotatedDataFrame to create an ExpressionSet class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; arraydat &lt;- matrix(data = seq(101, length.out = 20), ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000294241", "ENST00000355076",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         "ENST00000383706","ENST00000234812", "ENST00000383323"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("array1", "array2", "array3", "array4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; colDat &lt;- data.frame(slope53 = rnorm(4),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c("array1", "array2", "array3", "array4"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## SummarizedExperiment constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; exprdat &lt;- SummarizedExperiment::SummarizedExperiment(arraydat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     colData = colDat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample methylation dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; methyldat &lt;- matrix(data = seq(1, length.out = 25), ncol = 5,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000355076", "ENST00000383706",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "ENST00000383323", "ENST00000234812", "ENST00000294241"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("methyl1", "methyl2", "methyl3",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "methyl4", "methyl5")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample RNASeqGene dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rnadat &lt;- matrix(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     data = sample(c(46851, 5, 19, 13, 2197, 507,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         84318, 126, 17, 21, 23979, 614), size = 20, replace = TRUE),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("XIST", "RPS4Y1", "KDM5D", "ENST00000383323", "ENST00000234812"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("samparray1", "samparray2", "samparray3", "samparray4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a mock RangedSummarizedExperiment from a data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangedat &lt;- data.frame(chr="chr2", start = 11:15, end = 12:16,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     strand = c("+", "-", "+", "*", "."),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     samp0 = c(0,0,1,1,1), samp1 = c(1,0,1,0,1), samp2 = c(0,1,0,1,0),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c(paste0("ENST", "00000", 135411:135414), "ENST00000383323"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangeSE &lt;- SummarizedExperiment::makeSummarizedExperimentFromDataFrame(rangedat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Combine to a named list and call the ExperimentList constructor function</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; assayList &lt;- list(Affy = exprdat, Methyl450k = methyldat, RNASeqGene = rnadat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+                 GISTIC = rangeSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Use the ExperimentList constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExpList &lt;- ExperimentList(assayList)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create sample maps for each experiment</span></span></span>
<span class="r-in"><span><span class="va">exprmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jill"</span>, <span class="st">"Barbara"</span>, <span class="st">"Bob"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"array1"</span>, <span class="st">"array2"</span>, <span class="st">"array3"</span>, <span class="st">"array4"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">methylmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jack"</span>, <span class="st">"Jill"</span>, <span class="st">"Barbara"</span>, <span class="st">"Bob"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"methyl1"</span>, <span class="st">"methyl2"</span>, <span class="st">"methyl3"</span>, <span class="st">"methyl4"</span>, <span class="st">"methyl5"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">rnamap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jill"</span>, <span class="st">"Bob"</span>, <span class="st">"Barbara"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"samparray1"</span>, <span class="st">"samparray2"</span>, <span class="st">"samparray3"</span>, <span class="st">"samparray4"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gistmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Bob"</span>, <span class="st">"Jill"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"samp0"</span>, <span class="st">"samp1"</span>, <span class="st">"samp2"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Combine as a named list and convert to a DataFrame</span></span></span>
<span class="r-in"><span><span class="va">maplist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Affy <span class="op">=</span> <span class="va">exprmap</span>, Methyl450k <span class="op">=</span> <span class="va">methylmap</span>,</span></span>
<span class="r-in"><span>    RNASeqGene <span class="op">=</span> <span class="va">rnamap</span>, GISTIC <span class="op">=</span> <span class="va">gistmap</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create a sampleMap</span></span></span>
<span class="r-in"><span><span class="va">sampMap</span> <span class="op">&lt;-</span> <span class="fu"><a href="mapToList.html">listToMap</a></span><span class="op">(</span><span class="va">maplist</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">## Create an example phenotype data</span></span></span>
<span class="r-in"><span><span class="va">colDat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>sex <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"M"</span>, <span class="st">"F"</span>, <span class="st">"M"</span>, <span class="st">"F"</span><span class="op">)</span>, age <span class="op">=</span> <span class="fl">38</span><span class="op">:</span><span class="fl">41</span>,</span></span>
<span class="r-in"><span>    row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jill"</span>, <span class="st">"Bob"</span>, <span class="st">"Barbara"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create a MultiAssayExperiment instance</span></span></span>
<span class="r-in"><span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu">MultiAssayExperiment</span><span class="op">(</span>experiments <span class="op">=</span> <span class="va">ExpList</span>, colData <span class="op">=</span> <span class="va">colDat</span>,</span></span>
<span class="r-in"><span>    sampleMap <span class="op">=</span> <span class="va">sampMap</span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Marcel Ramos, Martin Morgan, Vincent J Carey, Levi Waldron.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

