<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>A group of helper functions for manipulating and cleaning a MultiAssayExperiment — MultiAssayExperiment-helpers • MultiAssayExperiment</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="A group of helper functions for manipulating and cleaning a MultiAssayExperiment — MultiAssayExperiment-helpers"><meta name="description" content="A set of helper functions were created to help clean and
manipulate a MultiAssayExperiment object. intersectRows also works
for ExperimentList objects.
complete.cases: Returns a logical vector corresponding to 'colData'
rows that have data across all experiments
isEmpty: Returns a logical TRUE value for zero length
MultiAssayExperiment objects
intersectRows: Takes all common rows across experiments,
excludes experiments with empty rownames
intersectColumns: A wrapper for complete.cases to return a
MultiAssayExperiment with only those biological units that have
measurements across all experiments
replicated: Identifies, via logical vectors, colnames that
originate from a single biological unit within each assay
replicates: Provides the replicate colnames found with
the replicated function by their name, empty list if none
anyReplicated: Whether the assay has replicate measurements
showReplicated: Displays the actual columns that are replicated per
assay and biological unit, i.e., primary value (colData
rowname) in the sampleMap
mergeReplicates: A function that combines replicated / repeated
measurements across all experiments and is guided by the replicated
return value
longFormat: A MultiAssayExperiment method that
returns a small and skinny DataFrame. The colDataCols
arguments allows the user to append colData columns to the data.
wideFormat: A function to reshape the data in a
MultiAssayExperiment to a &quot;wide&quot; format DataFrame. Each row in
the DataFrame represents an observation (corresponding to an entry in
the colData). If replicates are present, their data will be appended at
the end of the corresponding row and will generate additional NA data.
It is recommended to remove or consolidate technical replicates with
mergeReplicates. Optional colDataCols can be added when the
original object is a MultiAssayExperiment.
hasRowRanges: A function that identifies ExperimentList elements
that have a rowRanges method
getWithColData: A convenience function for extracting an assay
and associated colData
renamePrimary: A convenience function to rename the primary
biological units as represented in the rownames(colData)
renameColname: A convenience function to rename the colnames
of a particular assay

"><meta property="og:description" content="A set of helper functions were created to help clean and
manipulate a MultiAssayExperiment object. intersectRows also works
for ExperimentList objects.
complete.cases: Returns a logical vector corresponding to 'colData'
rows that have data across all experiments
isEmpty: Returns a logical TRUE value for zero length
MultiAssayExperiment objects
intersectRows: Takes all common rows across experiments,
excludes experiments with empty rownames
intersectColumns: A wrapper for complete.cases to return a
MultiAssayExperiment with only those biological units that have
measurements across all experiments
replicated: Identifies, via logical vectors, colnames that
originate from a single biological unit within each assay
replicates: Provides the replicate colnames found with
the replicated function by their name, empty list if none
anyReplicated: Whether the assay has replicate measurements
showReplicated: Displays the actual columns that are replicated per
assay and biological unit, i.e., primary value (colData
rowname) in the sampleMap
mergeReplicates: A function that combines replicated / repeated
measurements across all experiments and is guided by the replicated
return value
longFormat: A MultiAssayExperiment method that
returns a small and skinny DataFrame. The colDataCols
arguments allows the user to append colData columns to the data.
wideFormat: A function to reshape the data in a
MultiAssayExperiment to a &quot;wide&quot; format DataFrame. Each row in
the DataFrame represents an observation (corresponding to an entry in
the colData). If replicates are present, their data will be appended at
the end of the corresponding row and will generate additional NA data.
It is recommended to remove or consolidate technical replicates with
mergeReplicates. Optional colDataCols can be added when the
original object is a MultiAssayExperiment.
hasRowRanges: A function that identifies ExperimentList elements
that have a rowRanges method
getWithColData: A convenience function for extracting an assay
and associated colData
renamePrimary: A convenience function to rename the primary
biological units as represented in the rownames(colData)
renameColname: A convenience function to rename the colnames
of a particular assay

"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="default" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MultiAssayExperiment</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.30.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Quick Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><a class="dropdown-item" href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a></li>
    <li><a class="dropdown-item" href="../articles/MultiAssayExperiment.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a></li>
    <li><a class="dropdown-item" href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a></li>
  </ul></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news"><li><a class="dropdown-item" href="../news/index.html">Release Notes</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="nav-link" href="https://github.com/waldronlab/MultiAssayExperiment"><span class="fa fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>A group of helper functions for manipulating and cleaning a MultiAssayExperiment</h1>
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/RELEASE_3_19/R/MultiAssayExperiment-helpers.R"><code>R/MultiAssayExperiment-helpers.R</code></a></small>
      <div class="d-none name"><code>MultiAssayExperiment-helpers.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>A set of helper functions were created to help clean and
manipulate a MultiAssayExperiment object. <code>intersectRows</code> also works
for <code>ExperimentList</code> objects.</p><ul><li><p>complete.cases: Returns a logical vector corresponding to 'colData'
rows that have data across all experiments</p></li>
<li><p>isEmpty: Returns a logical <code>TRUE</code> value for zero length
<code>MultiAssayExperiment</code> objects</p></li>
<li><p>intersectRows: Takes all common rows across experiments,
excludes experiments with empty rownames</p></li>
<li><p>intersectColumns: A wrapper for <code>complete.cases</code> to return a
<code>MultiAssayExperiment</code> with only those biological units that have
measurements across all experiments</p></li>
<li><p>replicated: Identifies, via logical vectors, <code>colname</code>s that
originate from a single biological unit within each assay</p></li>
<li><p>replicates: Provides the replicate <code>colname</code>s found with
the <code>replicated</code> function by their name, empty list if none</p></li>
<li><p>anyReplicated: Whether the assay has replicate measurements</p></li>
<li><p>showReplicated: Displays the actual columns that are replicated per
assay and biological unit, i.e., <code>primary</code> value (<code>colData</code>
rowname) in the <code>sampleMap</code></p></li>
<li><p>mergeReplicates: A function that combines replicated / repeated
measurements across all experiments and is guided by the replicated
return value</p></li>
<li><p>longFormat: A <code>MultiAssayExperiment</code> method that
returns a small and skinny <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code>. The <code>colDataCols</code>
arguments allows the user to append <code>colData</code> columns to the data.</p></li>
<li><p>wideFormat: A function to reshape the data in a
<code>MultiAssayExperiment</code> to a "wide" format <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code>. Each row in
the <code>DataFrame</code> represents an observation (corresponding to an entry in
the <code>colData</code>). If replicates are present, their data will be appended at
the end of the corresponding row and will generate additional <code>NA</code> data.
It is recommended to remove or consolidate technical replicates with
<code>mergeReplicates</code>. Optional <code>colDataCols</code> can be added when the
original object is a <code>MultiAssayExperiment</code>.</p></li>
<li><p>hasRowRanges: A function that identifies ExperimentList elements
that have a <code>rowRanges</code> method</p></li>
<li><p>getWithColData: A convenience function for extracting an assay
and associated colData</p></li>
<li><p>renamePrimary: A convenience function to rename the primary
biological units as represented in the <code>rownames(colData)</code></p></li>
<li><p>renameColname: A convenience function to rename the colnames
of a particular assay</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu">isEmpty</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">intersectRows</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">intersectColumns</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">replicated</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu">replicated</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">anyReplicated</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu">anyReplicated</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">showReplicated</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu">showReplicated</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">replicates</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu">replicates</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">mergeReplicates</span><span class="op">(</span><span class="va">x</span>, replicates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>, simplify <span class="op">=</span> <span class="fu">BiocGenerics</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu">mergeReplicates</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  replicates <span class="op">=</span> <span class="fu">replicated</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>,</span>
<span>  simplify <span class="op">=</span> <span class="fu">BiocGenerics</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'ANY'</span></span>
<span><span class="fu">mergeReplicates</span><span class="op">(</span><span class="va">x</span>, replicates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>, simplify <span class="op">=</span> <span class="fu">BiocGenerics</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">longFormat</span><span class="op">(</span><span class="va">object</span>, colDataCols <span class="op">=</span> <span class="cn">NULL</span>, i <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">wideFormat</span><span class="op">(</span></span>
<span>  <span class="va">object</span>,</span>
<span>  colDataCols <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  check.names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  collapse <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>  i <span class="op">=</span> <span class="fl">1L</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">hasRowRanges</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu">hasRowRanges</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'ExperimentList'</span></span>
<span><span class="fu">hasRowRanges</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">getWithColData</span><span class="op">(</span><span class="va">x</span>, <span class="va">i</span>, mode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"append"</span>, <span class="st">"replace"</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">renamePrimary</span><span class="op">(</span><span class="va">x</span>, <span class="va">value</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">renameColname</span><span class="op">(</span><span class="va">x</span>, <span class="va">i</span>, <span class="va">value</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">splitAssays</span><span class="op">(</span><span class="va">x</span>, <span class="va">hitList</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'MultiAssayExperiment'</span></span>
<span><span class="fu">splitAssays</span><span class="op">(</span><span class="va">x</span>, <span class="va">hitList</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">makeHitList</span><span class="op">(</span><span class="va">x</span>, <span class="va">patternList</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments. See details for more information.</p></dd>


<dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p>A MultiAssayExperiment or ExperimentList</p></dd>


<dt id="arg-replicates">replicates<a class="anchor" aria-label="anchor" href="#arg-replicates"></a></dt>
<dd><p>A list of <code>LogicalList</code>s
indicating multiple / duplicate entries for each biological unit per assay,
see <code>replicated</code> (default <code>replicated(x)</code>).</p></dd>


<dt id="arg-simplify">simplify<a class="anchor" aria-label="anchor" href="#arg-simplify"></a></dt>
<dd><p>A function for merging repeat measurements in experiments
as indicated by the <code>replicated</code> method for <code>MultiAssayExperiment</code></p></dd>


<dt id="arg-object">object<a class="anchor" aria-label="anchor" href="#arg-object"></a></dt>
<dd><p>Any supported class object</p></dd>


<dt id="arg-coldatacols">colDataCols<a class="anchor" aria-label="anchor" href="#arg-coldatacols"></a></dt>
<dd><p>A <code>character</code>, <code>logical</code>, or <code>numeric</code>
index for <code>colData</code> columns to be included</p></dd>


<dt id="arg-i">i<a class="anchor" aria-label="anchor" href="#arg-i"></a></dt>
<dd><p>longFormat: The i-th assay in
<code>SummarizedExperiment</code>-like objects. A vector input is
supported in the case that the <code>SummarizedExperiment</code> object(s) has more
than one assay (default 1L),
renameColname: Either a <code>numeric</code> or <code>character</code> index
indicating the assay whose colnames are to be renamed</p></dd>


<dt id="arg-check-names">check.names<a class="anchor" aria-label="anchor" href="#arg-check-names"></a></dt>
<dd><p>(logical default TRUE) Column names of the output
<code>DataFrame</code> will be checked for syntactic validity and made unique,
if necessary</p></dd>


<dt id="arg-collapse">collapse<a class="anchor" aria-label="anchor" href="#arg-collapse"></a></dt>
<dd><p>(character default "_") A single string delimiter for output
column names. In <code>wideFormat</code>, experiments and rownames (and when
replicate samples are present, colnames) are seperated by this delimiter</p></dd>


<dt id="arg-mode">mode<a class="anchor" aria-label="anchor" href="#arg-mode"></a></dt>
<dd><p>String indicating how <code>MultiAssayExperiment</code>
column-level metadata should be added to the
<code>SummarizedExperiment</code> <code>colData</code>.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p><code>logical(1)</code> Whether to <code>suppressMessages</code> on subsetting
operations in <code>getWithColData</code> (default FALSE)</p></dd>


<dt id="arg-value">value<a class="anchor" aria-label="anchor" href="#arg-value"></a></dt>
<dd><p>renamePrimary: A <code>character</code> vector of the same length as
the existing <code>rownames(colData)</code> to use for replacement,
renameColname: A <code>CharacterList</code> or <code>list</code> with matching
<code>lengths</code> to replace <code>colnames(x)</code></p></dd>


<dt id="arg-hitlist">hitList<a class="anchor" aria-label="anchor" href="#arg-hitlist"></a></dt>
<dd><p>a named <code>list</code> or <code>List</code> of logical vectors that
indicate groupings in the assays</p></dd>


<dt id="arg-patternlist">patternList<a class="anchor" aria-label="anchor" href="#arg-patternlist"></a></dt>
<dd><p>a named <code>list</code> or <code>List</code> of atomic character
vectors that are the input to <code>grepl</code> for identifying groupings in
the assays</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>See the itemized list in the description section for details.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The <code>replicated</code> function finds replicate measurements in each
assay and returns a list of <code>LogicalList</code>s.
Each element in a single <code>LogicalList</code> corresponds to a
biological or <em>primary</em> unit as in the <code>sampleMap</code>. Below is a
small graphic for one particular biological unit in one assay, where the
logical vector corresponds to the number of measurements/samples in the
assay:</p><div class="sourceCode"><pre><code>
 &gt;      replicated(MultiAssayExperiment)
 (list str)       '-- $ AssayName
 (LogicalList str)      '-- [[ "Biological Unit" ]]
 Replicated if sum(...) &gt; 1          '-- TRUE TRUE FALSE FALSE
</code></pre></div><p><code>anyReplicated</code> determines if any of the assays have at least one
replicate. <em>Note</em>. These methods are not available for the
<code>ExperimentList</code> class due to a missing <code>sampleMap</code> structure
(by design).
<code>showReplicated</code> returns a list of <code>CharacterList</code>s where
each element corresponds to the the biological or <em>primary</em> units that
are replicated in that assay element. The values in the inner list are
the <code>colnames</code> in the assay that are technical replicates.</p>
<p>The <code>replicates</code> function (noun) returns the <code>colname</code>s
from the <code>sampleMap</code> that were identified as replicates. It returns a
list of <code>CharacterList</code>s for each assay present in the
<code>MultiAssayExperiment</code> and an inner entry for each biological unit
that has replicate observations in that assay.</p>
<p>The <code>mergeReplicates</code> function is a house-keeping method
for a <code>MultiAssayExperiment</code> where only <code>complete.cases</code> are
returned. This by-assay operation averages replicate measurements
(by default) and columns are aligned by the row order in <code>colData</code>.
Users can provide their own function for merging replicates with the
<code>simplify</code> functional argument. Additional inputs <code>...</code> are
sent to the 'simplify' function.</p>
<p>The <code>mergeReplicates</code> "ANY" method consolidates duplicate
measurements for rectangular data structures, returns object of the same
class (endomorphic). The ellipsis or <code>...</code> argument allows the
user to provide additional arguments to the <code>simplify</code> functional
argument.</p>
<p>The <code>longFormat</code> "ANY" class method, works with classes such as
<code>ExpressionSet</code> and <code>SummarizedExperiment</code> as well as <code>matrix</code>
to provide a consistent long and skinny <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code>.</p>
<p>The <code>hasRowRanges</code> method identifies assays that support
a <code>rowRanges</code> method <em>and</em>
return a <code>GRanges</code> object.</p>
    </div>
    <div class="section level2">
    <h2 id="mergereplicates">mergeReplicates<a class="anchor" aria-label="anchor" href="#mergereplicates"></a></h2>


<p>The <code>mergeReplicates</code> function makes use of the output from
<code>replicated</code> which will point out the duplicate measurements by
biological unit in the <code>MultiAssayExperiment</code>. This function will
return a <code>MultiAssayExperiment</code> with merged replicates. Additional
arguments can be provided to the simplify argument via the ellipsis
(<code>...</code>). For example, when replicates "TCGA-B" and "TCGA-A" are found in
the assay, the name of the first appearing replicate is taken (i.e., "B").
Note that a typical use case of merging replicates occurs when there are
multiple measurements on the <strong>same</strong> sample (within the same assay)
and can therefore be averaged.</p>
    </div>
    <div class="section level2">
    <h2 id="longformat">longFormat<a class="anchor" aria-label="anchor" href="#longformat"></a></h2>


<p>The 'longFormat' method takes data from the <code><a href="ExperimentList.html">ExperimentList</a></code>
in a <code>MultiAssayExperiment</code> and returns a uniform
<code>DataFrame</code>. The resulting DataFrame has columns indicating
primary, rowname, colname and value. This method can optionally include
columns of the MultiAssayExperiment colData named by <code>colDataCols</code> character
vector argument. (<code>MultiAssayExperiment</code> method only). The <code>i</code> argument
allows the user to specify the assay value for the
<code>SummarizedExperiment</code> assay function's <code>i</code> argument.</p>
    </div>
    <div class="section level2">
    <h2 id="wideformat">wideFormat<a class="anchor" aria-label="anchor" href="#wideformat"></a></h2>


<p>The <code>wideFormat</code> function returns standardized wide <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code>
where each row represents a biological unit as in the <code>colData</code>.
Depending on the data and setup, biological units can be patients, tumors,
specimens, etc. Metadata columns are
generated based on the names produced in the wide format
<code>DataFrame</code>. These can be accessed via the
<code><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">mcols()</a></code> function.
See the <code>wideFormat</code> section for description of the <code>colDataCols</code> and
<code>i</code> arguments.</p>
    </div>
    <div class="section level2">
    <h2 id="hasrowranges">hasRowRanges<a class="anchor" aria-label="anchor" href="#hasrowranges"></a></h2>


<p>The <code>hasRowRanges</code> method identifies assays with associated ranged
row data by directly testing the method on the object. The result from the
test must be a <code>GRanges</code> class object to satisfy the test.</p>
    </div>
    <div class="section level2">
    <h2 id="getwithcoldata">getWithColData<a class="anchor" aria-label="anchor" href="#getwithcoldata"></a></h2>


<p>The <code>getWithColData</code> function allows the user to conveniently extract
a particular assay as indicated by the <strong><code>i</code></strong> index argument. It
will also attempt to provide the
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></code>
along with the extracted object using the <code>colData&lt;-</code> replacement
method when possible. Typically, this method is available for
<code>SummarizedExperiment</code> and <code>RaggedExperiment</code> classes.</p>
<p>The setting of <code>mode</code> determines how the <code>colData</code>
is added. If <code>mode="append"</code>, the <code>MultiAssayExperiment</code>
metadata is appended onto that of the <code>SummarizedExperiment</code>.
If any fields are duplicated by name, the values in the
<code>SummarizedExperiment</code> are retained, with a warning emitted if
the values are different.  For <code>mode="replace"</code>, the
<code>MultiAssayExperiment</code> metadata replaces that of the
<code>SummarizedExperiment</code>, while for <code>mode="none"</code>,
no replacement or appending is performed.</p>
    </div>
    <div class="section level2">
    <h2 id="rename-">rename*<a class="anchor" aria-label="anchor" href="#rename-"></a></h2>


<p>The <code>renamePrimary</code> function allows the user to conveniently change the
actual names of the primary biological units as seen in
<code>rownames(colData)</code>. <code>renameColname</code> allows the user to change the
names of a particular assay based on index <code>i</code>. <code>i</code> can either be
a single numeric or character value. See <code>colnames&lt;-</code> method for
renaming multiple colnames in a <code>MultiAssayExperiment</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="splitassays">splitAssays<a class="anchor" aria-label="anchor" href="#splitassays"></a></h2>


<p>The <code>splitAssays</code> method separates columns in each of the assays based
on the <code>hitList</code> input. The <code>hitList</code> can be generated using
the <code>makeHitList</code> helper function. To use the <code>makeHitList</code>
helper, the user should input a list of patterns that will match on the
column names of each assay. These matches should be mutually exclusive as
to avoid repetition of columns across assays. See the examples section.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="va">MultiAssayExperiment</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Run the example ExperimentList</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; example("ExperimentList")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create an empty ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExperimentList()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExperimentList class object of length 0:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create array matrix and AnnotatedDataFrame to create an ExpressionSet class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; arraydat &lt;- matrix(data = seq(101, length.out = 20), ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000294241", "ENST00000355076",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         "ENST00000383706","ENST00000234812", "ENST00000383323"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("array1", "array2", "array3", "array4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; colDat &lt;- data.frame(slope53 = rnorm(4),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c("array1", "array2", "array3", "array4"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## SummarizedExperiment constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; exprdat &lt;- SummarizedExperiment::SummarizedExperiment(arraydat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     colData = colDat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample methylation dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; methyldat &lt;- matrix(data = seq(1, length.out = 25), ncol = 5,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000355076", "ENST00000383706",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "ENST00000383323", "ENST00000234812", "ENST00000294241"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("methyl1", "methyl2", "methyl3",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "methyl4", "methyl5")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample RNASeqGene dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rnadat &lt;- matrix(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     data = sample(c(46851, 5, 19, 13, 2197, 507,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         84318, 126, 17, 21, 23979, 614), size = 20, replace = TRUE),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("XIST", "RPS4Y1", "KDM5D", "ENST00000383323", "ENST00000234812"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("samparray1", "samparray2", "samparray3", "samparray4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a mock RangedSummarizedExperiment from a data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangedat &lt;- data.frame(chr="chr2", start = 11:15, end = 12:16,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     strand = c("+", "-", "+", "*", "."),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     samp0 = c(0,0,1,1,1), samp1 = c(1,0,1,0,1), samp2 = c(0,1,0,1,0),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c(paste0("ENST", "00000", 135411:135414), "ENST00000383323"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangeSE &lt;- SummarizedExperiment::makeSummarizedExperimentFromDataFrame(rangedat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Combine to a named list and call the ExperimentList constructor function</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; assayList &lt;- list(Affy = exprdat, Methyl450k = methyldat, RNASeqGene = rnadat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+                 GISTIC = rangeSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Use the ExperimentList constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExpList &lt;- ExperimentList(assayList)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create sample maps for each experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; exprmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("array1", "array2", "array3", "array4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; methylmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("methyl1", "methyl2", "methyl3", "methyl4", "methyl5"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; rnamap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Bob", "Barbara"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samparray1", "samparray2", "samparray3", "samparray4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; gistmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Bob", "Jill"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samp0", "samp1", "samp2"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Combine as a named list and convert to a DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; maplist &lt;- list(Affy = exprmap, Methyl450k = methylmap,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     RNASeqGene = rnamap, GISTIC = gistmap)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a sampleMap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; sampMap &lt;- listToMap(maplist)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create an example phenotype data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; colDat &lt;- data.frame(sex = c("M", "F", "M", "F"), age = 38:41,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     row.names = c("Jack", "Jill", "Bob", "Barbara"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a MultiAssayExperiment instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; mae &lt;- MultiAssayExperiment(experiments = ExpList, colData = colDat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     sampleMap = sampMap)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]  TRUE  TRUE  TRUE FALSE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">isEmpty</span><span class="op">(</span><span class="fu"><a href="MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## renaming biological units (primary)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">mae2</span> <span class="op">&lt;-</span> <span class="fu">renamePrimary</span><span class="op">(</span><span class="va">mae</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pt"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">colData</span><span class="op">(</span><span class="va">mae2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 4 rows and 2 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             sex       age</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     &lt;character&gt; &lt;integer&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pt1           M        38</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pt2           F        39</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pt3           M        40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pt4           F        41</span>
<span class="r-in"><span><span class="fu"><a href="MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">mae2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 16 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          assay     primary     colname</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   Affy               pt1      array1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   Affy               pt2      array2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   Affy               pt4      array3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   Affy               pt3      array4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   Methyl450k         pt1     methyl1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...        ...         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12  RNASeqGene         pt3  samparray3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13  RNASeqGene         pt4  samparray4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14  GISTIC             pt1       samp0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15  GISTIC             pt3       samp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16  GISTIC             pt2       samp2</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## renaming observational units (colname)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">mae2</span> <span class="op">&lt;-</span> <span class="fu">renameColname</span><span class="op">(</span><span class="va">mae</span>, i <span class="op">=</span> <span class="st">"Affy"</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"ARRAY"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mae2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CharacterList of length 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [["Affy"]] ARRAY1 ARRAY2 ARRAY3 ARRAY4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [["Methyl450k"]] methyl1 methyl2 methyl3 methyl4 methyl5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [["RNASeqGene"]] samparray1 samparray2 samparray3 samparray4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [["GISTIC"]] samp0 samp1 samp2</span>
<span class="r-in"><span><span class="fu"><a href="MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">mae2</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 16 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          assay     primary     colname</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   Affy              Jack      ARRAY1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   Affy              Jill      ARRAY2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   Affy           Barbara      ARRAY3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   Affy               Bob      ARRAY4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   Methyl450k        Jack     methyl1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...        ...         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12  RNASeqGene         Bob  samparray3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13  RNASeqGene     Barbara  samparray4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14  GISTIC            Jack       samp0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15  GISTIC             Bob       samp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16  GISTIC            Jill       samp2</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">patts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    normals <span class="op">=</span> <span class="st">"TCGA-[A-Z0-9]{2}-[A-Z0-9]{4}-11"</span>,</span></span>
<span class="r-in"><span>    tumors <span class="op">=</span> <span class="st">"TCGA-[A-Z0-9]{2}-[A-Z0-9]{4}-01"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"miniACC"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">makeHitList</span><span class="op">(</span><span class="va">miniACC</span>, <span class="va">patts</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## only turmors present</span></span></span>
<span class="r-in"><span><span class="fu">splitAssays</span><span class="op">(</span><span class="va">miniACC</span>, <span class="va">hits</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 5 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 5:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] tumors_RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] tumors_gistict: SummarizedExperiment with 198 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] tumors_RPPAArray: SummarizedExperiment with 33 rows and 46 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] tumors_Mutations: matrix with 97 rows and 90 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] tumors_miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Marcel Ramos, Martin Morgan, Vincent J Carey, Levi Waldron.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

