<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Accessing and modifying information in MultiAssayExperiment — MultiAssayExperiment-methods • MultiAssayExperiment</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Accessing and modifying information in MultiAssayExperiment — MultiAssayExperiment-methods"><meta property="og:description" content="A set of accessor and setter generic functions to extract
either the sampleMap, the ExperimentList,
colData, or metadata slots of a
MultiAssayExperiment object"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiAssayExperiment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.24.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/MultiAssayExperiment.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a>
    </li>
    <li>
      <a href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a>
    </li>
  </ul></li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../news/index.html">Release Notes</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/waldronlab/MultiAssayExperiment" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Accessing and modifying information in MultiAssayExperiment</h1>
    <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/HEAD/R/MultiAssayExperiment-class.R" class="external-link"><code>R/MultiAssayExperiment-class.R</code></a>, <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/HEAD/R/MultiAssayExperiment-methods.R" class="external-link"><code>R/MultiAssayExperiment-methods.R</code></a></small>
    <div class="hidden name"><code>MultiAssayExperiment-methods.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A set of accessor and setter generic functions to extract
either the <code>sampleMap</code>, the <code><a href="ExperimentList.html">ExperimentList</a></code>,
<code>colData</code>, or <code>metadata</code> slots of a
<code><a href="MultiAssayExperiment.html">MultiAssayExperiment</a></code> object</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre><code># S4 method for MultiAssayExperiment
sampleMap(x)

# S4 method for MultiAssayExperiment
experiments(x)

# S4 method for MultiAssayExperiment
colData(x, ...)

# S4 method for MultiAssayExperiment
metadata(x)

# S4 method for MultiAssayExperiment,DataFrame
sampleMap(object) &lt;- value

# S4 method for MultiAssayExperiment,ANY
sampleMap(object) &lt;- value

# S4 method for MultiAssayExperiment,ExperimentList
experiments(object) &lt;- value

# S4 method for MultiAssayExperiment,List
experiments(object) &lt;- value

# S4 method for MultiAssayExperiment,DataFrame
colData(x) &lt;- value

# S4 method for MultiAssayExperiment,ANY
colData(x) &lt;- value

# S4 method for MultiAssayExperiment
metadata(x, ...) &lt;- value

# S4 method for MultiAssayExperiment
$(x, name) &lt;- value

# S4 method for MultiAssayExperiment
names(x) &lt;- value

# S4 method for MultiAssayExperiment,List
colnames(x) &lt;- value

# S4 method for MultiAssayExperiment,list
colnames(x) &lt;- value

# S4 method for MultiAssayExperiment
$(x, name)</code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>...</dt>
<dd><p>Argument not in use</p></dd>


<dt>object, x</dt>
<dd><p>A <code>MultiAssayExperiment</code> object</p></dd>


<dt>value</dt>
<dd><p>See details.</p></dd>


<dt>name</dt>
<dd><p>A column in <code>colData</code></p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Accessors: Either a <code>sampleMap</code>, <code>ExperimentList</code>, or
<code>DataFrame</code> object</p>


<p>Setters: A <code>MultiAssayExperiment</code> object</p>
    </div>
    <div id="accessors">
    <h2>Accessors</h2>
    

<p>Eponymous names for accessing <code>MultiAssayExperiment</code> slots with the
exception of the <a href="ExperimentList.html">ExperimentList</a> accessor named <code>experiments</code>.</p><ul><li><p>colData: Access the <code>colData</code> slot</p></li>
<li><p>sampleMap: Access the <code>sampleMap</code> slot</p></li>
<li><p>experiments: Access the <a href="ExperimentList.html">ExperimentList</a> slot</p></li>
<li><p>`[[`: Access the <a href="ExperimentList.html">ExperimentList</a> slot</p></li>
<li><p>`$`: Access a column in <code>colData</code></p></li>
</ul></div>
    <div id="setters">
    <h2>Setters</h2>
    

<p>Setter method values (i.e., '<code>function(x) &lt;- value</code>'):</p><ul><li><p>experiments&lt;-: An <code><a href="ExperimentList.html">ExperimentList</a></code> object
    containing experiment data of supported classes</p></li>
<li><p>sampleMap&lt;-: A <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code> object relating
    samples to biological units and assays</p></li>
<li><p>colData&lt;-: A <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code> object describing the
    biological units</p></li>
<li><p>metadata&lt;-: A <code>list</code> object of metadata</p></li>
<li><p>`[[&lt;-`: Equivalent to the <code>experiments&lt;-</code> setter method for
    convenience</p></li>
<li><p>`$&lt;-`: A vector to replace the indicated column in <code>colData</code></p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Load example MultiAssayExperiment</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="va">MultiAssayExperiment</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Run the example ExperimentList</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; example("ExperimentList")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create an empty ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExperimentList()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExperimentList class object of length 0:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create array matrix and AnnotatedDataFrame to create an ExpressionSet class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; arraydat &lt;- matrix(data = seq(101, length.out = 20), ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000294241", "ENST00000355076",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         "ENST00000383706","ENST00000234812", "ENST00000383323"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("array1", "array2", "array3", "array4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; colDat &lt;- data.frame(slope53 = rnorm(4),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c("array1", "array2", "array3", "array4"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## SummarizedExperiment constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; exprdat &lt;- SummarizedExperiment::SummarizedExperiment(arraydat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     colData = colDat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample methylation dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; methyldat &lt;- matrix(data = seq(1, length.out = 25), ncol = 5,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000355076", "ENST00000383706",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "ENST00000383323", "ENST00000234812", "ENST00000294241"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("methyl1", "methyl2", "methyl3",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "methyl4", "methyl5")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample RNASeqGene dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rnadat &lt;- matrix(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     data = sample(c(46851, 5, 19, 13, 2197, 507,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         84318, 126, 17, 21, 23979, 614), size = 20, replace = TRUE),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("XIST", "RPS4Y1", "KDM5D", "ENST00000383323", "ENST00000234812"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("samparray1", "samparray2", "samparray3", "samparray4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a mock RangedSummarizedExperiment from a data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangedat &lt;- data.frame(chr="chr2", start = 11:15, end = 12:16,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     strand = c("+", "-", "+", "*", "."),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     samp0 = c(0,0,1,1,1), samp1 = c(1,0,1,0,1), samp2 = c(0,1,0,1,0),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c(paste0("ENST", "00000", 135411:135414), "ENST00000383323"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangeSE &lt;- SummarizedExperiment::makeSummarizedExperimentFromDataFrame(rangedat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Combine to a named list and call the ExperimentList constructor function</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; assayList &lt;- list(Affy = exprdat, Methyl450k = methyldat, RNASeqGene = rnadat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+                 GISTIC = rangeSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Use the ExperimentList constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExpList &lt;- ExperimentList(assayList)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create sample maps for each experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; exprmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("array1", "array2", "array3", "array4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; methylmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("methyl1", "methyl2", "methyl3", "methyl4", "methyl5"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; rnamap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Bob", "Barbara"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samparray1", "samparray2", "samparray3", "samparray4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; gistmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Bob", "Jill"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samp0", "samp1", "samp2"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Combine as a named list and convert to a DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; maplist &lt;- list(Affy = exprmap, Methyl450k = methylmap,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     RNASeqGene = rnamap, GISTIC = gistmap)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a sampleMap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; sampMap &lt;- listToMap(maplist)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create an example phenotype data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; colDat &lt;- data.frame(sex = c("M", "F", "M", "F"), age = 38:41,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     row.names = c("Jack", "Jill", "Bob", "Barbara"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a MultiAssayExperiment instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; mae &lt;- MultiAssayExperiment(experiments = ExpList, colData = colDat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     sampleMap = sampMap)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Access the sampleMap</span></span></span>
<span class="r-in"><span><span class="fu">sampleMap</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 16 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          assay     primary     colname</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   Affy              Jack      array1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   Affy              Jill      array2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   Affy           Barbara      array3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4   Affy               Bob      array4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5   Methyl450k        Jack     methyl1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...        ...         ...         ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12  RNASeqGene         Bob  samparray3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13  RNASeqGene     Barbara  samparray4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14  GISTIC            Jack       samp0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15  GISTIC             Bob       samp1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16  GISTIC            Jill       samp2</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Replacement method for a MultiAssayExperiment sampleMap</span></span></span>
<span class="r-in"><span><span class="fu">sampleMap</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> harmonizing input:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   removing 4 colData rownames not in sampleMap 'primary'</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Access the ExperimentList</span></span></span>
<span class="r-in"><span><span class="fu">experiments</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExperimentList class object of length 0:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Replace with an empty ExperimentList</span></span></span>
<span class="r-in"><span><span class="fu">experiments</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="ExperimentList.html">ExperimentList</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Access the metadata</span></span></span>
<span class="r-in"><span><span class="fu">metadata</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> list()</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Replace metadata with a list</span></span></span>
<span class="r-in"><span><span class="fu">metadata</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>runDate <span class="op">=</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/format.html" class="external-link">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.time</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"%B %d, %Y"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Access the colData</span></span></span>
<span class="r-in"><span><span class="fu">colData</span><span class="op">(</span><span class="va">mae</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 0 rows and 2 columns</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Access a column in colData</span></span></span>
<span class="r-in"><span><span class="va">mae</span><span class="op">$</span><span class="va">age</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> integer(0)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Replace a column in colData</span></span></span>
<span class="r-in"><span><span class="va">mae</span><span class="op">$</span><span class="va">age</span> <span class="op">&lt;-</span> <span class="va">mae</span><span class="op">$</span><span class="va">age</span> <span class="op">+</span> <span class="fl">1</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

