<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Prepare a MultiAssayExperiment instance — prepMultiAssay • MultiAssayExperiment</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Prepare a MultiAssayExperiment instance — prepMultiAssay" />
<meta property="og:description" content="The purpose of this helper function is to faciltate the creation of a
MultiAssayExperiment object by detecting any inconsistencies
with all types of names in either the ExperimentList,
the colData, or sampleMap." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiAssayExperiment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.18.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/MultiAssayExperiment.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a>
    </li>
    <li>
      <a href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../news/index.html">Release Notes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/waldronlab/MultiAssayExperiment">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prepare a <code>MultiAssayExperiment</code> instance</h1>
    <small class="dont-index">Source: <a href='https://github.com/waldronlab/MultiAssayExperiment/blob/master/R/prepMultiAssay.R'><code>R/prepMultiAssay.R</code></a></small>
    <div class="hidden name"><code>prepMultiAssay.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The purpose of this helper function is to faciltate the creation of a
<code><a href='MultiAssayExperiment.html'>MultiAssayExperiment</a></code> object by detecting any inconsistencies
with all types of names in either the <code><a href='ExperimentList.html'>ExperimentList</a></code>,
the <code>colData</code>, or <code><a href='MultiAssayExperiment-methods.html'>sampleMap</a></code>.</p>
    </div>

    <pre class="usage"><span class='fu'>prepMultiAssay</span><span class='op'>(</span><span class='va'>ExperimentList</span>, <span class='va'>colData</span>, <span class='va'>sampleMap</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExperimentList</th>
      <td><p>A <code>list</code> of all combined experiments</p></td>
    </tr>
    <tr>
      <th>colData</th>
      <td><p>A DataFrame of the phenotype
data for all participants</p></td>
    </tr>
    <tr>
      <th>sampleMap</th>
      <td><p>A DataFrame of sample identifiers, assay
samples, and assay names</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Optional arguments for the <code>MultiAssayExperiment</code> constructor
function such as <code>metadata</code> and <code>drops</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>list</code> containing all the essential components of a
<code><a href='MultiAssayExperiment.html'>MultiAssayExperiment</a></code> as well as a "drops" metadata element that
indicates non-matched names. The names of the resulting list correspond to
the arguments of the <code>MultiAssayExperiment</code> constructor function.</p>
    <h2 class="hasAnchor" id="checks"><a class="anchor" href="#checks"></a>Checks</h2>

    

<p>The <code>prepMultiAssay</code> function checks that all columns in the sampleMap
are <code>character</code>.</p>
<p>It checks that all names and lengths match in both the
<code><a href='ExperimentList.html'>ExperimentList</a></code> and in the unique assay names of the
<code><a href='MultiAssayExperiment-methods.html'>sampleMap</a></code>.</p>
<p>If <code><a href='ExperimentList.html'>ExperimentList</a></code> names and assay names only differ by case
and are not duplicated, the function will standardize all names to
lowercase.</p>
<p>If names cannot be matched between the colname column of the
<code><a href='MultiAssayExperiment-methods.html'>sampleMap</a></code> and the colnames of the <code>ExperimentList</code>, those
unmatched will be dropped and found in the "drops" element of the
resulting <code>list</code>.</p>
<p>Names in the "primary" column of the <code><a href='MultiAssayExperiment-methods.html'>sampleMap</a></code>, will be
matched to those in the <code>colData</code>. Unmatched "primary" column rows will
be dropped from the <code><a href='MultiAssayExperiment-methods.html'>sampleMap</a></code>. Suggestions for name fixes in
either the <code><a href='ExperimentList.html'>ExperimentList</a></code> or colnames will be made when
necessary.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Run example</span>
<span class='fu'><a href='https://rdrr.io/r/utils/example.html'>example</a></span><span class='op'>(</span><span class='st'>"MultiAssayExperiment"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; MltAsE&gt; ## Run the example ExperimentList
#&gt; MltAsE&gt; example("ExperimentList")
#&gt; 
#&gt; ExprmL&gt; ## Create an empty ExperimentList instance
#&gt; ExprmL&gt; ExperimentList()
#&gt; ExperimentList class object of length 0:
#&gt;  
#&gt; ExprmL&gt; ## Create array matrix and AnnotatedDataFrame to create an ExpressionSet class
#&gt; ExprmL&gt; arraydat &lt;- matrix(data = seq(101, length.out = 20), ncol = 4,
#&gt; ExprmL+     dimnames = list(
#&gt; ExprmL+         c("ENST00000294241", "ENST00000355076",
#&gt; ExprmL+         "ENST00000383706","ENST00000234812", "ENST00000383323"),
#&gt; ExprmL+         c("array1", "array2", "array3", "array4")
#&gt; ExprmL+     ))
#&gt; 
#&gt; ExprmL&gt; colDat &lt;- data.frame(slope53 = rnorm(4),
#&gt; ExprmL+     row.names = c("array1", "array2", "array3", "array4"))
#&gt; 
#&gt; ExprmL&gt; ## SummarizedExperiment constructor
#&gt; ExprmL&gt; exprdat &lt;- SummarizedExperiment::SummarizedExperiment(arraydat,
#&gt; ExprmL+     colData = colDat)
#&gt; 
#&gt; ExprmL&gt; ## Create a sample methylation dataset
#&gt; ExprmL&gt; methyldat &lt;- matrix(data = seq(1, length.out = 25), ncol = 5,
#&gt; ExprmL+     dimnames = list(
#&gt; ExprmL+         c("ENST00000355076", "ENST00000383706",
#&gt; ExprmL+           "ENST00000383323", "ENST00000234812", "ENST00000294241"),
#&gt; ExprmL+         c("methyl1", "methyl2", "methyl3",
#&gt; ExprmL+           "methyl4", "methyl5")
#&gt; ExprmL+     ))
#&gt; 
#&gt; ExprmL&gt; ## Create a sample RNASeqGene dataset
#&gt; ExprmL&gt; rnadat &lt;- matrix(
#&gt; ExprmL+     data = sample(c(46851, 5, 19, 13, 2197, 507,
#&gt; ExprmL+         84318, 126, 17, 21, 23979, 614), size = 20, replace = TRUE),
#&gt; ExprmL+     ncol = 4,
#&gt; ExprmL+     dimnames = list(
#&gt; ExprmL+         c("XIST", "RPS4Y1", "KDM5D", "ENST00000383323", "ENST00000234812"),
#&gt; ExprmL+         c("samparray1", "samparray2", "samparray3", "samparray4")
#&gt; ExprmL+     ))
#&gt; 
#&gt; ExprmL&gt; ## Create a mock RangedSummarizedExperiment from a data.frame
#&gt; ExprmL&gt; rangedat &lt;- data.frame(chr="chr2", start = 11:15, end = 12:16,
#&gt; ExprmL+     strand = c("+", "-", "+", "*", "."),
#&gt; ExprmL+     samp0 = c(0,0,1,1,1), samp1 = c(1,0,1,0,1), samp2 = c(0,1,0,1,0),
#&gt; ExprmL+     row.names = c(paste0("ENST", "00000", 135411:135414), "ENST00000383323"))
#&gt; 
#&gt; ExprmL&gt; rangeSE &lt;- SummarizedExperiment::makeSummarizedExperimentFromDataFrame(rangedat)
#&gt; 
#&gt; ExprmL&gt; ## Combine to a named list and call the ExperimentList constructor function
#&gt; ExprmL&gt; assayList &lt;- list(Affy = exprdat, Methyl450k = methyldat, RNASeqGene = rnadat,
#&gt; ExprmL+                 GISTIC = rangeSE)
#&gt; 
#&gt; ExprmL&gt; ## Use the ExperimentList constructor
#&gt; ExprmL&gt; ExpList &lt;- ExperimentList(assayList)
#&gt; 
#&gt; MltAsE&gt; ## Create sample maps for each experiment
#&gt; MltAsE&gt; exprmap &lt;- data.frame(
#&gt; MltAsE+     primary = c("Jack", "Jill", "Barbara", "Bob"),
#&gt; MltAsE+     colname = c("array1", "array2", "array3", "array4"),
#&gt; MltAsE+     stringsAsFactors = FALSE)
#&gt; 
#&gt; MltAsE&gt; methylmap &lt;- data.frame(
#&gt; MltAsE+     primary = c("Jack", "Jack", "Jill", "Barbara", "Bob"),
#&gt; MltAsE+     colname = c("methyl1", "methyl2", "methyl3", "methyl4", "methyl5"),
#&gt; MltAsE+     stringsAsFactors = FALSE)
#&gt; 
#&gt; MltAsE&gt; rnamap &lt;- data.frame(
#&gt; MltAsE+     primary = c("Jack", "Jill", "Bob", "Barbara"),
#&gt; MltAsE+     colname = c("samparray1", "samparray2", "samparray3", "samparray4"),
#&gt; MltAsE+     stringsAsFactors = FALSE)
#&gt; 
#&gt; MltAsE&gt; gistmap &lt;- data.frame(
#&gt; MltAsE+     primary = c("Jack", "Bob", "Jill"),
#&gt; MltAsE+     colname = c("samp0", "samp1", "samp2"),
#&gt; MltAsE+     stringsAsFactors = FALSE)
#&gt; 
#&gt; MltAsE&gt; ## Combine as a named list and convert to a DataFrame
#&gt; MltAsE&gt; maplist &lt;- list(Affy = exprmap, Methyl450k = methylmap,
#&gt; MltAsE+     RNASeqGene = rnamap, GISTIC = gistmap)
#&gt; 
#&gt; MltAsE&gt; ## Create a sampleMap
#&gt; MltAsE&gt; sampMap &lt;- listToMap(maplist)
#&gt; 
#&gt; MltAsE&gt; ## Create an example phenotype data
#&gt; MltAsE&gt; colDat &lt;- data.frame(sex = c("M", "F", "M", "F"), age = 38:41,
#&gt; MltAsE+     row.names = c("Jack", "Jill", "Bob", "Barbara"))
#&gt; 
#&gt; MltAsE&gt; ## Create a MultiAssayExperiment instance
#&gt; MltAsE&gt; mae &lt;- MultiAssayExperiment(experiments = ExpList, colData = colDat,
#&gt; MltAsE+     sampleMap = sampMap)</div><div class='input'>
<span class='co'>## Check if there are any inconsistencies within the different names</span>
<span class='va'>preparedMAE</span> <span class='op'>&lt;-</span> <span class='fu'>prepMultiAssay</span><span class='op'>(</span><span class='va'>ExpList</span>, <span class='va'>colDat</span>, <span class='va'>sampMap</span><span class='op'>)</span>

<span class='co'>## Results in a list of components for the MultiAssayExperiment constructor</span>
<span class='co'>## function</span>
<span class='fu'><a href='MultiAssayExperiment.html'>MultiAssayExperiment</a></span><span class='op'>(</span><span class='va'>preparedMAE</span><span class='op'>$</span><span class='va'>experiments</span>, <span class='va'>preparedMAE</span><span class='op'>$</span><span class='va'>colData</span>,
<span class='va'>preparedMAE</span><span class='op'>$</span><span class='va'>sampleMap</span><span class='op'>)</span>
</div><div class='output co'>#&gt; A MultiAssayExperiment object of 4 listed
#&gt;  experiments with user-defined names and respective classes.
#&gt;  Containing an ExperimentList class object of length 4:
#&gt;  [1] Affy: SummarizedExperiment with 5 rows and 4 columns
#&gt;  [2] Methyl450k: matrix with 5 rows and 5 columns
#&gt;  [3] RNASeqGene: matrix with 5 rows and 4 columns
#&gt;  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 3 columns
#&gt; Functionality:
#&gt;  experiments() - obtain the ExperimentList instance
#&gt;  colData() - the primary/phenotype DataFrame
#&gt;  sampleMap() - the sample coordination DataFrame
#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment
#&gt;  *Format() - convert into a long or wide DataFrame
#&gt;  assays() - convert ExperimentList to a SimpleList of matrices
#&gt;  exportClass() - save all data to files</div><div class='input'>
<span class='co'>## Alternatively, use the do.call function</span>
<span class='fu'><a href='https://rdrr.io/r/base/do.call.html'>do.call</a></span><span class='op'>(</span><span class='va'>MultiAssayExperiment</span>, <span class='va'>preparedMAE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; A MultiAssayExperiment object of 4 listed
#&gt;  experiments with user-defined names and respective classes.
#&gt;  Containing an ExperimentList class object of length 4:
#&gt;  [1] Affy: SummarizedExperiment with 5 rows and 4 columns
#&gt;  [2] Methyl450k: matrix with 5 rows and 5 columns
#&gt;  [3] RNASeqGene: matrix with 5 rows and 4 columns
#&gt;  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 3 columns
#&gt; Functionality:
#&gt;  experiments() - obtain the ExperimentList instance
#&gt;  colData() - the primary/phenotype DataFrame
#&gt;  sampleMap() - the sample coordination DataFrame
#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment
#&gt;  *Format() - convert into a long or wide DataFrame
#&gt;  assays() - convert ExperimentList to a SimpleList of matrices
#&gt;  exportClass() - save all data to files</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


