<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Prepare a MultiAssayExperiment instance — prepMultiAssay • MultiAssayExperiment</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare a MultiAssayExperiment instance — prepMultiAssay"><meta name="description" content="The purpose of this helper function is to faciltate the creation of a
MultiAssayExperiment object by detecting any inconsistencies
with all types of names in either the ExperimentList,
the colData, or sampleMap."><meta property="og:description" content="The purpose of this helper function is to faciltate the creation of a
MultiAssayExperiment object by detecting any inconsistencies
with all types of names in either the ExperimentList,
the colData, or sampleMap."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="default" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MultiAssayExperiment</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.30.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Quick Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes"><li><a class="dropdown-item" href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a></li>
    <li><a class="dropdown-item" href="../articles/MultiAssayExperiment.html">Getting Started</a></li>
    <li><a class="dropdown-item" href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a></li>
    <li><a class="dropdown-item" href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a></li>
  </ul></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news"><li><a class="dropdown-item" href="../news/index.html">Release Notes</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="nav-link" href="https://github.com/waldronlab/MultiAssayExperiment"><span class="fa fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Prepare a <code>MultiAssayExperiment</code> instance</h1>
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/RELEASE_3_19/R/prepMultiAssay.R"><code>R/prepMultiAssay.R</code></a></small>
      <div class="d-none name"><code>prepMultiAssay.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The purpose of this helper function is to faciltate the creation of a
<code><a href="MultiAssayExperiment.html">MultiAssayExperiment</a></code> object by detecting any inconsistencies
with all types of names in either the <code><a href="ExperimentList.html">ExperimentList</a></code>,
the <code>colData</code>, or <code><a href="MultiAssayExperiment-methods.html">sampleMap</a></code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prepMultiAssay</span><span class="op">(</span><span class="va">ExperimentList</span>, <span class="va">colData</span>, <span class="va">sampleMap</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-experimentlist">ExperimentList<a class="anchor" aria-label="anchor" href="#arg-experimentlist"></a></dt>
<dd><p>A <code>list</code> of all combined experiments</p></dd>


<dt id="arg-coldata">colData<a class="anchor" aria-label="anchor" href="#arg-coldata"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code> of the phenotype
data for all participants</p></dd>


<dt id="arg-samplemap">sampleMap<a class="anchor" aria-label="anchor" href="#arg-samplemap"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></code> of sample identifiers, assay
samples, and assay names</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Optional arguments for the <code>MultiAssayExperiment</code> constructor
function such as <code>metadata</code> and <code>drops</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code>list</code> containing all the essential components of a
<code><a href="MultiAssayExperiment.html">MultiAssayExperiment</a></code> as well as a "drops" metadata element that
indicates non-matched names. The names of the resulting list correspond to
the arguments of the <code>MultiAssayExperiment</code> constructor function.</p>
    </div>
    <div class="section level2">
    <h2 id="checks">Checks<a class="anchor" aria-label="anchor" href="#checks"></a></h2>


<p>The <code>prepMultiAssay</code> function checks that all columns in the sampleMap
are <code>character</code>.</p>
<p>It checks that all names and lengths match in both the
<code><a href="ExperimentList.html">ExperimentList</a></code> and in the unique assay names of the
<code><a href="MultiAssayExperiment-methods.html">sampleMap</a></code>.</p>
<p>If <code><a href="ExperimentList.html">ExperimentList</a></code> names and assay names only differ by case
and are not duplicated, the function will standardize all names to
lowercase.</p>
<p>If names cannot be matched between the colname column of the
<code><a href="MultiAssayExperiment-methods.html">sampleMap</a></code> and the colnames of the <code>ExperimentList</code>, those
unmatched will be dropped and found in the "drops" element of the
resulting <code>list</code>.</p>
<p>Names in the "primary" column of the <code><a href="MultiAssayExperiment-methods.html">sampleMap</a></code>, will be
matched to those in the <code>colData</code>. Unmatched "primary" column rows will
be dropped from the <code><a href="MultiAssayExperiment-methods.html">sampleMap</a></code>. Suggestions for name fixes in
either the <code><a href="ExperimentList.html">ExperimentList</a></code> or colnames will be made when
necessary.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Run example</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/example.html" class="external-link">example</a></span><span class="op">(</span><span class="st">"MultiAssayExperiment"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Run the example ExperimentList</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; example("ExperimentList")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create an empty ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExperimentList()</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExperimentList class object of length 0:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create array matrix and AnnotatedDataFrame to create an ExpressionSet class</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; arraydat &lt;- matrix(data = seq(101, length.out = 20), ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000294241", "ENST00000355076",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         "ENST00000383706","ENST00000234812", "ENST00000383323"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("array1", "array2", "array3", "array4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; colDat &lt;- data.frame(slope53 = rnorm(4),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c("array1", "array2", "array3", "array4"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## SummarizedExperiment constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; exprdat &lt;- SummarizedExperiment::SummarizedExperiment(arraydat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     colData = colDat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample methylation dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; methyldat &lt;- matrix(data = seq(1, length.out = 25), ncol = 5,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("ENST00000355076", "ENST00000383706",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "ENST00000383323", "ENST00000234812", "ENST00000294241"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("methyl1", "methyl2", "methyl3",</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+           "methyl4", "methyl5")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a sample RNASeqGene dataset</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rnadat &lt;- matrix(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     data = sample(c(46851, 5, 19, 13, 2197, 507,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         84318, 126, 17, 21, 23979, 614), size = 20, replace = TRUE),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ncol = 4,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     dimnames = list(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("XIST", "RPS4Y1", "KDM5D", "ENST00000383323", "ENST00000234812"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+         c("samparray1", "samparray2", "samparray3", "samparray4")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     ))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Create a mock RangedSummarizedExperiment from a data.frame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangedat &lt;- data.frame(chr="chr2", start = 11:15, end = 12:16,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     strand = c("+", "-", "+", "*", "."),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     samp0 = c(0,0,1,1,1), samp1 = c(1,0,1,0,1), samp2 = c(0,1,0,1,0),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+     row.names = c(paste0("ENST", "00000", 135411:135414), "ENST00000383323"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; rangeSE &lt;- SummarizedExperiment::makeSummarizedExperimentFromDataFrame(rangedat)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Combine to a named list and call the ExperimentList constructor function</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; assayList &lt;- list(Affy = exprdat, Methyl450k = methyldat, RNASeqGene = rnadat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL+                 GISTIC = rangeSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ## Use the ExperimentList constructor</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ExprmL&gt; ExpList &lt;- ExperimentList(assayList)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create sample maps for each experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; exprmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("array1", "array2", "array3", "array4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; methylmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jack", "Jill", "Barbara", "Bob"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("methyl1", "methyl2", "methyl3", "methyl4", "methyl5"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; rnamap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Jill", "Bob", "Barbara"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samparray1", "samparray2", "samparray3", "samparray4"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; gistmap &lt;- data.frame(</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     primary = c("Jack", "Bob", "Jill"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     colname = c("samp0", "samp1", "samp2"),</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     stringsAsFactors = FALSE)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Combine as a named list and convert to a DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; maplist &lt;- list(Affy = exprmap, Methyl450k = methylmap,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     RNASeqGene = rnamap, GISTIC = gistmap)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a sampleMap</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; sampMap &lt;- listToMap(maplist)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create an example phenotype data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; colDat &lt;- data.frame(sex = c("M", "F", "M", "F"), age = 38:41,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     row.names = c("Jack", "Jill", "Bob", "Barbara"))</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; ## Create a MultiAssayExperiment instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE&gt; mae &lt;- MultiAssayExperiment(experiments = ExpList, colData = colDat,</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MltAsE+     sampleMap = sampMap)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Check if there are any inconsistencies within the different names</span></span></span>
<span class="r-in"><span><span class="va">preparedMAE</span> <span class="op">&lt;-</span> <span class="fu">prepMultiAssay</span><span class="op">(</span><span class="va">ExpList</span>, <span class="va">colDat</span>, <span class="va">sampMap</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Results in a list of components for the MultiAssayExperiment constructor</span></span></span>
<span class="r-in"><span><span class="co">## function</span></span></span>
<span class="r-in"><span><span class="fu"><a href="MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span><span class="va">preparedMAE</span><span class="op">$</span><span class="va">experiments</span>, <span class="va">preparedMAE</span><span class="op">$</span><span class="va">colData</span>,</span></span>
<span class="r-in"><span><span class="va">preparedMAE</span><span class="op">$</span><span class="va">sampleMap</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 4 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 5 rows and 5 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Alternatively, use the do.call function</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">MultiAssayExperiment</span>, <span class="va">preparedMAE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> A MultiAssayExperiment object of 4 listed</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments with user-defined names and respective classes.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Containing an ExperimentList class object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] Affy: SummarizedExperiment with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] Methyl450k: matrix with 5 rows and 5 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] RNASeqGene: matrix with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] GISTIC: RangedSummarizedExperiment with 5 rows and 3 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Functionality:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  experiments() - obtain the ExperimentList instance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  colData() - the primary/phenotype DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  sampleMap() - the sample coordination DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  *Format() - convert into a long or wide DataFrame</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  exportClass() - save data to flat files</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Marcel Ramos, Martin Morgan, Vincent J Carey, Levi Waldron.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

