<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MultiAssayExperiment: The Integrative Bioconductor Container • MultiAssayExperiment</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MultiAssayExperiment: The Integrative Bioconductor Container">
<meta property="og:description" content="MultiAssayExperiment">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiAssayExperiment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.26.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/MultiAssayExperiment.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a>
    </li>
    <li>
      <a href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Release Notes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/MultiAssayExperiment" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MultiAssayExperiment: The Integrative
Bioconductor Container</h1>
                        <h4 data-toc-skip class="author">MultiAssay
Special Interest Group</h4>
            
            <h4 data-toc-skip class="date">April 28, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/HEAD/vignettes/MultiAssayExperiment.Rmd" class="external-link"><code>vignettes/MultiAssayExperiment.Rmd</code></a></small>
      <div class="hidden name"><code>MultiAssayExperiment.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MultiAssayExperiment"</span><span class="op">)</span></span></code></pre></div>
<p>Loading the packages:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RaggedExperiment</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="a-brief-description">A Brief Description<a class="anchor" aria-label="anchor" href="#a-brief-description"></a>
</h2>
<p><code>MultiAssayExperiment</code> offers a data structure for
representing and analyzing multi-omics experiments: a biological
analysis approach utilizing multiple types of observations, such as DNA
mutations and abundance of RNA and proteins, in the same biological
specimens.</p>
<div class="section level3">
<h3 id="choosing-the-appropriate-data-structure">Choosing the appropriate data structure<a class="anchor" aria-label="anchor" href="#choosing-the-appropriate-data-structure"></a>
</h3>
<p>For assays with different numbers of rows and even columns,
<code>MultiAssayExperiment</code> is recommended. For sets of assays
with the same information across all rows (e.g., genes or genomic
ranges), <code>SummarizedExperiment</code> is the recommended data
structure.</p>
</div>
</div>
<div class="section level2">
<h2 id="overview-of-the-multiassayexperiment-class">Overview of the <code>MultiAssayExperiment</code> class<a class="anchor" aria-label="anchor" href="#overview-of-the-multiassayexperiment-class"></a>
</h2>
<p>Here is an overview of the class and its constructors and
extractors:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">empty</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">empty</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 0 listed</span></span>
<span><span class="co">##  experiments with no user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 0:</span></span>
<span><span class="co">##  Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/methods/slot.html" class="external-link">slotNames</a></span><span class="op">(</span><span class="va">empty</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ExperimentList" "colData"        "sampleMap"      "drops"         </span></span>
<span><span class="co">## [5] "metadata"</span></span></code></pre>
<p>A visual representation of the MultiAssayExperiment class and its
accessor functions can be seen below. There are three main
components:</p>
<ol style="list-style-type: decimal">
<li><code>ExperimentList</code></li>
<li><code>colData</code></li>
<li><code>sampleMap</code></li>
</ol>
<div class="figure">
<img src="MultiAssayExperiment.png" alt="MultiAssayExperiment object schematic shows the design of the infrastructure class. The colData provides data about the patients, cell lines, or other biological units, with one row per unit and one column per variable. The experiments are a list of assay datasets of arbitrary class, with one column per observation. The sampleMap links a single table of patient data (colData) to a list of experiments via a simple but powerful table of experiment:patient edges (relationships), that can be created automatically in simple cases or in a spreadsheet if assay-specific sample identifiers are used. sampleMap relates each column (observation) in the assays (experiments) to exactly one row (biological unit) in colData; however, one row of colData may map to zero, one, or more columns per assay, allowing for missing and replicate assays. Green stripes indicate a mapping of one subject to multiple observations across experiments." width="\maxwidth"><p class="caption">
MultiAssayExperiment object schematic shows the design of the
infrastructure class. The colData provides data about the patients, cell
lines, or other biological units, with one row per unit and one column
per variable. The experiments are a list of assay datasets of arbitrary
class, with one column per observation. The sampleMap links a single
table of patient data (colData) to a list of experiments via a simple
but powerful table of experiment:patient edges (relationships), that can
be created automatically in simple cases or in a spreadsheet if
assay-specific sample identifiers are used. sampleMap relates each
column (observation) in the assays (experiments) to exactly one row
(biological unit) in colData; however, one row of colData may map to
zero, one, or more columns per assay, allowing for missing and replicate
assays. Green stripes indicate a mapping of one subject to multiple
observations across experiments.
</p>
</div>
<div class="section level3">
<h3 id="components-of-the-multiassayexperiment">Components of the <code>MultiAssayExperiment</code><a class="anchor" aria-label="anchor" href="#components-of-the-multiassayexperiment"></a>
</h3>
<div class="section level4">
<h4 id="experimentlist-experimental-data">
<code>ExperimentList</code>: experimental data<a class="anchor" aria-label="anchor" href="#experimentlist-experimental-data"></a>
</h4>
<p>The <code>ExperimentList</code> slot and class is the container
workhorse for the <code>MultiAssayExperiment</code> class. It contains
all the experimental data. It inherits from class
<code><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">S4Vectors::SimpleList</a></code> with one element/component per data
type.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">experiments</a></span><span class="op">(</span><span class="va">empty</span><span class="op">)</span><span class="op">)</span> <span class="co"># ExperimentList</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ExperimentList"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "MultiAssayExperiment"</span></span></code></pre>
<p>The elements of the <code>ExperimentList</code> can contain
<strong>ID-based</strong> and <strong>range-based</strong> data.
Requirements for all classes in the <code>ExperimentList</code> are
listed in the API.</p>
<p>The following base and Bioconductor classes are known to work as
elements of the ExperimentList:</p>
<ul>
<li><p><code><a href="https://rdrr.io/r/base/matrix.html" class="external-link">base::matrix</a></code>: the base class, can be used for
ID-based datasets such as gene expression summarized per-gene, microRNA,
metabolomics, or microbiome data.</p></li>
<li><p><code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment::SummarizedExperiment</a></code>: A richer
representation compared to a ordinary matrix of ID-based datasets
capable of storing additional assay- level metadata.</p></li>
<li><p><code><a href="https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html" class="external-link">Biobase::ExpressionSet</a></code>: A <em>legacy</em>
representation of ID-based datasets, supported for convenience and
supplanted by <code>SummarizedExperiment</code>.</p></li>
<li><p><code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment::RangedSummarizedExperiment</a></code>:
For rectangular range-based datasets, one set of genomic ranges are
assayed for multiple samples. It can be used for gene expression,
methylation, or other data types that refer to genomic
positions.</p></li>
<li><p><code><a href="https://rdrr.io/pkg/RaggedExperiment/man/RaggedExperiment-class.html" class="external-link">RaggedExperiment::RaggedExperiment</a></code>: For range-based
datasets, such as copy number and mutation data, the
<code>RaggedExperiment</code> class can be used to represent
measurements by genomic positions.</p></li>
</ul>
<div class="section level5">
<h5 id="class-requirements-within-experimentlist-container">Class requirements within <code>ExperimentList</code> container<a class="anchor" aria-label="anchor" href="#class-requirements-within-experimentlist-container"></a>
</h5>
<p>See the <a href="#application-programming-interface-api">API
section</a> for details on requirements for using other data classes. In
general, data classes meeting minimum requirements, including support
for square bracket <code>[</code> subsetting and <code><a href="https://rdrr.io/r/base/dimnames.html" class="external-link">dimnames()</a></code>
will work by default.</p>
<p>The datasets contained in elements of the <code>ExperimentList</code>
can have:</p>
<ul>
<li>column names (required)</li>
<li>row names (optional)</li>
</ul>
<p>The column names correspond to samples, and are used to match assay
data to specimen metadata stored in <code>colData</code>.</p>
<p>The row names can correspond to a variety of features in the data
including but not limited to gene names, probe IDs, proteins, and named
ranges. Note that the existence of “row” names does <em>not</em> mean
the data must be rectangular or matrix-like.</p>
<p>Classes contained in the <code>ExperimentList</code> must support the
following list of methods:</p>
<ul>
<li>
<code>[</code>: single square bracket subsetting, with a single
comma. It is assumed that values before the comma subset rows, and
values after the comma subset columns.</li>
<li>
<code><a href="https://rdrr.io/r/base/dimnames.html" class="external-link">dimnames()</a></code> : corresponding to features (such as genes,
proteins, etc.) and experimental samples</li>
<li>
<code><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim()</a></code>: returns a vector of the number of rows and
number of columns</li>
</ul>
</div>
</div>
<div class="section level4">
<h4 id="coldata-primary-data">
<code>colData</code>: primary data<a class="anchor" aria-label="anchor" href="#coldata-primary-data"></a>
</h4>
<p>The <code>MultiAssayExperiment</code> keeps one set of “primary”
metadata that describes the ‘biological unit’ which can refer to
specimens, experimental subjects, patients, etc. In this vignette, we
will refer to each experimental subject as a <em>patient</em>.</p>
<div class="section level5">
<h5 id="coldata-slot-requirements">
<code>colData</code> slot requirements<a class="anchor" aria-label="anchor" href="#coldata-slot-requirements"></a>
</h5>
<p>The <code>colData</code> dataset should be of class
<code>DataFrame</code> but can accept a <code>data.frame</code> class
object that will be coerced.</p>
<p>In order to relate metadata of the biological unit, the row names of
the <code>colData</code> dataset must contain patient identifiers.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">patient.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>sex<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"M"</span>, <span class="st">"F"</span>, <span class="st">"M"</span>, <span class="st">"F"</span><span class="op">)</span>,</span>
<span>    age<span class="op">=</span><span class="fl">38</span><span class="op">:</span><span class="fl">41</span>,</span>
<span>    row.names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jill"</span>, <span class="st">"Bob"</span>, <span class="st">"Barbara"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">patient.data</span></span></code></pre></div>
<pre><code><span><span class="co">##         sex age</span></span>
<span><span class="co">## Jack      M  38</span></span>
<span><span class="co">## Jill      F  39</span></span>
<span><span class="co">## Bob       M  40</span></span>
<span><span class="co">## Barbara   F  41</span></span></code></pre>
<p>Key points:</p>
<ul>
<li>one row of <code>colData</code> <em>can</em> map to zero, one, or
more columns in any <code>ExperimentList</code> element</li>
<li>each row of <code>colData</code> <em>must</em> map to at least one
column in at least one <code>ExperimentList</code> element.</li>
<li>each column of each <code>ExperimentList</code> element
<em>must</em> map to <em>exactly</em> one row of
<code>colData</code>.</li>
</ul>
<p>These relationships are defined by the <a href="#sampleMap">sampleMap</a>.</p>
</div>
<div class="section level5">
<h5 id="note-on-the-flexibility-of-the-dataframe">Note on the flexibility of the <code>DataFrame</code><a class="anchor" aria-label="anchor" href="#note-on-the-flexibility-of-the-dataframe"></a>
</h5>
<p>For many typical purposes the <code>DataFrame</code> and
<code>data.frame</code> behave equivalently; but the
<code>Dataframe</code> is more flexible as it allows any vector-like
data type to be stored in its columns. The flexibility of the
<code>DataFrame</code> permits, for example, storing multiple
dose-response values for a single cell line, even if the number of doses
and responses is not consistent across all cell lines. Doses could be
stored in one column of <code>colData</code> as a
<code>SimpleList</code>, and responses in another column, also as a
<code>SimpleList</code>. Or, dose-response values could be stored in a
single column of <code>colData</code> as a two-column matrix for each
cell line.</p>
</div>
</div>
<div class="section level4">
<h4 id="sampleMap">
<code>sampleMap</code>: relating <code>colData</code> to multiple
assays<a class="anchor" aria-label="anchor" href="#sampleMap"></a>
</h4>
<p>The <code>sampleMap</code> is a <code>DataFrame</code> that relates
the “primary” data (<code>colData</code>) to the experimental
assays:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/methods/is.html" class="external-link">is</a></span><span class="op">(</span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">empty</span><span class="op">)</span>, <span class="st">"DataFrame"</span><span class="op">)</span> <span class="co"># TRUE</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<p>The <code>sampleMap</code> provides an unambiguous map from every
experimental observation to <em>one and only one</em> row in
<code>colData</code>. It is, however, permissible for a row of
<code>colData</code> to be associated with multiple experimental
observations or no observations at all. In other words, there is a
“many-to-one” mapping from experimental observations to rows of
<code>colData</code>, and a “one-to-any-number” mapping from rows of
<code>colData</code> to experimental observations.</p>
<div class="section level5">
<h5 id="samplemap-structure">
<code>sampleMap</code> structure<a class="anchor" aria-label="anchor" href="#samplemap-structure"></a>
</h5>
<p>The <code>sampleMap</code> has three columns, with the following
column names:</p>
<ol style="list-style-type: decimal">
<li><p><strong>assay</strong> provides the names of the different
experiments / assays performed. These are user-defined, with the only
requirement that the names of the <code>ExperimentList</code>, where the
experimental assays are stored, must be contained in this
column.</p></li>
<li><p><strong>primary</strong> provides the “primary” sample names. All
values in this column must also be present in the rownames of
<code>colData(MultiAssayExperiment)</code>. In this example, allowable
values in this column are “Jack”, “Jill”, “Barbara”, and “Bob”.</p></li>
<li><p><strong>colname</strong> provides the sample names used by
experimental datasets, which in practice are often different than the
primary sample names. For each assay, all column names must be found in
this column. Otherwise, those assays would be orphaned: it would be
impossible to match them up to samples in the overall experiment. As
mentioned above, duplicate values are allowed, to represent replicates
with the same overall experiment-level annotation.</p></li>
</ol>
<p>This design is motivated by the following situations:</p>
<ol style="list-style-type: decimal">
<li>It allows flexibility for any amount of technical replication and
biological replication (such as tumor and matched normal for a single
patient) of individual assays.</li>
<li>It allows missing observations (such as RNA-seq performed only for
some of the patients).</li>
<li>It allows the use of different identifiers to be used for patients /
specimens and for each assay. These different identifiers are matched
unambiguously, and consistency between them is maintained during
subsetting and re-ordering.</li>
</ol>
<div class="section level6">
<h6 id="instances-where-samplemap-isnt-provided">Instances where <code>sampleMap</code> isn’t provided<a class="anchor" aria-label="anchor" href="#instances-where-samplemap-isnt-provided"></a>
</h6>
<p>If each assay uses the same colnames (i.e., if the same sample
identifiers are used for each experiment), a simple list of these
datasets is sufficient for the <code>MultiAssayExperiment</code>
constructor function. It is not necessary for them to have the same
rownames or colnames:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exprss1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">16</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">4</span>,</span>
<span>        dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"ENST00000%i"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">288754</span><span class="op">:</span><span class="fl">290000</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jill"</span>, <span class="st">"Bob"</span>, <span class="st">"Bobby"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">exprss2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">12</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">3</span>,</span>
<span>        dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"ENST00000%i"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">288754</span><span class="op">:</span><span class="fl">290000</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jane"</span>, <span class="st">"Bob"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">doubleExp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"methyl 2k"</span>  <span class="op">=</span> <span class="va">exprss1</span>, <span class="st">"methyl 3k"</span> <span class="op">=</span> <span class="va">exprss2</span><span class="op">)</span></span>
<span><span class="va">simpleMultiAssay</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span>experiments<span class="op">=</span><span class="va">doubleExp</span><span class="op">)</span></span>
<span><span class="va">simpleMultiAssay</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] methyl 2k: matrix with 4 rows and 4 columns</span></span>
<span><span class="co">##  [2] methyl 3k: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>In the above example, the user did not provide the
<code>colData</code> argument so the constructor function filled it with
an empty <code>DataFrame</code>:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">simpleMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 5 rows and 0 columns</span></span></code></pre>
<p>But the <code>colData</code> can be provided. Here, note that any
assay sample (column) that cannot be mapped to a corresponding row in
the provided <code>colData</code> gets dropped. This is part of ensuring
internal validity of the <code>MultiAssayExperiment</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simpleMultiAssay2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span>experiments<span class="op">=</span><span class="va">doubleExp</span>,</span>
<span>                                          colData<span class="op">=</span><span class="va">patient.data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Data dropped from ExperimentList (element - column):</span></span>
<span><span class="co">##  methyl 2k - Bobby</span></span>
<span><span class="co">##  methyl 3k - Jane</span></span>
<span><span class="co">## Unable to map to rows of colData</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 1 colData rownames not in sampleMap 'primary'</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simpleMultiAssay2</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] methyl 2k: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [2] methyl 3k: matrix with 4 rows and 2 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">simpleMultiAssay2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 3 rows and 2 columns</span></span>
<span><span class="co">##              sex       age</span></span>
<span><span class="co">##      &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## Jack           M        38</span></span>
<span><span class="co">## Jill           F        39</span></span>
<span><span class="co">## Bob            M        40</span></span></code></pre>
</div>
</div>
</div>
<div class="section level4">
<h4 id="metadata">metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h4>
<p>Metadata can be added at different levels of the
<code>MultiAssayExperiment</code>.</p>
<p>Can be of <em>ANY</em> class, for storing study-wide metadata, such
as citation information. For an empty <code>MultiAssayExperiment</code>
object, it is NULL.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">empty</span><span class="op">)</span><span class="op">)</span> <span class="co"># NULL (class "ANY")</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "list"</span></span></code></pre>
<p>At the <code>ExperimentList</code> level, the <code>metadata</code>
function would allow the user to enter metadata as a
<code>list</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">experiments</a></span><span class="op">(</span><span class="va">empty</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## list()</span></span></code></pre>
<p>At the individual assay level, certain classes <em>may</em> support
metadata, for example, <code>metadata</code> and <code>mcols</code> for
a <code>SummarizedExperiment</code>. It is recommended to use
<code>metadata</code> at the <code>ExperimentList</code> level.</p>
<p style="text-align: right;">
<a href="#header">back to top</a>
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="creating-a-multiassayexperiment-object-a-rich-example">Creating a <code>MultiAssayExperiment</code> object: a rich
example<a class="anchor" aria-label="anchor" href="#creating-a-multiassayexperiment-object-a-rich-example"></a>
</h2>
<p>In this section we demonstrate all core supported data classes, using
different sample ID conventions for each assay, with primary
<code>colData</code>. The some supported classes such as,
<code>matrix</code>, <code>SummarizedExperiment</code>, and
<code>RangedSummarizedExperiment</code>.</p>
<div class="section level3">
<h3 id="create-toy-datasets-demonstrating-all-supported-data-types">Create toy datasets demonstrating all supported data types<a class="anchor" aria-label="anchor" href="#create-toy-datasets-demonstrating-all-supported-data-types"></a>
</h3>
<p>We have three matrix-like datasets. First, let’s represent expression
data as a <code>SummarizedExperiment</code>:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">arraydat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">101</span>, <span class="fl">108</span><span class="op">)</span>, ncol<span class="op">=</span><span class="fl">4</span>,</span>
<span>    dimnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENST00000294241"</span>, <span class="st">"ENST00000355076"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"array1"</span>, <span class="st">"array2"</span>, <span class="st">"array3"</span>, <span class="st">"array4"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 array1 array2 array3 array4</span></span>
<span><span class="co">## ENST00000294241    101    103    105    107</span></span>
<span><span class="co">## ENST00000355076    102    104    106    108</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coldat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>slope53<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span>,</span>
<span>    row.names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"array1"</span>, <span class="st">"array2"</span>, <span class="st">"array3"</span>, <span class="st">"array4"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">exprdat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span><span class="va">arraydat</span>, colData<span class="op">=</span><span class="va">coldat</span><span class="op">)</span></span>
<span><span class="va">exprdat</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 2 4 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(2): ENST00000294241 ENST00000355076</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(4): array1 array2 array3 array4</span></span>
<span><span class="co">## colData names(1): slope53</span></span></code></pre>
<p>The following map matches <code>colData</code> sample names to
<code>exprdata</code> sample names. Note that row orders aren’t
initially matched up, and this is OK.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">exprmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>primary<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">patient.data</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">3</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                       colname<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"array1"</span>, <span class="st">"array2"</span>, <span class="st">"array3"</span>, <span class="st">"array4"</span><span class="op">)</span>,</span>
<span>                       stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   primary colname</span></span>
<span><span class="co">## 1    Jack  array1</span></span>
<span><span class="co">## 2    Jill  array2</span></span>
<span><span class="co">## 3 Barbara  array3</span></span>
<span><span class="co">## 4     Bob  array4</span></span></code></pre>
<p>Now methylation data, which we will represent as a
<code>matrix</code>. It uses gene identifiers also, but measures a
partially overlapping set of genes. Now, let’s store this as a simple
matrix which can contains a replicate for one of the patients.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">methyldat</span> <span class="op">&lt;-</span></span>
<span>   <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, ncol<span class="op">=</span><span class="fl">5</span>,</span>
<span>          dimnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENST00000355076"</span>, <span class="st">"ENST00000383706"</span><span class="op">)</span>,</span>
<span>                        <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"methyl1"</span>, <span class="st">"methyl2"</span>, <span class="st">"methyl3"</span>,</span>
<span>                          <span class="st">"methyl4"</span>, <span class="st">"methyl5"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 methyl1 methyl2 methyl3 methyl4 methyl5</span></span>
<span><span class="co">## ENST00000355076       1       3       5       7       9</span></span>
<span><span class="co">## ENST00000383706       2       4       6       8      10</span></span></code></pre>
<p>The following map matches <code>colData</code> sample names to
<code>methyldat</code> sample names.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">methylmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jack"</span>, <span class="st">"Jill"</span>, <span class="st">"Barbara"</span>, <span class="st">"Bob"</span><span class="op">)</span>,</span>
<span>    colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"methyl1"</span>, <span class="st">"methyl2"</span>, <span class="st">"methyl3"</span>, <span class="st">"methyl4"</span>, <span class="st">"methyl5"</span><span class="op">)</span>,</span>
<span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   primary colname</span></span>
<span><span class="co">## 1    Jack methyl1</span></span>
<span><span class="co">## 2    Jack methyl2</span></span>
<span><span class="co">## 3    Jill methyl3</span></span>
<span><span class="co">## 4 Barbara methyl4</span></span>
<span><span class="co">## 5     Bob methyl5</span></span></code></pre>
<p>Now we have a microRNA platform, which has no common identifiers with
the other datasets, and which we also represent as a
<code>matrix</code>. It is also missing data for “Jill”. We will use the
same sample naming convention as we did for arrays.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">microdat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">201</span><span class="op">:</span><span class="fl">212</span>, ncol<span class="op">=</span><span class="fl">3</span>,</span>
<span>                    dimnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"hsa-miR-21"</span>, <span class="st">"hsa-miR-191"</span>,</span>
<span>                                    <span class="st">"hsa-miR-148a"</span>, <span class="st">"hsa-miR148b"</span><span class="op">)</span>,</span>
<span>                                  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"micro1"</span>, <span class="st">"micro2"</span>, <span class="st">"micro3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##              micro1 micro2 micro3</span></span>
<span><span class="co">## hsa-miR-21      201    205    209</span></span>
<span><span class="co">## hsa-miR-191     202    206    210</span></span>
<span><span class="co">## hsa-miR-148a    203    207    211</span></span>
<span><span class="co">## hsa-miR148b     204    208    212</span></span></code></pre>
<p>And the following map matches <code>colData</code> sample names to
<code>microdat</code> sample names.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">micromap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Barbara"</span>, <span class="st">"Bob"</span><span class="op">)</span>,</span>
<span>    colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"micro1"</span>, <span class="st">"micro2"</span>, <span class="st">"micro3"</span><span class="op">)</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   primary colname</span></span>
<span><span class="co">## 1    Jack  micro1</span></span>
<span><span class="co">## 2 Barbara  micro2</span></span>
<span><span class="co">## 3     Bob  micro3</span></span></code></pre>
<p>Finally, we create a dataset of class
<code>RangedSummarizedExperiment</code>:</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nrows</span> <span class="op">&lt;-</span> <span class="fl">5</span>; <span class="va">ncols</span> <span class="op">&lt;-</span> <span class="fl">4</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">nrows</span> <span class="op">*</span> <span class="va">ncols</span>, <span class="fl">1</span>, <span class="fl">1e4</span><span class="op">)</span>, <span class="va">nrows</span><span class="op">)</span></span>
<span><span class="va">rowRanges</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="va">nrows</span> <span class="op">-</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">floor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">nrows</span>, <span class="fl">1e5</span>, <span class="fl">1e6</span><span class="op">)</span><span class="op">)</span>, width<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,</span>
<span>    strand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span><span class="op">)</span>, <span class="va">nrows</span>, <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>    feature_id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"ID\\%03d"</span>, <span class="fl">1</span><span class="op">:</span><span class="va">nrows</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">rowRanges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">letters</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="va">colData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span>Treatment<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ChIP"</span>, <span class="st">"Input"</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>    row.names<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mysnparray1"</span>, <span class="st">"mysnparray2"</span>, <span class="st">"mysnparray3"</span>, <span class="st">"mysnparray4"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">rse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span>assays<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">counts</span><span class="op">)</span>,</span>
<span>    rowRanges<span class="op">=</span><span class="va">rowRanges</span>, colData<span class="op">=</span><span class="va">colData</span><span class="op">)</span></span></code></pre></div>
<p>And we map the <code>colData</code> samples to the
<code>RangedSummarizedExperiment</code>:</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">rangemap</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>primary <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Jack"</span>, <span class="st">"Jill"</span>, <span class="st">"Bob"</span>, <span class="st">"Barbara"</span><span class="op">)</span>,</span>
<span>    colname <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mysnparray1"</span>, <span class="st">"mysnparray2"</span>, <span class="st">"mysnparray3"</span>, <span class="st">"mysnparray4"</span><span class="op">)</span>,</span>
<span>        stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   primary     colname</span></span>
<span><span class="co">## 1    Jack mysnparray1</span></span>
<span><span class="co">## 2    Jill mysnparray2</span></span>
<span><span class="co">## 3     Bob mysnparray3</span></span>
<span><span class="co">## 4 Barbara mysnparray4</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="samplemap-creation">
<code>sampleMap</code> creation<a class="anchor" aria-label="anchor" href="#samplemap-creation"></a>
</h3>
<p>The <code>MultiAssayExperiment</code> constructor function can create
the <code>sampleMap</code> automatically if a single naming convention
is used, but in this example it cannot because we used platform-specific
sample identifiers (e.g. mysnparray1, etc). So we must provide an ID map
that matches the samples of each experiment back to the
<code>colData</code>, as a three-column <code>data.frame</code> or
<code>DataFrame</code> with three columns named “assay”, primary”, and
“colname”. Here we start with a list:</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">listmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">exprmap</span>, <span class="va">methylmap</span>, <span class="va">micromap</span>, <span class="va">rangemap</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">listmap</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Affy"</span>, <span class="st">"Methyl 450k"</span>, <span class="st">"Mirna"</span>, <span class="st">"CNV gistic"</span><span class="op">)</span></span>
<span><span class="va">listmap</span></span></code></pre></div>
<pre><code><span><span class="co">## $Affy</span></span>
<span><span class="co">##   primary colname</span></span>
<span><span class="co">## 1    Jack  array1</span></span>
<span><span class="co">## 2    Jill  array2</span></span>
<span><span class="co">## 3 Barbara  array3</span></span>
<span><span class="co">## 4     Bob  array4</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`Methyl 450k`</span></span>
<span><span class="co">##   primary colname</span></span>
<span><span class="co">## 1    Jack methyl1</span></span>
<span><span class="co">## 2    Jack methyl2</span></span>
<span><span class="co">## 3    Jill methyl3</span></span>
<span><span class="co">## 4 Barbara methyl4</span></span>
<span><span class="co">## 5     Bob methyl5</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Mirna</span></span>
<span><span class="co">##   primary colname</span></span>
<span><span class="co">## 1    Jack  micro1</span></span>
<span><span class="co">## 2 Barbara  micro2</span></span>
<span><span class="co">## 3     Bob  micro3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`CNV gistic`</span></span>
<span><span class="co">##   primary     colname</span></span>
<span><span class="co">## 1    Jack mysnparray1</span></span>
<span><span class="co">## 2    Jill mysnparray2</span></span>
<span><span class="co">## 3     Bob mysnparray3</span></span>
<span><span class="co">## 4 Barbara mysnparray4</span></span></code></pre>
<p>and use the convenience function <code>listToMap</code> to convert
the list of <code>data.frame</code> objects to a valid object for the
<code>sampleMap</code>:</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dfmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mapToList.html">listToMap</a></span><span class="op">(</span><span class="va">listmap</span><span class="op">)</span></span>
<span><span class="va">dfmap</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 16 rows and 3 columns</span></span>
<span><span class="co">##           assay     primary     colname</span></span>
<span><span class="co">##        &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## 1   Affy               Jack      array1</span></span>
<span><span class="co">## 2   Affy               Jill      array2</span></span>
<span><span class="co">## 3   Affy            Barbara      array3</span></span>
<span><span class="co">## 4   Affy                Bob      array4</span></span>
<span><span class="co">## 5   Methyl 450k        Jack     methyl1</span></span>
<span><span class="co">## ...         ...         ...         ...</span></span>
<span><span class="co">## 12   Mirna              Bob      micro3</span></span>
<span><span class="co">## 13   CNV gistic        Jack mysnparray1</span></span>
<span><span class="co">## 14   CNV gistic        Jill mysnparray2</span></span>
<span><span class="co">## 15   CNV gistic         Bob mysnparray3</span></span>
<span><span class="co">## 16   CNV gistic     Barbara mysnparray4</span></span></code></pre>
<p>Note, <code>dfmap</code> can be reverted to a list with another
provided function:</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/mapToList.html">mapToList</a></span><span class="op">(</span><span class="va">dfmap</span>, <span class="st">"assay"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="experimental-data-as-a-list">Experimental data as a <code>list()</code><a class="anchor" aria-label="anchor" href="#experimental-data-as-a-list"></a>
</h3>
<p>Create an named list of experiments for the
<code>MultiAssayExperiment</code> function. All of these names must be
found within in the third column of <code>dfmap</code>:</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"Affy"</span> <span class="op">=</span> <span class="va">exprdat</span>, <span class="st">"Methyl 450k"</span> <span class="op">=</span> <span class="va">methyldat</span>,</span>
<span>    <span class="st">"Mirna"</span> <span class="op">=</span> <span class="va">microdat</span>, <span class="st">"CNV gistic"</span> <span class="op">=</span> <span class="va">rse</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="creation-of-the-multiassayexperiment-class-object">Creation of the <code>MultiAssayExperiment</code> class object<a class="anchor" aria-label="anchor" href="#creation-of-the-multiassayexperiment-class-object"></a>
</h3>
<p>We recommend using the <code>MultiAssayExperiment</code> constructor
function:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span><span class="va">objlist</span>, <span class="va">patient.data</span>, <span class="va">dfmap</span><span class="op">)</span></span>
<span><span class="va">myMultiAssay</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>The following extractor functions can be used to get extract data
from the object:</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">experiments</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 4 rows and 2 columns</span></span>
<span><span class="co">##                 sex       age</span></span>
<span><span class="co">##         &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## Jack              M        38</span></span>
<span><span class="co">## Jill              F        39</span></span>
<span><span class="co">## Bob               M        40</span></span>
<span><span class="co">## Barbara           F        41</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 16 rows and 3 columns</span></span>
<span><span class="co">##           assay     primary     colname</span></span>
<span><span class="co">##        &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## 1   Affy               Jack      array1</span></span>
<span><span class="co">## 2   Affy               Jill      array2</span></span>
<span><span class="co">## 3   Affy            Barbara      array3</span></span>
<span><span class="co">## 4   Affy                Bob      array4</span></span>
<span><span class="co">## 5   Methyl 450k        Jack     methyl1</span></span>
<span><span class="co">## ...         ...         ...         ...</span></span>
<span><span class="co">## 12   Mirna              Bob      micro3</span></span>
<span><span class="co">## 13   CNV gistic        Jack mysnparray1</span></span>
<span><span class="co">## 14   CNV gistic        Jill mysnparray2</span></span>
<span><span class="co">## 15   CNV gistic         Bob mysnparray3</span></span>
<span><span class="co">## 16   CNV gistic     Barbara mysnparray4</span></span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## list()</span></span></code></pre>
<p>Note that the <code>ExperimentList</code> class extends the
<code>SimpleList</code> class to add some validity checks specific to
<code>MultiAssayExperiment</code>. It can be used like a list.</p>
</div>
<div class="section level3">
<h3 id="helper-function-to-create-a-multiassayexperiment-object">Helper function to create a <code>MultiAssayExperiment</code>
object<a class="anchor" aria-label="anchor" href="#helper-function-to-create-a-multiassayexperiment-object"></a>
</h3>
<p>The <code>prepMultiAssay</code> function helps diagnose common
problems when creating a <code>MultiAssayExperiment</code> object. It
provides error messages and/or warnings in instances where names (either
<code>colnames</code> or <code>ExperimentList</code> element names) are
inconsistent with those found in the sampleMap. Input arguments are the
same as those in the <code>MultiAssayExperiment</code> (i.e.,
<code>ExperimentList</code>, <code>colData</code>,
<code>sampleMap</code>). The resulting output of the
<code>prepMultiAssay</code> function is a list of inputs including a
“metadata$drops” element for names that were not able to be matched.</p>
<p>Instances where <code>ExperimentList</code> is created without names
will prompt an error from <code>prepMultiAssay</code>. Named
<code>ExperimentList</code> elements are essential for checks in
<code>MultiAssayExperiment</code>.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objlist3</span> <span class="op">&lt;-</span> <span class="va">objlist</span></span>
<span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">objlist3</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="fu"><a href="../reference/prepMultiAssay.html">prepMultiAssay</a></span><span class="op">(</span><span class="va">objlist3</span>, <span class="va">patient.data</span>, <span class="va">dfmap</span><span class="op">)</span><span class="op">$</span><span class="va">experiments</span>,</span>
<span>    outFile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/showConnections.html" class="external-link">stdout</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Error in prepMultiAssay(objlist3, patient.data, dfmap) : </span></span>
<span><span class="co">##   ExperimentList does not have names, assign names</span></span></code></pre>
<p>Non-matching names may also be present in the
<code>ExperimentList</code> elements and the “assay” column of the
<code>sampleMap</code>. If names only differ by case and are identical
and unique, names will be standardized to lower case and replaced.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">objlist3</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">toupper</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">objlist</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">objlist3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "AFFY"        "METHYL 450K" "MIRNA"       "CNV GISTIC"</span></span></code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">dfmap</span><span class="op">[</span>, <span class="st">"assay"</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] Affy        Methyl 450k Mirna       CNV gistic </span></span>
<span><span class="co">## Levels: Affy Methyl 450k Mirna CNV gistic</span></span></code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/prepMultiAssay.html">prepMultiAssay</a></span><span class="op">(</span><span class="va">objlist3</span>, <span class="va">patient.data</span>, <span class="va">dfmap</span><span class="op">)</span><span class="op">$</span><span class="va">experiments</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Names in the ExperimentList do not match sampleMap assay</span></span>
<span><span class="co">## standardizing will be attempted...</span></span></code></pre>
<pre><code><span><span class="co">##  - names set to lowercase</span></span></code></pre>
<pre><code><span><span class="co">## ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] methyl 450k: matrix with 2 rows and 5 columns</span></span>
<span><span class="co">##  [3] mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] cnv gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span></code></pre>
<p>When <code>colnames</code> in the <code>ExperimentList</code> cannot
be matched back to the primary data (<code>colData</code>), these will
be dropped and added to the drops element.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleMap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">simpleMultiAssay2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">doubleExp</span>, <span class="va">colnames</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $`methyl 2k`</span></span>
<span><span class="co">## [1] "Jack"  "Jill"  "Bob"   "Bobby"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`methyl 3k`</span></span>
<span><span class="co">## [1] "Jack" "Jane" "Bob"</span></span></code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exampleMap</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 5 rows and 3 columns</span></span>
<span><span class="co">##       assay     primary     colname</span></span>
<span><span class="co">##    &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## 1 methyl 2k        Jack        Jack</span></span>
<span><span class="co">## 2 methyl 2k        Jill        Jill</span></span>
<span><span class="co">## 3 methyl 2k         Bob         Bob</span></span>
<span><span class="co">## 4 methyl 3k        Jack        Jack</span></span>
<span><span class="co">## 5 methyl 3k         Bob         Bob</span></span></code></pre>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/prepMultiAssay.html">prepMultiAssay</a></span><span class="op">(</span><span class="va">doubleExp</span>, <span class="va">patient.data</span>, <span class="va">exampleMap</span><span class="op">)</span><span class="op">$</span><span class="va">metadata</span><span class="op">$</span><span class="va">drops</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Not all colnames in the ExperimentList are found in the </span></span>
<span><span class="co">## sampleMap, dropping samples from ExperimentList...</span></span></code></pre>
<pre><code><span><span class="co">## $`methyl 2k`</span></span>
<span><span class="co">## [1] "Bobby"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`methyl 3k`</span></span>
<span><span class="co">## [1] "Jane"</span></span></code></pre>
<pre><code><span><span class="co">## $`columns.methyl 2k`</span></span>
<span><span class="co">## [1] "Bobby"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`columns.methyl 3k`</span></span>
<span><span class="co">## [1] "Jane"</span></span></code></pre>
<p>A similar operation is performed for checking “primary”
<code>sampleMap</code> names and <code>colData</code> rownames. In this
example, we add a row corresponding to “Joe” that does not have a match
in the experimental data.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exMap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">dfmap</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/DataFrame-class.html" class="external-link">DataFrame</a></span><span class="op">(</span>assay <span class="op">=</span> <span class="st">"New methyl"</span>, primary <span class="op">=</span> <span class="st">"Joe"</span>,</span>
<span>        colname <span class="op">=</span> <span class="st">"Joe"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="../reference/prepMultiAssay.html">prepMultiAssay</a></span><span class="op">(</span><span class="va">objlist</span>, <span class="va">patient.data</span>, <span class="va">exMap</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in prepMultiAssay(objlist, patient.data, exMap): </span></span>
<span><span class="co">## Lengths of names in the ExperimentList and sampleMap</span></span>
<span><span class="co">##  are not equal</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Not all names in the primary column of the sampleMap</span></span>
<span><span class="co">##  could be matched to the colData rownames; see $drops</span></span></code></pre>
<pre><code><span><span class="co">## DataFrame with 1 row and 3 columns</span></span>
<span><span class="co">##        assay     primary     colname</span></span>
<span><span class="co">##     &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## 1 New methyl         Joe         Joe</span></span></code></pre>
<p>To create a <code>MultiAssayExperiment</code> from the results of the
<code>prepMultiAssay</code> function, take each corresponding element
from the resulting list and enter them as arguments to the
<code>MultiAssayExperiment</code> constructor function.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prepped</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepMultiAssay.html">prepMultiAssay</a></span><span class="op">(</span><span class="va">objlist</span>, <span class="va">patient.data</span>, <span class="va">exMap</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in prepMultiAssay(objlist, patient.data, exMap): </span></span>
<span><span class="co">## Lengths of names in the ExperimentList and sampleMap</span></span>
<span><span class="co">##  are not equal</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Not all names in the primary column of the sampleMap</span></span>
<span><span class="co">##  could be matched to the colData rownames; see $drops</span></span></code></pre>
<pre><code><span><span class="co">## DataFrame with 1 row and 3 columns</span></span>
<span><span class="co">##        assay     primary     colname</span></span>
<span><span class="co">##     &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## 1 New methyl         Joe         Joe</span></span></code></pre>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">preppedMulti</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span><span class="va">prepped</span><span class="op">$</span><span class="va">experiments</span>, <span class="va">prepped</span><span class="op">$</span><span class="va">colData</span>,</span>
<span>    <span class="va">prepped</span><span class="op">$</span><span class="va">sampleMap</span>, <span class="va">prepped</span><span class="op">$</span><span class="va">metadata</span><span class="op">)</span></span>
<span><span class="va">preppedMulti</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>Alternatively, use the <code>do.call</code> function to easily create
a <code>MultiAssayExperiment</code> from the output of
<code>prepMultiAssay</code> function:</p>
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">MultiAssayExperiment</span>, <span class="va">prepped</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="helper-functions-to-create-bioconductor-classes-from-raw-data">Helper functions to create <code>Bioconductor</code> classes from
raw data<a class="anchor" aria-label="anchor" href="#helper-functions-to-create-bioconductor-classes-from-raw-data"></a>
</h3>
<p>Recent updates to the <code>GenomicRanges</code> and
<code>SummarizedExperiment</code> packages allow the user to create
standard <em>Bioconductor</em> classes from raw data. Raw data read in
as either <code>data.frame</code> or <code>DataFrame</code> can be
converted to <code>GRangesList</code> or
<code>SummarizedExperiment</code> classes depending on the type of
data.</p>
<p>The function to create a <code>GRangesList</code> from a
<code>data.frame</code>, called
<code>makeGRangesListFromDataFrame</code> can be found in the
<code>GenomicRanges</code> package.
<code>makeSummarizedExperimentFromDataFrame</code> is available in the
<code>SummarizedExperiment</code> package. It is also possible to create
a <code>RangedSummarizedExperiment</code> class object from raw data
when ranged data is available.</p>
<p>A simple example can be obtained from the function documentation in
<code>GenomicRanges</code>:</p>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">grlls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="va">nrows</span><span class="op">)</span>, start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">11</span>, <span class="fl">15</span><span class="op">)</span>,</span>
<span>    end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">12</span>, <span class="fl">16</span><span class="op">)</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span>, <span class="st">"*"</span>, <span class="st">"*"</span><span class="op">)</span>,</span>
<span>    score <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">5</span><span class="op">)</span>, specimen <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"a"</span>, <span class="st">"b"</span>, <span class="st">"b"</span>, <span class="st">"c"</span><span class="op">)</span>,</span>
<span>    gene_symbols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"GENE"</span>, <span class="va">letters</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">nrows</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">grldf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">grlls</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">GRL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesListFromDataFrame.html" class="external-link">makeGRangesListFromDataFrame</a></span><span class="op">(</span><span class="va">grldf</span>, split.field <span class="op">=</span> <span class="st">"specimen"</span>,</span>
<span>    names.field <span class="op">=</span> <span class="st">"gene_symbols"</span><span class="op">)</span></span></code></pre></div>
<p>This can then be converted to a <code>RaggedExperiment</code> object
for a rectangular representation that will conform more easily to the
<code>MultiAssayExperiment</code> API requirements.</p>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/RaggedExperiment/man/RaggedExperiment-class.html" class="external-link">RaggedExperiment</a></span><span class="op">(</span><span class="va">GRL</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RaggedExperiment </span></span>
<span><span class="co">## dim: 5 3 </span></span>
<span><span class="co">## assays(0):</span></span>
<span><span class="co">## rownames(5): GENEa GENEb GENEc GENEd GENEe</span></span>
<span><span class="co">## colnames(3): a b c</span></span>
<span><span class="co">## colData names(0):</span></span></code></pre>
<p><em>Note</em>. See the <code>RaggedExperiment</code> vignette for
more details.</p>
<p>In the <code>SummarizedExperiment</code> package:</p>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"chr2"</span>, <span class="va">nrows</span><span class="op">)</span>, start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">11</span>, <span class="fl">15</span><span class="op">)</span>,</span>
<span>    end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">12</span>, <span class="fl">16</span><span class="op">)</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span>, <span class="st">"*"</span>, <span class="st">"*"</span><span class="op">)</span>,</span>
<span>    expr0 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">7</span><span class="op">)</span>, expr1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">8</span>, <span class="fl">12</span><span class="op">)</span>, expr2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">12</span>, <span class="fl">16</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sedf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">sels</span>,</span>
<span>    row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"GENE"</span>, <span class="va">letters</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">nrows</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>,</span>
<span>    stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">sedf</span></span></code></pre></div>
<pre><code><span><span class="co">##        chr start end strand expr0 expr1 expr2</span></span>
<span><span class="co">## GENEe chr2    11  12      +     3     8    12</span></span>
<span><span class="co">## GENEd chr2    12  13      -     4     9    13</span></span>
<span><span class="co">## GENEc chr2    13  14      +     5    10    14</span></span>
<span><span class="co">## GENEb chr2    14  15      *     6    11    15</span></span>
<span><span class="co">## GENEa chr2    15  16      *     7    12    16</span></span></code></pre>
<div class="sourceCode" id="cb98"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/makeSummarizedExperimentFromDataFrame.html" class="external-link">makeSummarizedExperimentFromDataFrame</a></span><span class="op">(</span><span class="va">sedf</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## class: RangedSummarizedExperiment </span></span>
<span><span class="co">## dim: 5 3 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(5): GENEe GENEd GENEc GENEb GENEa</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(3): expr0 expr1 expr2</span></span>
<span><span class="co">## colData names(0):</span></span></code></pre>
<p style="text-align: right;">
<a href="#header">back to top</a>
</p>
</div>
</div>
<div class="section level2">
<h2 id="integrated-subsetting-across-experiments">Integrated subsetting across experiments<a class="anchor" aria-label="anchor" href="#integrated-subsetting-across-experiments"></a>
</h2>
<p><code>MultiAssayExperiment</code> allows subsetting by rows, columns,
and assays, rownames, and colnames, across all experiments
simultaneously while guaranteeing continued matching of samples.</p>
<p>Subsetting can be done most compactly by the square bracket method,
or more verbosely and potentially more flexibly by the
<code>subsetBy*()</code> methods.</p>
<div class="section level3">
<h3 id="subsetting-by-square-bracket">Subsetting by square bracket <code>[</code><a class="anchor" aria-label="anchor" href="#subsetting-by-square-bracket"></a>
</h3>
<p>The three positions within the bracket operator indicate rows,
columns, and assays, respectively (pseudocode):</p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span><span class="va">rows</span>, <span class="va">columns</span>, <span class="va">assays</span><span class="op">]</span></span></code></pre></div>
<p>For example, to select the gene “ENST00000355076”:</p>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span><span class="st">"ENST00000355076"</span>, , <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 1 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 1 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 0 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 0 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>The above operation works across all types of assays, whether
ID-based (e.g. <code>matrix</code>, <code>ExpressionSet</code>,
<code>SummarizedExperiment</code>) or range-based
(e.g. <code>RangedSummarizedExperiment</code>). Note that when using the
bracket method <code>[</code>, the drop argument is <em>TRUE</em> by
default.</p>
<p>You can subset by rows, columns, and assays in a single bracket
operation, and they will be performed in that order (rows, then columns,
then assays). The following selects the <code>ENST00000355076</code>
gene across all samples, then the first two samples of each assay, and
finally the Affy and Methyl 450k assays:</p>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span><span class="st">"ENST00000355076"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Affy"</span>, <span class="st">"Methyl 450k"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 3 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 1 rows and 2 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 1 rows and 3 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="subsetting-by-character-integer-and-logical">Subsetting by character, integer, and logical<a class="anchor" aria-label="anchor" href="#subsetting-by-character-integer-and-logical"></a>
</h3>
<p>By columns - character, integer, and logical are all allowed, for
example:</p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, <span class="st">"Jack"</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 1 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 2 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 1 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 1 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, <span class="fl">1</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 1 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 2 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 1 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 1 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 1 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 2 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 1 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 1 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>By assay - character, integer, and logical are allowed:</p>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="st">"Mirna"</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 13 sampleMap rows not in names(experiments)</span></span>
<span><span class="co">##   removing 1 colData rownames not in sampleMap 'primary'</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 1 listed</span></span>
<span><span class="co">##  experiment with a user-defined name and respective class.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 1:</span></span>
<span><span class="co">##  [1] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="fl">3</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 13 sampleMap rows not in names(experiments)</span></span>
<span><span class="co">##   removing 1 colData rownames not in sampleMap 'primary'</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 1 listed</span></span>
<span><span class="co">##  experiment with a user-defined name and respective class.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 1:</span></span>
<span><span class="co">##  [1] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 13 sampleMap rows not in names(experiments)</span></span>
<span><span class="co">##   removing 1 colData rownames not in sampleMap 'primary'</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 1 listed</span></span>
<span><span class="co">##  experiment with a user-defined name and respective class.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 1:</span></span>
<span><span class="co">##  [1] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="the-drop-argument">the “drop” argument<a class="anchor" aria-label="anchor" href="#the-drop-argument"></a>
</h3>
<p>Specify <code>drop=FALSE</code> to keep assays with zero rows or zero
columns, e.g.:</p>
<div class="sourceCode" id="cb125"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span><span class="st">"ENST00000355076"</span>, , , drop<span class="op">=</span><span class="cn">FALSE</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 1 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 1 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 0 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 0 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>Using the default <code>drop=TRUE</code>, assays with no rows or no
columns are removed:</p>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span><span class="st">"ENST00000355076"</span>, , , drop<span class="op">=</span><span class="cn">TRUE</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span>
<span></span>
<span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 7 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 1 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 1 rows and 5 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="more-on-subsetting-by-columns">More on subsetting by columns<a class="anchor" aria-label="anchor" href="#more-on-subsetting-by-columns"></a>
</h3>
<p>Experimental samples are stored in the rows of <code>colData</code>
but the columns of elements of <code>ExperimentList</code>, so when we
refer to subsetting by columns, we are referring to columns of the
experimental assays. Subsetting by samples / columns will be more
obvious after recalling the <code>colData</code>:</p>
<div class="sourceCode" id="cb131"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 4 rows and 2 columns</span></span>
<span><span class="co">##                 sex       age</span></span>
<span><span class="co">##         &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## Jack              M        38</span></span>
<span><span class="co">## Jill              F        39</span></span>
<span><span class="co">## Bob               M        40</span></span>
<span><span class="co">## Barbara           F        41</span></span></code></pre>
<p>Subsetting by samples identifies the selected samples in rows of the
colData DataFrame, then selects all columns of the
<code>ExperimentList</code> corresponding to these rows. Here we use an
integer to keep the first two rows of colData, and all experimental
assays associated to those two primary samples:</p>
<div class="sourceCode" id="cb133"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 2 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 3 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 1 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 2 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>Note that the above operation keeps different numbers of columns /
samples from each assay, reflecting the reality that some samples may
not have been assayed in all experiments, and may have replicates in
some.</p>
<p>Columns can be subset using a logical vector. Here the dollar sign
operator (<code>$</code>) accesses one of the columns in
<code>colData</code>.</p>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">malesMultiAssay</span> <span class="op">&lt;-</span> <span class="va">myMultiAssay</span><span class="op">[</span>, <span class="va">myMultiAssay</span><span class="op">$</span><span class="va">sex</span> <span class="op">==</span> <span class="st">"M"</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">malesMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 2 rows and 2 columns</span></span>
<span><span class="co">##              sex       age</span></span>
<span><span class="co">##      &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## Jack           M        38</span></span>
<span><span class="co">## Bob            M        40</span></span></code></pre>
<p>Finally, for special use cases you can exert detailed control of row
or column subsetting, by using a <code>list</code> or
<code>CharacterList</code> to subset. The following creates a
<code>CharacterList</code> of the column names of each assay:</p>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">allsamples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span>
<span><span class="va">allsamples</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 4</span></span>
<span><span class="co">## [["Affy"]] array1 array2 array3 array4</span></span>
<span><span class="co">## [["Methyl 450k"]] methyl1 methyl2 methyl3 methyl4 methyl5</span></span>
<span><span class="co">## [["Mirna"]] micro1 micro2 micro3</span></span>
<span><span class="co">## [["CNV gistic"]] mysnparray1 mysnparray2 mysnparray3 mysnparray4</span></span></code></pre>
<p>Now let’s get rid of three Methyl 450k arrays, those in positions 3,
4, and 5:</p>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">allsamples</span><span class="op">[[</span><span class="st">"Methyl 450k"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">allsamples</span><span class="op">[[</span><span class="st">"Methyl 450k"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="op">-</span><span class="fl">3</span><span class="op">:</span><span class="op">-</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="va">myMultiAssay</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">allsamples</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 3 sampleMap rows with 'colname' not in colnames of experiments</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 2 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb142"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/subsetBy.html">subsetByColumn</a></span><span class="op">(</span><span class="va">myMultiAssay</span>,  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">allsamples</span><span class="op">)</span><span class="op">)</span>  <span class="co">#equivalent</span></span></code></pre></div>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 3 sampleMap rows with 'colname' not in colnames of experiments</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 2 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="subsetting-assays">Subsetting assays<a class="anchor" aria-label="anchor" href="#subsetting-assays"></a>
</h3>
<p>You can select certain assays / experiments using subset, by
providing a character, logical, or integer vector. An example using
character:</p>
<div class="sourceCode" id="cb145"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Affy"</span>, <span class="st">"CNV gistic"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 8 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>You can subset assays also using logical or integer vectors:</p>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">is.cnv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"CNV"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">experiments</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">is.cnv</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] FALSE FALSE FALSE  TRUE</span></span></code></pre>
<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="va">is.cnv</span><span class="op">]</span>  <span class="co">#logical subsetting</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 12 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 1 listed</span></span>
<span><span class="co">##  experiment with a user-defined name and respective class.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 1:</span></span>
<span><span class="co">##  [1] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">is.cnv</span><span class="op">)</span><span class="op">]</span> <span class="co">#integer subsetting</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 12 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 1 listed</span></span>
<span><span class="co">##  experiment with a user-defined name and respective class.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 1:</span></span>
<span><span class="co">##  [1] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="subsetting-rows-features-by-ids-integers-or-logicals">Subsetting rows (features) by IDs, integers, or logicals<a class="anchor" aria-label="anchor" href="#subsetting-rows-features-by-ids-integers-or-logicals"></a>
</h3>
<p>Rows of the assays correspond to assay features or measurements, such
as genes. Regardless of whether the assay is ID-based (e.g.,
<code>matrix</code>, <code>ExpressionSet</code>) or range-based (e.g.,
<code>RangedSummarizedExperiment</code>), they can be subset using any
of the following:</p>
<ul>
<li><p>a <strong>character vector</strong> of IDs that will be matched
to rownames in each assay</p></li>
<li><p>an <strong>integer vector</strong> that will select rows of this
position from each assay. This probably doesn’t make sense unless every
<code>ExperimentList</code> element represents the same measurements in
the same order and will generate an error if any of the integer elements
exceeds the number of rows in any <code>ExperimentList</code> element.
The most likely use of integer subsetting would be as a
<code>head</code> function, for example to look at the first 6 rows of
each assay.</p></li>
<li><p>a <strong>logical vector</strong> that will be passed directly to
the row subsetting operation for each assay.</p></li>
<li><p>a <strong>list</strong> or <strong>List</strong> with element
names matching those in the <code>ExperimentList</code>. Each element of
the subsetting list will be passed on exactly to subset rows of the
corresponding element of the <code>ExperimentList</code>.</p></li>
</ul>
<p>Any <code>list</code> or <code>List</code> input allows for selective
subsetting. The subsetting is applied only to the matching element names
in the <code>ExperimentList</code>. For example, to only take the first
two rows of the microRNA dataset, we use a named <code>list</code> to
indicate what element we want to subset along with the
<code>drop = FALSE</code> argument.</p>
<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Mirna <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span>, , <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 2 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb161"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## equivalently</span></span>
<span><span class="fu"><a href="../reference/subsetBy.html">subsetByRow</a></span><span class="op">(</span><span class="va">myMultiAssay</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Mirna <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 2 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>Again, these operations always return a
<code>MultiAssayExperiment</code> class, unless <code>drop=TRUE</code>
is passed to the <code>[</code> backet subset, with any
<code>ExperimentList</code> element not containing the feature having
zero rows.</p>
<p>For example, return a MultiAssayExperiment where <code>Affy</code>
and <code>Methyl 450k</code> contain only “ENST0000035076”” row, and
“Mirna” and “CNV gistic” have zero rows (<code>drop</code> argument is
set to <code>FALSE</code> by default in <code>subsetBy*</code>):</p>
<div class="sourceCode" id="cb163"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">featSub0</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetBy.html">subsetByRow</a></span><span class="op">(</span><span class="va">myMultiAssay</span>, <span class="st">"ENST00000355076"</span><span class="op">)</span></span>
<span><span class="va">featSub1</span> <span class="op">&lt;-</span> <span class="va">myMultiAssay</span><span class="op">[</span><span class="st">"ENST00000355076"</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span> <span class="co">#equivalent</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">featSub0</span>, <span class="va">featSub1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb165"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">featSub1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "MultiAssayExperiment"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "MultiAssayExperiment"</span></span></code></pre>
<div class="sourceCode" id="cb167"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">experiments</a></span><span class="op">(</span><span class="va">featSub1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ExperimentList"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "MultiAssayExperiment"</span></span></code></pre>
<div class="sourceCode" id="cb169"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">experiments</a></span><span class="op">(</span><span class="va">featSub1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 1 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 1 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 0 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 0 rows and 4 columns</span></span></code></pre>
<p>In the following, <code>Affy</code> <code>SummarizedExperiment</code>
keeps both rows but with their order reversed, and
<code>Methyl 450k</code> keeps only its second row.</p>
<div class="sourceCode" id="cb171"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">featSubsetted</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/subsetBy.html">subsetByRow</a></span><span class="op">(</span><span class="va">myMultiAssay</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENST00000355076"</span>, <span class="st">"ENST00000294241"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">myMultiAssay</span>, <span class="fl">1L</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 array1 array2 array3 array4</span></span>
<span><span class="co">## ENST00000294241    101    103    105    107</span></span>
<span><span class="co">## ENST00000355076    102    104    106    108</span></span></code></pre>
<div class="sourceCode" id="cb173"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">featSubsetted</span>, <span class="fl">1L</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 array1 array2 array3 array4</span></span>
<span><span class="co">## ENST00000355076    102    104    106    108</span></span>
<span><span class="co">## ENST00000294241    101    103    105    107</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="subsetting-rows-features-by-genomicranges">Subsetting rows (features) by <code>GenomicRanges</code><a class="anchor" aria-label="anchor" href="#subsetting-rows-features-by-genomicranges"></a>
</h3>
<p>For <code>MultiAssayExperiment</code> objects containing range-based
objects (currently <code>RangedSummarizedExperiment</code>), these can
be subset using a <code>GRanges</code> object, for example:</p>
<div class="sourceCode" id="cb175"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr1"</span>, <span class="st">"chr2"</span><span class="op">)</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"-"</span>, <span class="st">"+"</span>, <span class="st">"+"</span><span class="op">)</span>,</span>
<span>              ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">230602</span>, <span class="fl">443625</span>, <span class="fl">934533</span><span class="op">)</span>,</span>
<span>                               end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">330701</span>, <span class="fl">443724</span>, <span class="fl">934632</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Now do the subsetting. The function doing the work here is
<code><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">IRanges::subsetByOverlaps</a></code> - see its arguments for flexible
types of subsetting by range. The first three arguments here are for
<code>subset</code>, the rest passed on to
<code><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">IRanges::subsetByOverlaps</a></code> through “…”:</p>
<div class="sourceCode" id="cb176"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">subsetted</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subsetBy.html">subsetByRow</a></span><span class="op">(</span><span class="va">myMultiAssay</span>, <span class="va">gr</span>, maxgap <span class="op">=</span> <span class="fl">2L</span>, type <span class="op">=</span> <span class="st">"within"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">experiments</a></span><span class="op">(</span><span class="va">subsetted</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 0 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 0 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 0 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 0 rows and 4 columns</span></span></code></pre>
<div class="sourceCode" id="cb178"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">subsetted</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 0 ranges and 1 metadata column:</span></span>
<span><span class="co">##    seqnames    ranges strand |  feature_id</span></span>
<span><span class="co">##       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Square bracket subsetting can still be used here, but passing on
arguments to <code><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">IRanges::subsetByOverlaps</a></code> through “…” is
simpler using <code><a href="../reference/subsetBy.html">subsetByRow()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="subsetting-is-endomorphic">Subsetting is endomorphic<a class="anchor" aria-label="anchor" href="#subsetting-is-endomorphic"></a>
</h3>
<p><code>subsetByRow</code>, <code>subsetByColumn</code>,
<code>subsetByAssay</code>, and square bracket subsetting are all
“endomorphic” operations, in that they always return another
<code>MultiAssayExperiment</code> object.</p>
</div>
<div class="section level3">
<h3 id="double-bracket-subsetting-to-select-experiments">Double-bracket subsetting to select experiments<a class="anchor" aria-label="anchor" href="#double-bracket-subsetting-to-select-experiments"></a>
</h3>
<p>A double-bracket subset operation refers to an experiment, and will
return the object contained within an <code>ExperimentList</code>
element. It is <strong>not</strong> endomorphic. For example, the first
<code>ExperimentList</code> element is called “Affy” and contains a
<code>SummarizedExperiment</code>:</p>
<div class="sourceCode" id="cb180"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Affy"        "Methyl 450k" "Mirna"       "CNV gistic"</span></span></code></pre>
<div class="sourceCode" id="cb182"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 2 4 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(2): ENST00000294241 ENST00000355076</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(4): array1 array2 array3 array4</span></span>
<span><span class="co">## colData names(1): slope53</span></span></code></pre>
<div class="sourceCode" id="cb184"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[[</span><span class="st">"Affy"</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 2 4 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(2): ENST00000294241 ENST00000355076</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(4): array1 array2 array3 array4</span></span>
<span><span class="co">## colData names(1): slope53</span></span></code></pre>
<p style="text-align: right;">
<a href="#header">back to top</a>
</p>
</div>
</div>
<div class="section level2">
<h2 id="helpers-for-data-clean-up-and-management">Helpers for data clean-up and management<a class="anchor" aria-label="anchor" href="#helpers-for-data-clean-up-and-management"></a>
</h2>
<div class="section level3">
<h3 id="complete-cases">
<code>complete.cases</code><a class="anchor" aria-label="anchor" href="#complete-cases"></a>
</h3>
<p>The <code>complete.cases</code> function returns a logical vector of
<code>colData</code> rows identifying which primary units have data for
all experiments. Recall that <code>myMultiAssay</code> provides data for
four individuals:</p>
<div class="sourceCode" id="cb186"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 4 rows and 2 columns</span></span>
<span><span class="co">##                 sex       age</span></span>
<span><span class="co">##         &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## Jack              M        38</span></span>
<span><span class="co">## Jill              F        39</span></span>
<span><span class="co">## Bob               M        40</span></span>
<span><span class="co">## Barbara           F        41</span></span></code></pre>
<p>Of these, only Jack has data for all 5 experiments:</p>
<div class="sourceCode" id="cb188"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  TRUE FALSE  TRUE  TRUE</span></span></code></pre>
<p>But all four have complete cases for Affy and Methyl 450k:</p>
<div class="sourceCode" id="cb190"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 7 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## [1] TRUE TRUE TRUE TRUE</span></span></code></pre>
<p>This output can be used to select individuals with complete data:</p>
<div class="sourceCode" id="cb194"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myMultiAssay</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html" class="external-link">complete.cases</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 3 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 4 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 3 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="replicated">
<code>replicated</code><a class="anchor" aria-label="anchor" href="#replicated"></a>
</h3>
<p>The <code>replicated</code> function identifies <code>primary</code>
column values or biological units that have multiple observations per
<code>assay</code>. It returns a <code>list</code> of
<code>LogicalList</code>s that indicate what biological units have one
or more replicate measurements. This output is used for merging
replicates by default.</p>
<div class="sourceCode" id="cb196"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">replicated</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $Affy</span></span>
<span><span class="co">## LogicalList of length 4</span></span>
<span><span class="co">## [["Barbara"]] FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Bob"]] FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Jack"]] FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Jill"]] FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`Methyl 450k`</span></span>
<span><span class="co">## LogicalList of length 4</span></span>
<span><span class="co">## [["Barbara"]] FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Bob"]] FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Jack"]] TRUE TRUE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Jill"]] FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $Mirna</span></span>
<span><span class="co">## LogicalList of length 3</span></span>
<span><span class="co">## [["Barbara"]] FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Bob"]] FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Jack"]] FALSE FALSE FALSE</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $`CNV gistic`</span></span>
<span><span class="co">## LogicalList of length 4</span></span>
<span><span class="co">## [["Barbara"]] FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Bob"]] FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Jack"]] FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">## [["Jill"]] FALSE FALSE FALSE FALSE</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="intersectrows">
<code>intersectRows</code><a class="anchor" aria-label="anchor" href="#intersectrows"></a>
</h3>
<p>The <code>intersectRows</code> function takes all common rownames
across all experiments and returns a <code>MultiAssayExperiment</code>
with those rows.</p>
<div class="sourceCode" id="cb198"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">ensmblMatches</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">intersectRows</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 7 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 1 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 1 rows and 5 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb202"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ensmblMatches</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 2</span></span>
<span><span class="co">## [["Affy"]] ENST00000355076</span></span>
<span><span class="co">## [["Methyl 450k"]] ENST00000355076</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="intersectcolumns">
<code>intersectColumns</code><a class="anchor" aria-label="anchor" href="#intersectcolumns"></a>
</h3>
<p>A call to <code>intersectColumns</code> returns another
<code>MultiAssayExperiment</code> where the columns of each element of
the <code>ExperimentList</code> correspond exactly to the rows of
<code>colData</code>. In many cases, this operation returns a 1-to-1
correspondence of samples to patients for each experiment assay unless
replicates are present in the data.</p>
<div class="sourceCode" id="cb204"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">intersectColumns</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 3 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 4 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 3 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="mergereplicates">
<code>mergeReplicates</code><a class="anchor" aria-label="anchor" href="#mergereplicates"></a>
</h3>
<p>The <code>mergeReplicates</code> function allows the user to specify
a function (default: <code>mean</code>) for combining replicate columns
in each assay element. This can be combined with
<code>intersectColumns</code> to create a
<code>MultiAssayExperiment</code> object with one measurement in each
experiment per biological unit.</p>
<div class="sourceCode" id="cb206"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">mergeReplicates</a></span><span class="op">(</span><span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">intersectColumns</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 1 sampleMap rows with 'colname' not in colnames of experiments</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 4 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 4:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 3 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 3 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 3 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="combine-c">combine <code>c</code><a class="anchor" aria-label="anchor" href="#combine-c"></a>
</h3>
<p>The combine <code>c</code> function allows the user to append an
experiment to the list of experiments already present in
<code>MultiAssayExperiment</code>. In the case that additional
observations on the same set of samples were performed, the
<code>c</code> function can conveniently be referenced to an existing
assay that contains the same ordering of sample measurements.</p>
<p>The <code>mapFrom</code> argument indicates what experiment has the
exact same organization of samples that will be introduced by the new
experiment dataset. If the number of columns in the new experiment do
not match those in the reference experiment, an error will be
thrown.</p>
<p>Here we introduce a toy dataset created on the fly:</p>
<div class="sourceCode" id="cb209"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">myMultiAssay</span>, ExpScores <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>, ncol <span class="op">=</span> <span class="fl">4</span>,</span>
<span>dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSMBL0001"</span>, <span class="st">"ENSMBL0002"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pt"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>mapFrom <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Assuming column order in the data provided </span></span>
<span><span class="co">##  matches the order in 'mapFrom' experiment(s) colnames</span></span></code></pre>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 5 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 5:</span></span>
<span><span class="co">##  [1] Affy: SummarizedExperiment with 2 rows and 4 columns</span></span>
<span><span class="co">##  [2] Methyl 450k: matrix with 2 rows and 5 columns</span></span>
<span><span class="co">##  [3] Mirna: matrix with 4 rows and 3 columns</span></span>
<span><span class="co">##  [4] CNV gistic: RangedSummarizedExperiment with 5 rows and 4 columns</span></span>
<span><span class="co">##  [5] ExpScores: matrix with 2 rows and 4 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p><em>Note</em>: Alternatively, a <code>sampleMap</code> for the
additional dataset can be provided.</p>
<p style="text-align: right;">
<a href="#header">back to top</a>
</p>
</div>
</div>
<div class="section level2">
<h2 id="extractor-functions">Extractor functions<a class="anchor" aria-label="anchor" href="#extractor-functions"></a>
</h2>
<p>Extractor functions convert a <code>MultiAssayExperiment</code> into
other forms that are convenient for analyzing. These would normally be
called after any desired subsetting has been performed.</p>
<div class="section level3">
<h3 id="getwithcoldata">
<code>getWithColData</code><a class="anchor" aria-label="anchor" href="#getwithcoldata"></a>
</h3>
<p>Provides a single assay along with any associated ‘colData’ columns
while keeping the assay class constant.</p>
<div class="sourceCode" id="cb212"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">affex</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">getWithColData</a></span><span class="op">(</span><span class="va">myMultiAssay</span>, <span class="fl">1L</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 2 4 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(2): ENST00000294241 ENST00000355076</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(4): array1 array2 array3 array4</span></span>
<span><span class="co">## colData names(3): slope53 sex age</span></span></code></pre>
<div class="sourceCode" id="cb215"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">affex</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 4 rows and 3 columns</span></span>
<span><span class="co">##           slope53         sex       age</span></span>
<span><span class="co">##         &lt;numeric&gt; &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## array1 -0.4780168           M        38</span></span>
<span><span class="co">## array2 -0.0997548           F        39</span></span>
<span><span class="co">## array3  1.2418972           F        41</span></span>
<span><span class="co">## array4 -0.7127357           M        40</span></span></code></pre>
<div class="sourceCode" id="cb217"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">affex</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "SummarizedExperiment"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "SummarizedExperiment"</span></span></code></pre>
<p>It will error when the target data class does not support a
<code>colData</code> replacement method, meaning that it typically works
with <code>SummarizedExperiment</code> and <code>RaggedExperiment</code>
assays and their extensions.</p>
</div>
<div class="section level3">
<h3 id="longformat-wideformat">
<code>longFormat</code> &amp; <code>wideFormat</code><a class="anchor" aria-label="anchor" href="#longformat-wideformat"></a>
</h3>
<p>Produces <em>long</em> (default) or <em>wide</em>
<code>DataFrame</code> objects. The following produces a long
<code>DataFrame</code> (the default) for the first two assays:</p>
<div class="sourceCode" id="cb219"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">longFormat</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 7 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## DataFrame with 18 rows and 5 columns</span></span>
<span><span class="co">##           assay     primary         rowname     colname     value</span></span>
<span><span class="co">##     &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">## 1          Affy        Jack ENST00000294241      array1       101</span></span>
<span><span class="co">## 2          Affy        Jack ENST00000355076      array1       102</span></span>
<span><span class="co">## 3          Affy        Jill ENST00000294241      array2       103</span></span>
<span><span class="co">## 4          Affy        Jill ENST00000355076      array2       104</span></span>
<span><span class="co">## 5          Affy     Barbara ENST00000294241      array3       105</span></span>
<span><span class="co">## ...         ...         ...             ...         ...       ...</span></span>
<span><span class="co">## 14  Methyl 450k        Jill ENST00000383706     methyl3         6</span></span>
<span><span class="co">## 15  Methyl 450k     Barbara ENST00000355076     methyl4         7</span></span>
<span><span class="co">## 16  Methyl 450k     Barbara ENST00000383706     methyl4         8</span></span>
<span><span class="co">## 17  Methyl 450k         Bob ENST00000355076     methyl5         9</span></span>
<span><span class="co">## 18  Methyl 450k         Bob ENST00000383706     methyl5        10</span></span></code></pre>
<p>This is especially useful for performing regression against patient
or sample data from <code>colData</code> using the
<code>pDataCols</code> argument:</p>
<div class="sourceCode" id="cb223"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">longFormat</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">[</span>, , <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, colDataCols<span class="op">=</span><span class="st">"age"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: 'experiments' dropped; see 'metadata'</span></span></code></pre>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 7 sampleMap rows not in names(experiments)</span></span></code></pre>
<pre><code><span><span class="co">## DataFrame with 18 rows and 6 columns</span></span>
<span><span class="co">##           assay     primary         rowname     colname     value       age</span></span>
<span><span class="co">##     &lt;character&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt; &lt;integer&gt; &lt;integer&gt;</span></span>
<span><span class="co">## 1          Affy        Jack ENST00000294241      array1       101        38</span></span>
<span><span class="co">## 2          Affy        Jack ENST00000355076      array1       102        38</span></span>
<span><span class="co">## 3          Affy        Jill ENST00000294241      array2       103        39</span></span>
<span><span class="co">## 4          Affy        Jill ENST00000355076      array2       104        39</span></span>
<span><span class="co">## 5          Affy     Barbara ENST00000294241      array3       105        41</span></span>
<span><span class="co">## ...         ...         ...             ...         ...       ...       ...</span></span>
<span><span class="co">## 14  Methyl 450k        Jill ENST00000383706     methyl3         6        39</span></span>
<span><span class="co">## 15  Methyl 450k     Barbara ENST00000355076     methyl4         7        41</span></span>
<span><span class="co">## 16  Methyl 450k     Barbara ENST00000383706     methyl4         8        41</span></span>
<span><span class="co">## 17  Methyl 450k         Bob ENST00000355076     methyl5         9        40</span></span>
<span><span class="co">## 18  Methyl 450k         Bob ENST00000383706     methyl5        10        40</span></span></code></pre>
<p>The “wide” format is useful for calculating correlations or
performing regression against different genomic features. Wide format is
in general not possible with replicate measurements, so we demonstrate
on the cleaned <code>MultiAssayExperiment</code> for the first 5
columns:</p>
<div class="sourceCode" id="cb227"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">maemerge</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">mergeReplicates</a></span><span class="op">(</span><span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">intersectColumns</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## harmonizing input:</span></span>
<span><span class="co">##   removing 1 sampleMap rows with 'colname' not in colnames of experiments</span></span></code></pre>
<div class="sourceCode" id="cb229"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment-helpers.html">wideFormat</a></span><span class="op">(</span><span class="va">maemerge</span>, colDataCols<span class="op">=</span><span class="st">"sex"</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 3 rows and 5 columns</span></span>
<span><span class="co">##       primary         sex Affy_ENST00000294241 Affy_ENST00000355076</span></span>
<span><span class="co">##   &lt;character&gt; &lt;character&gt;            &lt;integer&gt;            &lt;integer&gt;</span></span>
<span><span class="co">## 1        Jack           M                  101                  102</span></span>
<span><span class="co">## 2         Bob           M                  107                  108</span></span>
<span><span class="co">## 3     Barbara           F                  105                  106</span></span>
<span><span class="co">##   Methyl.450k_ENST00000355076</span></span>
<span><span class="co">##                     &lt;numeric&gt;</span></span>
<span><span class="co">## 1                           2</span></span>
<span><span class="co">## 2                           9</span></span>
<span><span class="co">## 3                           7</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="assay-assays">
<code>assay</code> / <code>assays</code><a class="anchor" aria-label="anchor" href="#assay-assays"></a>
</h3>
<p>The <code>assay</code> (singular) function takes a particular
experiment and returns a matrix. By default, it will return the
<em>first</em> experiment as a matrix.</p>
<div class="sourceCode" id="cb231"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 array1 array2 array3 array4</span></span>
<span><span class="co">## ENST00000294241    101    103    105    107</span></span>
<span><span class="co">## ENST00000355076    102    104    106    108</span></span></code></pre>
<p>The <code>assays</code> (plural) function returns a
<code>SimpleList</code> of data matrices from the
<code>ExperimentList</code>:</p>
<div class="sourceCode" id="cb233"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of length 4</span></span>
<span><span class="co">## names(4): Affy Methyl 450k Mirna CNV gistic</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="the-cancer-genome-atlas-and-multiassayexperiment">The Cancer Genome Atlas and MultiAssayExperiment<a class="anchor" aria-label="anchor" href="#the-cancer-genome-atlas-and-multiassayexperiment"></a>
</h2>
<p>Our most recent efforts include the release of the experiment data
package, <code>curatedTCGAData</code>. This package will allow users to
selectively download cancer datasets from The Cancer Genome Atlas (TCGA)
and represent the data as <code>MultiAssayExperiment</code> objects.
Please see the package vignette for more details.</p>
<div class="sourceCode" id="cb235"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"curatedTCGAData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="dimension-names-rownames-and-colnames">Dimension names: <code>rownames</code> and
<code>colnames</code><a class="anchor" aria-label="anchor" href="#dimension-names-rownames-and-colnames"></a>
</h2>
<p><code>rownames</code> and <code>colnames</code> return a
<code>CharacterList</code> of row names and column names across all the
assays. A <code>CharacterList</code> is an efficient alternative to
<code>list</code> used when each element contains a character vector. It
also provides a nice show method:</p>
<div class="sourceCode" id="cb236"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 4</span></span>
<span><span class="co">## [["Affy"]] ENST00000294241 ENST00000355076</span></span>
<span><span class="co">## [["Methyl 450k"]] ENST00000355076 ENST00000383706</span></span>
<span><span class="co">## [["Mirna"]] hsa-miR-21 hsa-miR-191 hsa-miR-148a hsa-miR148b</span></span>
<span><span class="co">## [["CNV gistic"]] a b c d e</span></span></code></pre>
<div class="sourceCode" id="cb238"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">myMultiAssay</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CharacterList of length 4</span></span>
<span><span class="co">## [["Affy"]] array1 array2 array3 array4</span></span>
<span><span class="co">## [["Methyl 450k"]] methyl1 methyl2 methyl3 methyl4 methyl5</span></span>
<span><span class="co">## [["Mirna"]] micro1 micro2 micro3</span></span>
<span><span class="co">## [["CNV gistic"]] mysnparray1 mysnparray2 mysnparray3 mysnparray4</span></span></code></pre>
<p style="text-align: right;">
<a href="#header">back to top</a>
</p>
</div>
<div class="section level2">
<h2 id="requirements-for-support-of-additional-data-classes">Requirements for support of additional data classes<a class="anchor" aria-label="anchor" href="#requirements-for-support-of-additional-data-classes"></a>
</h2>
<p>Any data classes in the <code>ExperimentList</code> object must
support the following methods:</p>
<ul>
<li><code>dimnames</code></li>
<li><code>[</code></li>
<li><code><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim()</a></code></li>
</ul>
<p>Here is what happens if one of the methods doesn’t:</p>
<div class="sourceCode" id="cb240"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">objlist2</span> <span class="op">&lt;-</span> <span class="va">objlist</span></span>
<span><span class="va">objlist2</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">objlist2</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/conditions.html" class="external-link">tryCatch</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span><span class="va">objlist2</span>, <span class="va">patient.data</span>, <span class="va">dfmap</span><span class="op">)</span>,</span>
<span>    error <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">e</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/conditions.html" class="external-link">conditionMessage</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "invalid class \"ExperimentList\" object: \n    'integer' class is not supported, use a rectangular class"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="application-programming-interface-api">Application Programming Interface (API)<a class="anchor" aria-label="anchor" href="#application-programming-interface-api"></a>
</h2>
<p>For more information on the formal API of
<code>MultiAssayExperiment</code>, please see the <a href="https://github.com/waldronlab/MultiAssayExperiment/wiki/MultiAssayExperiment-API" class="external-link">API
wiki</a> document on GitHub. An API package is available for download on
GitHub via <code>install("waldronlab/MultiAssayShiny")</code>. It
provides visual exploration of available methods in
<code>MultiAssayExperiment</code>.</p>
<p style="text-align: right;">
<a href="#header">back to top</a>
</p>
</div>
<div class="section level2">
<h2 id="methods-for-multiassayexperiment">Methods for MultiAssayExperiment<a class="anchor" aria-label="anchor" href="#methods-for-multiassayexperiment"></a>
</h2>
<p>The following methods are defined for
<code>MultiAssayExperiment</code>:</p>
<div class="sourceCode" id="cb242"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/methods.html" class="external-link">methods</a></span><span class="op">(</span>class<span class="op">=</span><span class="st">"MultiAssayExperiment"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] [               [[              [[&lt;-            [&lt;-            </span></span>
<span><span class="co">##  [5] $               $&lt;-             anyReplicated   assay          </span></span>
<span><span class="co">##  [9] assays          c               coerce          colData        </span></span>
<span><span class="co">## [13] colData&lt;-       colnames&lt;-      complete.cases  dimnames       </span></span>
<span><span class="co">## [17] experiments     experiments&lt;-   exportClass     hasRowRanges   </span></span>
<span><span class="co">## [21] isEmpty         length          mergeReplicates metadata       </span></span>
<span><span class="co">## [25] metadata&lt;-      names           names&lt;-         replicated     </span></span>
<span><span class="co">## [29] replicates      sampleMap       sampleMap&lt;-     show           </span></span>
<span><span class="co">## [33] showReplicated  splitAssays     subsetByAssay   subsetByColData</span></span>
<span><span class="co">## [37] subsetByColumn  subsetByRow     updateObject   </span></span>
<span><span class="co">## see '?methods' for accessing help and source code</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="citing-multiassayexperiment">Citing MultiAssayExperiment<a class="anchor" aria-label="anchor" href="#citing-multiassayexperiment"></a>
</h2>
<p>We are excited to announce the official citation for
MultiAssayExperiment in <em>Cancer Research</em>.</p>
<div class="sourceCode" id="cb244"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"MultiAssayExperiment"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## To cite MultiAssayExperiment in publications use:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Marcel Ramos et al. Software For The Integration Of Multiomics</span></span>
<span><span class="co">##   Experiments In Bioconductor. Cancer Research, 2017 November 1;</span></span>
<span><span class="co">##   77(21); e39-42. DOI: 10.1158/0008-5472.CAN-17-0344</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   @Article{,</span></span>
<span><span class="co">##     title = {Software For The Integration Of Multi-Omics Experiments In Bioconductor},</span></span>
<span><span class="co">##     author = {Marcel Ramos and Lucas Schiffer and Angela Re and Rimsha Azhar and Azfar Basunia and Carmen Rodriguez Cabrera and Tiffany Chan and Philip Chapman and Sean Davis and David Gomez-Cabrero and Aedin C. Culhane and Benjamin Haibe-Kains and Kasper Hansen and Hanish Kodali and Marie Stephie Louis and Arvind Singh Mer and Markus Reister and Martin Morgan and Vincent Carey and Levi Waldron},</span></span>
<span><span class="co">##     journal = {Cancer Research},</span></span>
<span><span class="co">##     year = {2017},</span></span>
<span><span class="co">##     volume = {77(21); e39-42},</span></span>
<span><span class="co">##   }</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="sessioninfo">sessionInfo()<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb246"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] RaggedExperiment_1.24.0     MultiAssayExperiment_1.26.0</span></span>
<span><span class="co">##  [3] SummarizedExperiment_1.30.0 Biobase_2.60.0             </span></span>
<span><span class="co">##  [5] GenomicRanges_1.52.0        GenomeInfoDb_1.36.0        </span></span>
<span><span class="co">##  [7] IRanges_2.34.0              S4Vectors_0.38.0           </span></span>
<span><span class="co">##  [9] BiocGenerics_0.46.0         MatrixGenerics_1.12.0      </span></span>
<span><span class="co">## [11] matrixStats_0.63.0          BiocStyle_2.28.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] utf8_1.2.3              generics_0.1.3          tidyr_1.3.0            </span></span>
<span><span class="co">##  [4] sass_0.4.5              bitops_1.0-7            stringi_1.7.12         </span></span>
<span><span class="co">##  [7] lattice_0.21-8          digest_0.6.31           magrittr_2.0.3         </span></span>
<span><span class="co">## [10] evaluate_0.20           grid_4.3.0              bookdown_0.33          </span></span>
<span><span class="co">## [13] fastmap_1.1.1           rprojroot_2.0.3         jsonlite_1.8.4         </span></span>
<span><span class="co">## [16] Matrix_1.5-4            BiocManager_1.30.20     fansi_1.0.4            </span></span>
<span><span class="co">## [19] purrr_1.0.1             textshaping_0.3.6       jquerylib_0.1.4        </span></span>
<span><span class="co">## [22] cli_3.6.1               rlang_1.1.0             XVector_0.40.0         </span></span>
<span><span class="co">## [25] withr_2.5.0             cachem_1.0.7            DelayedArray_0.25.0    </span></span>
<span><span class="co">## [28] yaml_2.3.7              BiocBaseUtils_1.2.0     tools_4.3.0            </span></span>
<span><span class="co">## [31] memoise_2.0.1           dplyr_1.1.2             GenomeInfoDbData_1.2.10</span></span>
<span><span class="co">## [34] vctrs_0.6.2             R6_2.5.1                lifecycle_1.0.3        </span></span>
<span><span class="co">## [37] zlibbioc_1.46.0         stringr_1.5.0           fs_1.6.2               </span></span>
<span><span class="co">## [40] ragg_1.2.5              pkgconfig_2.0.3         desc_1.4.2             </span></span>
<span><span class="co">## [43] pillar_1.9.0            pkgdown_2.0.7           bslib_0.4.2            </span></span>
<span><span class="co">## [46] glue_1.6.2              systemfonts_1.0.4       tidyselect_1.2.0       </span></span>
<span><span class="co">## [49] tibble_3.2.1            xfun_0.39               highr_0.10             </span></span>
<span><span class="co">## [52] knitr_1.42              htmltools_0.5.5         rmarkdown_2.21         </span></span>
<span><span class="co">## [55] compiler_4.3.0          RCurl_1.98-1.12</span></span></code></pre>
<p style="text-align: right;">
<a href="#header">back to top</a>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
