<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using DelayedMatrix with MultiAssayExperiment â€¢ MultiAssayExperiment</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using DelayedMatrix with MultiAssayExperiment">
<meta property="og:description" content="MultiAssayExperiment">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiAssayExperiment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/MultiAssayExperiment.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a>
    </li>
    <li>
      <a href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Release Notes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/MultiAssayExperiment">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="UsingHDF5Array_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using DelayedMatrix with MultiAssayExperiment</h1>
                        <h4 class="author">MultiAssay Special Interest Group</h4>
            
            <h4 class="date">November 04, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/master/vignettes/UsingHDF5Array.Rmd"><code>vignettes/UsingHDF5Array.Rmd</code></a></small>
      <div class="hidden name"><code>UsingHDF5Array.Rmd</code></div>

    </div>

    
    
<div id="integrating-an-hdf5-backend-for-multiassayexperiment" class="section level1">
<h1 class="hasAnchor">
<a href="#integrating-an-hdf5-backend-for-multiassayexperiment" class="anchor"></a>Integrating an HDF5 backend for MultiAssayExperiment</h1>
<div id="dependencies" class="section level2">
<h2 class="hasAnchor">
<a href="#dependencies" class="anchor"></a>Dependencies</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/">MultiAssayExperiment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/HDF5Array">HDF5Array</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="hdf5array-and-delayedarray-constructor" class="section level2">
<h2 class="hasAnchor">
<a href="#hdf5array-and-delayedarray-constructor" class="anchor"></a>HDF5Array and DelayedArray Constructor</h2>
<p>The <code>HDF5Array</code> package provides an on-disk representation of large datasets without the need to load them into memory. Convenient lazy evaluation operations allow the user to manipulate such large data files based on metadata. The <code>DelayedMatrix</code> class in the <code>DelayedArray</code> package provides a way to connect to a large matrix that is stored on disk.</p>
<p>First, we create a small matrix for constructing the <code>DelayedMatrix</code> class.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">smallMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">10e5</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></code></pre></div>
<p>We add rownames and column names to the matrix object for compatibility with the <code>MultiAssayExperiment</code> representation.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">rownames</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"GENE"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">nrow</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html">colnames</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"SampleID"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html">ncol</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Here we use the <code>DelayedArray</code> constructor function to create a <code>DelayedMatrix</code> object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">smallMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DelayedArray/man/DelayedArray-class.html">DelayedArray</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "DelayedMatrix"
## attr(,"package")
## [1] "DelayedArray"</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># show method</span>
<span class="va">smallMatrix</span></code></pre></div>
<pre><code>## &lt;50000 x 20&gt; matrix of class DelayedMatrix and type "double":
##             SampleID1   SampleID2   SampleID3 ... SampleID19 SampleID20
##     GENE1  0.13738341  0.77323598  3.28452674   .  0.6186079  0.3020248
##     GENE2 -0.87860056 -1.01130551 -0.02062963   . -1.2493849  0.9099543
##     GENE3  0.46202231  0.10810840 -0.03158766   . -0.2727587 -0.8245425
##     GENE4  2.18794749 -0.22359621  0.20310116   . -1.1107794  0.6975085
##     GENE5  0.09081977 -1.63047285  1.01488152   .  0.2661830 -0.4129355
##       ...           .           .           .   .          .          .
## GENE49996  -0.3392032  -0.9772606  -0.2087721   . -0.9750892  0.4190485
## GENE49997  -0.6325679  -0.5500055  -0.5641244   .  0.2709898 -0.1287319
## GENE49998  -0.3304345   0.3519324   0.8782357   .  0.2653960 -2.2548703
## GENE49999  -0.4608000   0.7515588  -0.4034890   .  1.6098185 -0.8777167
## GENE50000   0.7400389  -0.3136337   0.4103519   .  0.2609142  0.4276916</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 50000    20</code></pre>
</div>
<div id="writing-to-a-file-with-dimnames" class="section level2">
<h2 class="hasAnchor">
<a href="#writing-to-a-file-with-dimnames" class="anchor"></a>Writing to a file with dimnames</h2>
<p>Finally, the <code>rhdf5</code> package stores <code>dimnames</code> in a standard location.</p>
<p>In order to make use of this functionality, we would use <code>writeHDF5Array</code> with the <code>with.dimnames</code> argument:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">testh5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".h5"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/HDF5Array/man/writeHDF5Array.html">writeHDF5Array</a></span><span class="op">(</span><span class="va">smallMatrix</span>, filepath <span class="op">=</span> <span class="va">testh5</span>, name <span class="op">=</span> <span class="st">"smallMatrix"</span>,
    with.dimnames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## &lt;50000 x 20&gt; matrix of class HDF5Matrix and type "double":
##             SampleID1   SampleID2   SampleID3 ... SampleID19 SampleID20
##     GENE1  0.13738341  0.77323598  3.28452674   .  0.6186079  0.3020248
##     GENE2 -0.87860056 -1.01130551 -0.02062963   . -1.2493849  0.9099543
##     GENE3  0.46202231  0.10810840 -0.03158766   . -0.2727587 -0.8245425
##     GENE4  2.18794749 -0.22359621  0.20310116   . -1.1107794  0.6975085
##     GENE5  0.09081977 -1.63047285  1.01488152   .  0.2661830 -0.4129355
##       ...           .           .           .   .          .          .
## GENE49996  -0.3392032  -0.9772606  -0.2087721   . -0.9750892  0.4190485
## GENE49997  -0.6325679  -0.5500055  -0.5641244   .  0.2709898 -0.1287319
## GENE49998  -0.3304345   0.3519324   0.8782357   .  0.2653960 -2.2548703
## GENE49999  -0.4608000   0.7515588  -0.4034890   .  1.6098185 -0.8777167
## GENE50000   0.7400389  -0.3136337   0.4103519   .  0.2609142  0.4276916</code></pre>
<p>To see the file structure we use <code>h5ls</code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/HDF5Array/man/h5ls.html">h5ls</a></span><span class="op">(</span><span class="va">testh5</span><span class="op">)</span></code></pre></div>
<pre><code>##                    group                  name       otype dclass        dim
## 0                      / .smallMatrix_dimnames   H5I_GROUP                  
## 1 /.smallMatrix_dimnames                     1 H5I_DATASET STRING      50000
## 2 /.smallMatrix_dimnames                     2 H5I_DATASET STRING         20
## 3                      /           smallMatrix H5I_DATASET  FLOAT 50000 x 20</code></pre>
</div>
<div id="importing-hdf5-files" class="section level2">
<h2 class="hasAnchor">
<a href="#importing-hdf5-files" class="anchor"></a>Importing HDF5 files</h2>
<p>Note that a large matrix from an HDF5 file can also be loaded using the <code>HDF5ArraySeed</code> and <code>DelayedArray</code> functions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hdf5Data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HDF5Array/man/HDF5ArraySeed-class.html">HDF5ArraySeed</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">testh5</span>, name <span class="op">=</span> <span class="st">"smallMatrix"</span><span class="op">)</span>
<span class="va">newDelayedMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DelayedArray/man/DelayedArray-class.html">DelayedArray</a></span><span class="op">(</span><span class="va">hdf5Data</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">newDelayedMatrix</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "HDF5Matrix"
## attr(,"package")
## [1] "HDF5Array"</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">newDelayedMatrix</span></code></pre></div>
<pre><code>## &lt;50000 x 20&gt; matrix of class HDF5Matrix and type "double":
##             SampleID1   SampleID2   SampleID3 ... SampleID19 SampleID20
##     GENE1  0.13738341  0.77323598  3.28452674   .  0.6186079  0.3020248
##     GENE2 -0.87860056 -1.01130551 -0.02062963   . -1.2493849  0.9099543
##     GENE3  0.46202231  0.10810840 -0.03158766   . -0.2727587 -0.8245425
##     GENE4  2.18794749 -0.22359621  0.20310116   . -1.1107794  0.6975085
##     GENE5  0.09081977 -1.63047285  1.01488152   .  0.2661830 -0.4129355
##       ...           .           .           .   .          .          .
## GENE49996  -0.3392032  -0.9772606  -0.2087721   . -0.9750892  0.4190485
## GENE49997  -0.6325679  -0.5500055  -0.5641244   .  0.2709898 -0.1287319
## GENE49998  -0.3304345   0.3519324   0.8782357   .  0.2653960 -2.2548703
## GENE49999  -0.4608000   0.7515588  -0.4034890   .  1.6098185 -0.8777167
## GENE50000   0.7400389  -0.3136337   0.4103519   .  0.2609142  0.4276916</code></pre>
</div>
<div id="using-a-delayedmatrix-with-multiassayexperiment" class="section level2">
<h2 class="hasAnchor">
<a href="#using-a-delayedmatrix-with-multiassayexperiment" class="anchor"></a>Using a <code>DelayedMatrix</code> with <code>MultiAssayExperiment</code>
</h2>
<p>A <code>DelayedMatrix</code> alone conforms to the <code>MultiAssayExperiment</code> API requirements. Shown below, the <code>DelayedMatrix</code> can be put into a named <code>list</code> and passed into the <code>MultiAssayExperiment</code> constructor function.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">HDF5MAE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span>experiments <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>smallMatrix <span class="op">=</span> <span class="va">smallMatrix</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">HDF5MAE</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 20 rows and 3 columns
##           assay     primary     colname
##        &lt;factor&gt; &lt;character&gt; &lt;character&gt;
## 1   smallMatrix   SampleID1   SampleID1
## 2   smallMatrix   SampleID2   SampleID2
## 3   smallMatrix   SampleID3   SampleID3
## 4   smallMatrix   SampleID4   SampleID4
## 5   smallMatrix   SampleID5   SampleID5
## ...         ...         ...         ...
## 16  smallMatrix  SampleID16  SampleID16
## 17  smallMatrix  SampleID17  SampleID17
## 18  smallMatrix  SampleID18  SampleID18
## 19  smallMatrix  SampleID19  SampleID19
## 20  smallMatrix  SampleID20  SampleID20</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">HDF5MAE</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 20 rows and 0 columns</code></pre>
<div id="summarizedexperiment-with-delayedmatrix-backend" class="section level3">
<h3 class="hasAnchor">
<a href="#summarizedexperiment-with-delayedmatrix-backend" class="anchor"></a><code>SummarizedExperiment</code> with <code>DelayedMatrix</code> backend</h3>
<p>A more information rich <code>DelayedMatrix</code> can be created when used in conjunction with the <code>SummarizedExperiment</code> class and it can even include <code>rowRanges</code>. The flexibility of the <code>MultiAssayExperiment</code> API supports classes with minimal requirements. Additionally, this <code>SummarizedExperiment</code> with the <code>DelayedMatrix</code> backend can be part of a bigger <code>MultiAssayExperiment</code> object. Below is a minimal example of how this would work:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">HDF5SE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html">SummarizedExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="va">smallMatrix</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">HDF5SE</span><span class="op">)</span></code></pre></div>
<pre><code>## &lt;50000 x 20&gt; matrix of class DelayedMatrix and type "double":
##             SampleID1   SampleID2   SampleID3 ... SampleID19 SampleID20
##     GENE1  0.13738341  0.77323598  3.28452674   .  0.6186079  0.3020248
##     GENE2 -0.87860056 -1.01130551 -0.02062963   . -1.2493849  0.9099543
##     GENE3  0.46202231  0.10810840 -0.03158766   . -0.2727587 -0.8245425
##     GENE4  2.18794749 -0.22359621  0.20310116   . -1.1107794  0.6975085
##     GENE5  0.09081977 -1.63047285  1.01488152   .  0.2661830 -0.4129355
##       ...           .           .           .   .          .          .
## GENE49996  -0.3392032  -0.9772606  -0.2087721   . -0.9750892  0.4190485
## GENE49997  -0.6325679  -0.5500055  -0.5641244   .  0.2709898 -0.1287319
## GENE49998  -0.3304345   0.3519324   0.8782357   .  0.2653960 -2.2548703
## GENE49999  -0.4608000   0.7515588  -0.4034890   .  1.6098185 -0.8777167
## GENE50000   0.7400389  -0.3136337   0.4103519   .  0.2609142  0.4276916</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>HDF5SE <span class="op">=</span> <span class="va">HDF5SE</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## A MultiAssayExperiment object of 1 listed
##  experiment with a user-defined name and respective class.
##  Containing an ExperimentList class object of length 1:
##  [1] HDF5SE: SummarizedExperiment with 50000 rows and 20 columns
## Functionality:
##  experiments() - obtain the ExperimentList instance
##  colData() - the primary/phenotype DataFrame
##  sampleMap() - the sample coordination DataFrame
##  `$`, `[`, `[[` - extract colData columns, subset, or experiment
##  *Format() - convert into a long or wide DataFrame
##  assays() - convert ExperimentList to a SimpleList of matrices
##  exportClass() - save data to flat files</code></pre>
<p>Additional scenarios are currently in development where an <code>HDF5Matrix</code> is hosted remotely. Many opportunities exist when considering on-disk and off-disk representations of data with <code>MultiAssayExperiment</code>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
