<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using DelayedMatrix with MultiAssayExperiment â€¢ MultiAssayExperiment</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using DelayedMatrix with MultiAssayExperiment">
<meta property="og:description" content="MultiAssayExperiment">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MultiAssayExperiment</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.28.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Quick Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/QuickStartMultiAssay.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/MultiAssayExperiment.html">Getting Started</a>
    </li>
    <li>
      <a href="../articles/UsingHDF5Array.html">Using DelayedMatrix</a>
    </li>
    <li>
      <a href="../articles/extras/MAE-curatedMetagenomicData.html">MultiAssayExperiment in Practice</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../news/index.html">Release Notes</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/MultiAssayExperiment" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using DelayedMatrix with
MultiAssayExperiment</h1>
                        <h4 data-toc-skip class="author">MultiAssay
Special Interest Group</h4>
            
            <h4 data-toc-skip class="date">January 24, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/MultiAssayExperiment/blob/HEAD/vignettes/UsingHDF5Array.Rmd" class="external-link"><code>vignettes/UsingHDF5Array.Rmd</code></a></small>
      <div class="hidden name"><code>UsingHDF5Array.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="integrating-an-hdf5-backend-for-multiassayexperiment">Integrating an HDF5 backend for MultiAssayExperiment<a class="anchor" aria-label="anchor" href="#integrating-an-hdf5-backend-for-multiassayexperiment"></a>
</h2>
<div class="section level3">
<h3 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/HDF5Array" class="external-link">HDF5Array</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="hdf5array-and-delayedarray-constructor">HDF5Array and DelayedArray Constructor<a class="anchor" aria-label="anchor" href="#hdf5array-and-delayedarray-constructor"></a>
</h3>
<p>The <code>HDF5Array</code> package provides an on-disk representation
of large datasets without the need to load them into memory. Convenient
lazy evaluation operations allow the user to manipulate such large data
files based on metadata. The <code>DelayedMatrix</code> class in the
<code>DelayedArray</code> package provides a way to connect to a large
matrix that is stored on disk.</p>
<p>First, we create a small matrix for constructing the
<code>DelayedMatrix</code> class.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">smallMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">10e5</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p>We add rownames and column names to the matrix object for
compatibility with the <code>MultiAssayExperiment</code>
representation.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"GENE"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"SampleID"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Here we use the <code>DelayedArray</code> constructor function to
create a <code>DelayedMatrix</code> object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">smallMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DelayedArray/man/DelayedArray-class.html" class="external-link">DelayedArray</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "DelayedMatrix"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "DelayedArray"</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show method</span></span>
<span><span class="va">smallMatrix</span></span></code></pre></div>
<pre><code><span><span class="co">## &lt;50000 x 20&gt; DelayedMatrix object of type "double":</span></span>
<span><span class="co">##             SampleID1   SampleID2   SampleID3 ...  SampleID19  SampleID20</span></span>
<span><span class="co">##     GENE1  -0.3875569  -0.9245979  -0.4860061   .  0.88458768 -1.61563992</span></span>
<span><span class="co">##     GENE2  -0.2644658   0.3048011  -0.9763176   .  0.62873303  0.11844208</span></span>
<span><span class="co">##     GENE3  -1.0037401  -0.6172604   0.2084938   . -1.03015889 -1.61369308</span></span>
<span><span class="co">##     GENE4  -1.1815745  -1.5556907   0.6032805   . -1.30802616  0.07655268</span></span>
<span><span class="co">##     GENE5  -2.9104018   2.4867479   3.2304527   . -0.39682861  2.10575923</span></span>
<span><span class="co">##       ...           .           .           .   .           .           .</span></span>
<span><span class="co">## GENE49996  0.08974065  0.65075160 -0.07995008   .   0.5827023   0.1526415</span></span>
<span><span class="co">## GENE49997  0.70751975  1.37734494 -2.00313995   .   0.6547790   1.4318531</span></span>
<span><span class="co">## GENE49998  0.74370046 -0.04946058 -0.31213299   .  -2.4141129  -1.5732835</span></span>
<span><span class="co">## GENE49999 -2.61420539 -1.09089104  0.91170686   .   1.0285334  -0.3192227</span></span>
<span><span class="co">## GENE50000  0.42807965 -0.71061527  0.39352464   .   0.2416611  -0.7500389</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">smallMatrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 50000    20</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="writing-to-a-file-with-dimnames">Writing to a file with dimnames<a class="anchor" aria-label="anchor" href="#writing-to-a-file-with-dimnames"></a>
</h3>
<p>Finally, the <code>rhdf5</code> package stores <code>dimnames</code>
in a standard location.</p>
<p>In order to make use of this functionality, we would use
<code>writeHDF5Array</code> with the <code>with.dimnames</code>
argument:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">testh5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>fileext <span class="op">=</span> <span class="st">".h5"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/HDF5Array/man/writeHDF5Array.html" class="external-link">writeHDF5Array</a></span><span class="op">(</span><span class="va">smallMatrix</span>, filepath <span class="op">=</span> <span class="va">testh5</span>, name <span class="op">=</span> <span class="st">"smallMatrix"</span>,</span>
<span>    with.dimnames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## &lt;50000 x 20&gt; HDF5Matrix object of type "double":</span></span>
<span><span class="co">##             SampleID1   SampleID2   SampleID3 ...  SampleID19  SampleID20</span></span>
<span><span class="co">##     GENE1  -0.3875569  -0.9245979  -0.4860061   .  0.88458768 -1.61563992</span></span>
<span><span class="co">##     GENE2  -0.2644658   0.3048011  -0.9763176   .  0.62873303  0.11844208</span></span>
<span><span class="co">##     GENE3  -1.0037401  -0.6172604   0.2084938   . -1.03015889 -1.61369308</span></span>
<span><span class="co">##     GENE4  -1.1815745  -1.5556907   0.6032805   . -1.30802616  0.07655268</span></span>
<span><span class="co">##     GENE5  -2.9104018   2.4867479   3.2304527   . -0.39682861  2.10575923</span></span>
<span><span class="co">##       ...           .           .           .   .           .           .</span></span>
<span><span class="co">## GENE49996  0.08974065  0.65075160 -0.07995008   .   0.5827023   0.1526415</span></span>
<span><span class="co">## GENE49997  0.70751975  1.37734494 -2.00313995   .   0.6547790   1.4318531</span></span>
<span><span class="co">## GENE49998  0.74370046 -0.04946058 -0.31213299   .  -2.4141129  -1.5732835</span></span>
<span><span class="co">## GENE49999 -2.61420539 -1.09089104  0.91170686   .   1.0285334  -0.3192227</span></span>
<span><span class="co">## GENE50000  0.42807965 -0.71061527  0.39352464   .   0.2416611  -0.7500389</span></span></code></pre>
<p>To see the file structure we use <code>h5ls</code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">h5ls</span><span class="op">(</span><span class="va">testh5</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    group                  name       otype dclass        dim</span></span>
<span><span class="co">## 0                      / .smallMatrix_dimnames   H5I_GROUP                  </span></span>
<span><span class="co">## 1 /.smallMatrix_dimnames                     1 H5I_DATASET STRING      50000</span></span>
<span><span class="co">## 2 /.smallMatrix_dimnames                     2 H5I_DATASET STRING         20</span></span>
<span><span class="co">## 3                      /           smallMatrix H5I_DATASET  FLOAT 50000 x 20</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="importing-hdf5-files">Importing HDF5 files<a class="anchor" aria-label="anchor" href="#importing-hdf5-files"></a>
</h3>
<p>Note that a large matrix from an HDF5 file can also be loaded using
the <code>HDF5ArraySeed</code> and <code>DelayedArray</code>
functions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hdf5Data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HDF5Array/man/HDF5ArraySeed-class.html" class="external-link">HDF5ArraySeed</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">testh5</span>, name <span class="op">=</span> <span class="st">"smallMatrix"</span><span class="op">)</span></span>
<span><span class="va">newDelayedMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DelayedArray/man/DelayedArray-class.html" class="external-link">DelayedArray</a></span><span class="op">(</span><span class="va">hdf5Data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">newDelayedMatrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "HDF5Matrix"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "HDF5Array"</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">newDelayedMatrix</span></span></code></pre></div>
<pre><code><span><span class="co">## &lt;50000 x 20&gt; HDF5Matrix object of type "double":</span></span>
<span><span class="co">##             SampleID1   SampleID2   SampleID3 ...  SampleID19  SampleID20</span></span>
<span><span class="co">##     GENE1  -0.3875569  -0.9245979  -0.4860061   .  0.88458768 -1.61563992</span></span>
<span><span class="co">##     GENE2  -0.2644658   0.3048011  -0.9763176   .  0.62873303  0.11844208</span></span>
<span><span class="co">##     GENE3  -1.0037401  -0.6172604   0.2084938   . -1.03015889 -1.61369308</span></span>
<span><span class="co">##     GENE4  -1.1815745  -1.5556907   0.6032805   . -1.30802616  0.07655268</span></span>
<span><span class="co">##     GENE5  -2.9104018   2.4867479   3.2304527   . -0.39682861  2.10575923</span></span>
<span><span class="co">##       ...           .           .           .   .           .           .</span></span>
<span><span class="co">## GENE49996  0.08974065  0.65075160 -0.07995008   .   0.5827023   0.1526415</span></span>
<span><span class="co">## GENE49997  0.70751975  1.37734494 -2.00313995   .   0.6547790   1.4318531</span></span>
<span><span class="co">## GENE49998  0.74370046 -0.04946058 -0.31213299   .  -2.4141129  -1.5732835</span></span>
<span><span class="co">## GENE49999 -2.61420539 -1.09089104  0.91170686   .   1.0285334  -0.3192227</span></span>
<span><span class="co">## GENE50000  0.42807965 -0.71061527  0.39352464   .   0.2416611  -0.7500389</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="using-a-delayedmatrix-with-multiassayexperiment">Using a <code>DelayedMatrix</code> with
<code>MultiAssayExperiment</code><a class="anchor" aria-label="anchor" href="#using-a-delayedmatrix-with-multiassayexperiment"></a>
</h3>
<p>A <code>DelayedMatrix</code> alone conforms to the
<code>MultiAssayExperiment</code> API requirements. Shown below, the
<code>DelayedMatrix</code> can be put into a named <code>list</code> and
passed into the <code>MultiAssayExperiment</code> constructor
function.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HDF5MAE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span>experiments <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>smallMatrix <span class="op">=</span> <span class="va">smallMatrix</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/MultiAssayExperiment-methods.html">sampleMap</a></span><span class="op">(</span><span class="va">HDF5MAE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 20 rows and 3 columns</span></span>
<span><span class="co">##           assay     primary     colname</span></span>
<span><span class="co">##        &lt;factor&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## 1   smallMatrix   SampleID1   SampleID1</span></span>
<span><span class="co">## 2   smallMatrix   SampleID2   SampleID2</span></span>
<span><span class="co">## 3   smallMatrix   SampleID3   SampleID3</span></span>
<span><span class="co">## 4   smallMatrix   SampleID4   SampleID4</span></span>
<span><span class="co">## 5   smallMatrix   SampleID5   SampleID5</span></span>
<span><span class="co">## ...         ...         ...         ...</span></span>
<span><span class="co">## 16  smallMatrix  SampleID16  SampleID16</span></span>
<span><span class="co">## 17  smallMatrix  SampleID17  SampleID17</span></span>
<span><span class="co">## 18  smallMatrix  SampleID18  SampleID18</span></span>
<span><span class="co">## 19  smallMatrix  SampleID19  SampleID19</span></span>
<span><span class="co">## 20  smallMatrix  SampleID20  SampleID20</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">HDF5MAE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 20 rows and 0 columns</span></span></code></pre>
<div class="section level4">
<h4 id="summarizedexperiment-with-delayedmatrix-backend">
<code>SummarizedExperiment</code> with <code>DelayedMatrix</code>
backend<a class="anchor" aria-label="anchor" href="#summarizedexperiment-with-delayedmatrix-backend"></a>
</h4>
<p>A more information rich <code>DelayedMatrix</code> can be created
when used in conjunction with the <code>SummarizedExperiment</code>
class and it can even include <code>rowRanges</code>. The flexibility of
the <code>MultiAssayExperiment</code> API supports classes with minimal
requirements. Additionally, this <code>SummarizedExperiment</code> with
the <code>DelayedMatrix</code> backend can be part of a bigger
<code>MultiAssayExperiment</code> object. Below is a minimal example of
how this would work:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HDF5SE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span>assays <span class="op">=</span> <span class="va">smallMatrix</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">HDF5SE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## &lt;50000 x 20&gt; DelayedMatrix object of type "double":</span></span>
<span><span class="co">##             SampleID1   SampleID2   SampleID3 ...  SampleID19  SampleID20</span></span>
<span><span class="co">##     GENE1  -0.3875569  -0.9245979  -0.4860061   .  0.88458768 -1.61563992</span></span>
<span><span class="co">##     GENE2  -0.2644658   0.3048011  -0.9763176   .  0.62873303  0.11844208</span></span>
<span><span class="co">##     GENE3  -1.0037401  -0.6172604   0.2084938   . -1.03015889 -1.61369308</span></span>
<span><span class="co">##     GENE4  -1.1815745  -1.5556907   0.6032805   . -1.30802616  0.07655268</span></span>
<span><span class="co">##     GENE5  -2.9104018   2.4867479   3.2304527   . -0.39682861  2.10575923</span></span>
<span><span class="co">##       ...           .           .           .   .           .           .</span></span>
<span><span class="co">## GENE49996  0.08974065  0.65075160 -0.07995008   .   0.5827023   0.1526415</span></span>
<span><span class="co">## GENE49997  0.70751975  1.37734494 -2.00313995   .   0.6547790   1.4318531</span></span>
<span><span class="co">## GENE49998  0.74370046 -0.04946058 -0.31213299   .  -2.4141129  -1.5732835</span></span>
<span><span class="co">## GENE49999 -2.61420539 -1.09089104  0.91170686   .   1.0285334  -0.3192227</span></span>
<span><span class="co">## GENE50000  0.42807965 -0.71061527  0.39352464   .   0.2416611  -0.7500389</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiAssayExperiment.html">MultiAssayExperiment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>HDF5SE <span class="op">=</span> <span class="va">HDF5SE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 1 listed</span></span>
<span><span class="co">##  experiment with a user-defined name and respective class.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 1:</span></span>
<span><span class="co">##  [1] HDF5SE: SummarizedExperiment with 50000 rows and 20 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<p>Additional scenarios are currently in development where an
<code>HDF5Matrix</code> is hosted remotely. Many opportunities exist
when considering on-disk and off-disk representations of data with
<code>MultiAssayExperiment</code>.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] HDF5Array_1.30.0            rhdf5_2.46.1               </span></span>
<span><span class="co">##  [3] DelayedArray_0.28.0         SparseArray_1.2.3          </span></span>
<span><span class="co">##  [5] S4Arrays_1.2.0              abind_1.4-5                </span></span>
<span><span class="co">##  [7] Matrix_1.6-5                MultiAssayExperiment_1.28.0</span></span>
<span><span class="co">##  [9] SummarizedExperiment_1.32.0 Biobase_2.62.0             </span></span>
<span><span class="co">## [11] GenomicRanges_1.54.1        GenomeInfoDb_1.38.5        </span></span>
<span><span class="co">## [13] IRanges_2.36.0              S4Vectors_0.40.2           </span></span>
<span><span class="co">## [15] BiocGenerics_0.48.1         MatrixGenerics_1.14.0      </span></span>
<span><span class="co">## [17] matrixStats_1.2.0           BiocStyle_2.30.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] sass_0.4.8              bitops_1.0-7            stringi_1.8.3          </span></span>
<span><span class="co">##  [4] lattice_0.22-5          digest_0.6.34           magrittr_2.0.3         </span></span>
<span><span class="co">##  [7] evaluate_0.23           grid_4.3.2              bookdown_0.37          </span></span>
<span><span class="co">## [10] fastmap_1.1.1           jsonlite_1.8.8          BiocManager_1.30.22    </span></span>
<span><span class="co">## [13] purrr_1.0.2             textshaping_0.3.7       jquerylib_0.1.4        </span></span>
<span><span class="co">## [16] cli_3.6.2               rlang_1.1.3             crayon_1.5.2           </span></span>
<span><span class="co">## [19] XVector_0.42.0          cachem_1.0.8            yaml_2.3.8             </span></span>
<span><span class="co">## [22] tools_4.3.2             memoise_2.0.1           Rhdf5lib_1.24.1        </span></span>
<span><span class="co">## [25] GenomeInfoDbData_1.2.11 vctrs_0.6.5             R6_2.5.1               </span></span>
<span><span class="co">## [28] lifecycle_1.0.4         zlibbioc_1.48.0         stringr_1.5.1          </span></span>
<span><span class="co">## [31] fs_1.6.3                ragg_1.2.7              desc_1.4.3             </span></span>
<span><span class="co">## [34] pkgdown_2.0.7           bslib_0.6.1             glue_1.7.0             </span></span>
<span><span class="co">## [37] systemfonts_1.0.5       xfun_0.41               rhdf5filters_1.14.1    </span></span>
<span><span class="co">## [40] knitr_1.45              htmltools_0.5.7         rmarkdown_2.25         </span></span>
<span><span class="co">## [43] compiler_4.3.2          RCurl_1.98-1.14</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Marcel Ramos, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
